<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jakob Danel">
<meta name="author" content="Federick Bruch">
<meta name="dcterms.date" content="2024-02-15">

<title>Evaluating Tree Species Diversity in Forest Ecosystems Using LiDAR Data: A Exploration in NRW</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body>

<div id="quarto-search-results"></div>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="2">
    <h2 id="toc-title">Contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods"><span class="header-section-number">2</span> Methods</a>
  <ul class="collapse">
  <li><a href="#data-acquisition" id="toc-data-acquisition" class="nav-link" data-scroll-target="#data-acquisition"><span class="header-section-number">2.1</span> Data acquisition</a></li>
  <li><a href="#preprocessing" id="toc-preprocessing" class="nav-link" data-scroll-target="#preprocessing"><span class="header-section-number">2.2</span> Preprocessing</a></li>
  <li><a href="#analysis-of-different-distributions" id="toc-analysis-of-different-distributions" class="nav-link" data-scroll-target="#analysis-of-different-distributions"><span class="header-section-number">2.3</span> Analysis of different distributions</a></li>
  <li><a href="#random-forest-for-predicting-species" id="toc-random-forest-for-predicting-species" class="nav-link" data-scroll-target="#random-forest-for-predicting-species"><span class="header-section-number">2.4</span> Random Forest for predicting species</a></li>
  <li><a href="#implementation" id="toc-implementation" class="nav-link" data-scroll-target="#implementation"><span class="header-section-number">2.5</span> Implementation</a></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results"><span class="header-section-number">3</span> Results</a>
  <ul class="collapse">
  <li><a href="#researched-areas" id="toc-researched-areas" class="nav-link" data-scroll-target="#researched-areas"><span class="header-section-number">3.1</span> Researched areas</a></li>
  <li><a href="#distribution-of-tree-characteristics" id="toc-distribution-of-tree-characteristics" class="nav-link" data-scroll-target="#distribution-of-tree-characteristics"><span class="header-section-number">3.2</span> Distribution of tree characteristics</a>
  <ul class="collapse">
  <li><a href="#tree-heights" id="toc-tree-heights" class="nav-link" data-scroll-target="#tree-heights"><span class="header-section-number">3.2.1</span> Tree Heights</a></li>
  <li><a href="#number-of-returns" id="toc-number-of-returns" class="nav-link" data-scroll-target="#number-of-returns"><span class="header-section-number">3.2.2</span> Number of returns</a></li>
  <li><a href="#n-nearest-neighbours" id="toc-n-nearest-neighbours" class="nav-link" data-scroll-target="#n-nearest-neighbours"><span class="header-section-number">3.2.3</span> n-nearest Neighbours</a></li>
  <li><a href="#density-of-forest-patches" id="toc-density-of-forest-patches" class="nav-link" data-scroll-target="#density-of-forest-patches"><span class="header-section-number">3.2.4</span> Density of forest patches</a></li>
  <li><a href="#canopy-height-model" id="toc-canopy-height-model" class="nav-link" data-scroll-target="#canopy-height-model"><span class="header-section-number">3.2.5</span> Canopy Height Model</a></li>
  </ul></li>
  <li><a href="#random-forest-predictions" id="toc-random-forest-predictions" class="nav-link" data-scroll-target="#random-forest-predictions"><span class="header-section-number">3.3</span> Random Forest Predictions</a>
  <ul class="collapse">
  <li><a href="#use-neighbors-and-height" id="toc-use-neighbors-and-height" class="nav-link" data-scroll-target="#use-neighbors-and-height"><span class="header-section-number">3.3.1</span> Use neighbors and height</a></li>
  <li><a href="#enrich-neighbors-with-segmentation-data" id="toc-enrich-neighbors-with-segmentation-data" class="nav-link" data-scroll-target="#enrich-neighbors-with-segmentation-data"><span class="header-section-number">3.3.2</span> Enrich Neighbors with segmentation data</a></li>
  <li><a href="#train-with-patch-level-information" id="toc-train-with-patch-level-information" class="nav-link" data-scroll-target="#train-with-patch-level-information"><span class="header-section-number">3.3.3</span> Train with patch level information</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion"><span class="header-section-number">4</span> Discussion</a>
  <ul class="collapse">
  <li><a href="#findings" id="toc-findings" class="nav-link" data-scroll-target="#findings"><span class="header-section-number">4.1</span> Findings</a></li>
  <li><a href="#limitations" id="toc-limitations" class="nav-link" data-scroll-target="#limitations"><span class="header-section-number">4.2</span> Limitations</a></li>
  <li><a href="#further-work" id="toc-further-work" class="nav-link" data-scroll-target="#further-work"><span class="header-section-number">4.3</span> Further Work</a></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion"><span class="header-section-number">4.4</span> Conclusion</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"><span class="header-section-number">5</span> References</a></li>
  <li><a href="#appendix" id="toc-appendix" class="nav-link" data-scroll-target="#appendix"><span class="header-section-number">6</span> Appendix</a>
  <ul class="collapse">
  <li><a href="#contributions" id="toc-contributions" class="nav-link" data-scroll-target="#contributions"><span class="header-section-number">6.1</span> Contributions</a>
  <ul class="collapse">
  <li><a href="#frederick" id="toc-frederick" class="nav-link" data-scroll-target="#frederick"><span class="header-section-number">6.1.1</span> Frederick</a></li>
  <li><a href="#jakob" id="toc-jakob" class="nav-link" data-scroll-target="#jakob"><span class="header-section-number">6.1.2</span> Jakob</a></li>
  </ul></li>
  <li><a href="#sec-appendix-preprocessing" id="toc-sec-appendix-preprocessing" class="nav-link" data-scroll-target="#sec-appendix-preprocessing"><span class="header-section-number">6.2</span> Script which can be used to do all preprocessing</a></li>
  <li><a href="#sec-appendix-chms" id="toc-sec-appendix-chms" class="nav-link" data-scroll-target="#sec-appendix-chms"><span class="header-section-number">6.3</span> Canopy Height Models</a></li>
  <li><a href="#quantitative-results" id="toc-quantitative-results" class="nav-link" data-scroll-target="#quantitative-results"><span class="header-section-number">6.4</span> Quantitative Results</a>
  <ul class="collapse">
  <li><a href="#distribution-of-z-values" id="toc-distribution-of-z-values" class="nav-link" data-scroll-target="#distribution-of-z-values"><span class="header-section-number">6.4.1</span> Distribution of Z-Values</a></li>
  <li><a href="#nearest-neighbours" id="toc-nearest-neighbours" class="nav-link" data-scroll-target="#nearest-neighbours"><span class="header-section-number">6.4.2</span> Nearest Neighbours</a></li>
  <li><a href="#distribution-of-the-number-of-returns" id="toc-distribution-of-the-number-of-returns" class="nav-link" data-scroll-target="#distribution-of-the-number-of-returns"><span class="header-section-number">6.4.3</span> Distribution of the number of returns</a></li>
  </ul></li>
  <li><a href="#sec-documentation" id="toc-sec-documentation" class="nav-link" data-scroll-target="#sec-documentation"><span class="header-section-number">6.5</span> Documentation</a>
  <ul class="collapse">
  <li><a href="#lfa_calculate_patch_density" id="toc-lfa_calculate_patch_density" class="nav-link" data-scroll-target="#lfa_calculate_patch_density"><span class="header-section-number">6.5.1</span> <code>lfa_calculate_patch_density</code></a></li>
  <li><a href="#lfa_calculate_rf_metrics" id="toc-lfa_calculate_rf_metrics" class="nav-link" data-scroll-target="#lfa_calculate_rf_metrics"><span class="header-section-number">6.5.2</span> <code>lfa_calculate_rf_metrics</code></a></li>
  <li><a href="#lfa_capitalize_first_char" id="toc-lfa_capitalize_first_char" class="nav-link" data-scroll-target="#lfa_capitalize_first_char"><span class="header-section-number">6.5.3</span> <code>lfa_capitalize_first_char</code></a></li>
  <li><a href="#lfa_check_flag" id="toc-lfa_check_flag" class="nav-link" data-scroll-target="#lfa_check_flag"><span class="header-section-number">6.5.4</span> <code>lfa_check_flag</code></a></li>
  <li><a href="#lfa_chm" id="toc-lfa_chm" class="nav-link" data-scroll-target="#lfa_chm"><span class="header-section-number">6.5.5</span> <code>lfa_chm</code></a></li>
  <li><a href="#lfa_combine_sf_obj" id="toc-lfa_combine_sf_obj" class="nav-link" data-scroll-target="#lfa_combine_sf_obj"><span class="header-section-number">6.5.6</span> <code>lfa_combine_sf_obj</code></a></li>
  <li><a href="#lfa_count_returns_all_areas" id="toc-lfa_count_returns_all_areas" class="nav-link" data-scroll-target="#lfa_count_returns_all_areas"><span class="header-section-number">6.5.7</span> <code>lfa_count_returns_all_areas</code></a></li>
  <li><a href="#lfa_count_returns_per_tree" id="toc-lfa_count_returns_per_tree" class="nav-link" data-scroll-target="#lfa_count_returns_per_tree"><span class="header-section-number">6.5.8</span> <code>lfa_count_returns_per_tree</code></a></li>
  <li><a href="#lfa_create_boxplot" id="toc-lfa_create_boxplot" class="nav-link" data-scroll-target="#lfa_create_boxplot"><span class="header-section-number">6.5.9</span> <code>lfa_create_boxplot</code></a></li>
  <li><a href="#lfa_create_density_plots" id="toc-lfa_create_density_plots" class="nav-link" data-scroll-target="#lfa_create_density_plots"><span class="header-section-number">6.5.10</span> <code>lfa_create_density_plots</code></a></li>
  <li><a href="#lfa_create_grouped_bar_plot" id="toc-lfa_create_grouped_bar_plot" class="nav-link" data-scroll-target="#lfa_create_grouped_bar_plot"><span class="header-section-number">6.5.11</span> <code>lfa_create_grouped_bar_plot</code></a></li>
  <li><a href="#lfa_create_neighbor_mean_curves" id="toc-lfa_create_neighbor_mean_curves" class="nav-link" data-scroll-target="#lfa_create_neighbor_mean_curves"><span class="header-section-number">6.5.12</span> <code>lfa_create_neighbor_mean_curves</code></a></li>
  <li><a href="#lfa_create_plot_per_area" id="toc-lfa_create_plot_per_area" class="nav-link" data-scroll-target="#lfa_create_plot_per_area"><span class="header-section-number">6.5.13</span> <code>lfa_create_plot_per_area</code></a></li>
  <li><a href="#lfa_create_ppp_from_area" id="toc-lfa_create_ppp_from_area" class="nav-link" data-scroll-target="#lfa_create_ppp_from_area"><span class="header-section-number">6.5.14</span> <code>lfa_create_ppp_from_area</code></a></li>
  <li><a href="#lfa_create_stacked_distributions_plot" id="toc-lfa_create_stacked_distributions_plot" class="nav-link" data-scroll-target="#lfa_create_stacked_distributions_plot"><span class="header-section-number">6.5.15</span> <code>lfa_create_stacked_distributions_plot</code></a></li>
  <li><a href="#lfa_create_stacked_histogram" id="toc-lfa_create_stacked_histogram" class="nav-link" data-scroll-target="#lfa_create_stacked_histogram"><span class="header-section-number">6.5.16</span> <code>lfa_create_stacked_histogram</code></a></li>
  <li><a href="#lfa_create_tile_location_objects" id="toc-lfa_create_tile_location_objects" class="nav-link" data-scroll-target="#lfa_create_tile_location_objects"><span class="header-section-number">6.5.17</span> <code>lfa_create_tile_location_objects</code></a></li>
  <li><a href="#lfa_detection" id="toc-lfa_detection" class="nav-link" data-scroll-target="#lfa_detection"><span class="header-section-number">6.5.18</span> <code>lfa_detection</code></a></li>
  <li><a href="#lfa_download_areas" id="toc-lfa_download_areas" class="nav-link" data-scroll-target="#lfa_download_areas"><span class="header-section-number">6.5.19</span> <code>lfa_download_areas</code></a></li>
  <li><a href="#lfa_download" id="toc-lfa_download" class="nav-link" data-scroll-target="#lfa_download"><span class="header-section-number">6.5.20</span> <code>lfa_download</code></a></li>
  <li><a href="#lfa_find_n_nearest_trees" id="toc-lfa_find_n_nearest_trees" class="nav-link" data-scroll-target="#lfa_find_n_nearest_trees"><span class="header-section-number">6.5.21</span> <code>lfa_find_n_nearest_trees</code></a></li>
  <li><a href="#lfa_generate_result_table_tests" id="toc-lfa_generate_result_table_tests" class="nav-link" data-scroll-target="#lfa_generate_result_table_tests"><span class="header-section-number">6.5.22</span> <code>lfa_generate_result_table_tests</code></a></li>
  <li><a href="#lfa_get_all_areas" id="toc-lfa_get_all_areas" class="nav-link" data-scroll-target="#lfa_get_all_areas"><span class="header-section-number">6.5.23</span> <code>lfa_get_all_areas</code></a></li>
  <li><a href="#lfa_get_detection_area" id="toc-lfa_get_detection_area" class="nav-link" data-scroll-target="#lfa_get_detection_area"><span class="header-section-number">6.5.24</span> <code>lfa_get_detection_area</code></a></li>
  <li><a href="#lfa_get_detections_species" id="toc-lfa_get_detections_species" class="nav-link" data-scroll-target="#lfa_get_detections_species"><span class="header-section-number">6.5.25</span> <code>lfa_get_detections_species</code></a></li>
  <li><a href="#lfa_get_detections" id="toc-lfa_get_detections" class="nav-link" data-scroll-target="#lfa_get_detections"><span class="header-section-number">6.5.26</span> <code>lfa_get_detections</code></a></li>
  <li><a href="#lfa_get_flag_path" id="toc-lfa_get_flag_path" class="nav-link" data-scroll-target="#lfa_get_flag_path"><span class="header-section-number">6.5.27</span> <code>lfa_get_flag_path</code></a></li>
  <li><a href="#lfa_get_neighbor_paths" id="toc-lfa_get_neighbor_paths" class="nav-link" data-scroll-target="#lfa_get_neighbor_paths"><span class="header-section-number">6.5.28</span> <code>lfa_get_neighbor_paths</code></a></li>
  <li><a href="#lfa_get_species" id="toc-lfa_get_species" class="nav-link" data-scroll-target="#lfa_get_species"><span class="header-section-number">6.5.29</span> <code>lfa_get_species</code></a></li>
  <li><a href="#lfa_ground_correction" id="toc-lfa_ground_correction" class="nav-link" data-scroll-target="#lfa_ground_correction"><span class="header-section-number">6.5.30</span> <code>lfa_ground_correction</code></a></li>
  <li><a href="#lfa_init_data_structure" id="toc-lfa_init_data_structure" class="nav-link" data-scroll-target="#lfa_init_data_structure"><span class="header-section-number">6.5.31</span> <code>lfa_init_data_structure</code></a></li>
  <li><a href="#lfa_init" id="toc-lfa_init" class="nav-link" data-scroll-target="#lfa_init"><span class="header-section-number">6.5.32</span> <code>lfa_init</code></a></li>
  <li><a href="#lfa_intersect_areas" id="toc-lfa_intersect_areas" class="nav-link" data-scroll-target="#lfa_intersect_areas"><span class="header-section-number">6.5.33</span> <code>lfa_intersect_areas</code></a></li>
  <li><a href="#lfa_jsd_from_vec" id="toc-lfa_jsd_from_vec" class="nav-link" data-scroll-target="#lfa_jsd_from_vec"><span class="header-section-number">6.5.34</span> <code>lfa_jsd_from_vec</code></a></li>
  <li><a href="#lfa_jsd" id="toc-lfa_jsd" class="nav-link" data-scroll-target="#lfa_jsd"><span class="header-section-number">6.5.35</span> <code>lfa_jsd</code></a></li>
  <li><a href="#lfa_kld_from_vec" id="toc-lfa_kld_from_vec" class="nav-link" data-scroll-target="#lfa_kld_from_vec"><span class="header-section-number">6.5.36</span> <code>lfa_kld_from_vec</code></a></li>
  <li><a href="#lfa_kld" id="toc-lfa_kld" class="nav-link" data-scroll-target="#lfa_kld"><span class="header-section-number">6.5.37</span> <code>lfa_kld</code></a></li>
  <li><a href="#lfa_ks_test" id="toc-lfa_ks_test" class="nav-link" data-scroll-target="#lfa_ks_test"><span class="header-section-number">6.5.38</span> <code>lfa_ks_test</code></a></li>
  <li><a href="#lfa_load_ctg_if_not_present" id="toc-lfa_load_ctg_if_not_present" class="nav-link" data-scroll-target="#lfa_load_ctg_if_not_present"><span class="header-section-number">6.5.39</span> <code>lfa_load_ctg_if_not_present</code></a></li>
  <li><a href="#lfa_map_tile_locations" id="toc-lfa_map_tile_locations" class="nav-link" data-scroll-target="#lfa_map_tile_locations"><span class="header-section-number">6.5.40</span> <code>lfa_map_tile_locations</code></a></li>
  <li><a href="#lfa_merge_and_save" id="toc-lfa_merge_and_save" class="nav-link" data-scroll-target="#lfa_merge_and_save"><span class="header-section-number">6.5.41</span> <code>lfa_merge_and_save</code></a></li>
  <li><a href="#lfa_plot_confusion_matrix" id="toc-lfa_plot_confusion_matrix" class="nav-link" data-scroll-target="#lfa_plot_confusion_matrix"><span class="header-section-number">6.5.42</span> <code>lfa_plot_confusion_matrix</code></a></li>
  <li><a href="#lfa_precision_per_class" id="toc-lfa_precision_per_class" class="nav-link" data-scroll-target="#lfa_precision_per_class"><span class="header-section-number">6.5.43</span> <code>lfa_precision_per_class</code></a></li>
  <li><a href="#lfa_random_forest" id="toc-lfa_random_forest" class="nav-link" data-scroll-target="#lfa_random_forest"><span class="header-section-number">6.5.44</span> <code>lfa_random_forest</code></a></li>
  <li><a href="#lfa_rasterize_chunk" id="toc-lfa_rasterize_chunk" class="nav-link" data-scroll-target="#lfa_rasterize_chunk"><span class="header-section-number">6.5.45</span> <code>lfa_rasterize_chunk</code></a></li>
  <li><a href="#lfa_rd_to_qmd" id="toc-lfa_rd_to_qmd" class="nav-link" data-scroll-target="#lfa_rd_to_qmd"><span class="header-section-number">6.5.46</span> <code>lfa_rd_to_qmd</code></a></li>
  <li><a href="#lfa_rd_to_results" id="toc-lfa_rd_to_results" class="nav-link" data-scroll-target="#lfa_rd_to_results"><span class="header-section-number">6.5.47</span> <code>lfa_rd_to_results</code></a></li>
  <li><a href="#lfa_read_area_as_catalog" id="toc-lfa_read_area_as_catalog" class="nav-link" data-scroll-target="#lfa_read_area_as_catalog"><span class="header-section-number">6.5.48</span> <code>lfa_read_area_as_catalog</code></a></li>
  <li><a href="#lfa_recall_per_class" id="toc-lfa_recall_per_class" class="nav-link" data-scroll-target="#lfa_recall_per_class"><span class="header-section-number">6.5.49</span> <code>lfa_recall_per_class</code></a></li>
  <li><a href="#lfa_run_test_asymmetric" id="toc-lfa_run_test_asymmetric" class="nav-link" data-scroll-target="#lfa_run_test_asymmetric"><span class="header-section-number">6.5.50</span> <code>lfa_run_test_asymmetric</code></a></li>
  <li><a href="#lfa_run_test_symmetric" id="toc-lfa_run_test_symmetric" class="nav-link" data-scroll-target="#lfa_run_test_symmetric"><span class="header-section-number">6.5.51</span> <code>lfa_run_test_symmetric</code></a></li>
  <li><a href="#lfa_save_all_neighbours" id="toc-lfa_save_all_neighbours" class="nav-link" data-scroll-target="#lfa_save_all_neighbours"><span class="header-section-number">6.5.52</span> <code>lfa_save_all_neighbours</code></a></li>
  <li><a href="#lfa_segmentation" id="toc-lfa_segmentation" class="nav-link" data-scroll-target="#lfa_segmentation"><span class="header-section-number">6.5.53</span> <code>lfa_segmentation</code></a></li>
  <li><a href="#lfa_set_flag" id="toc-lfa_set_flag" class="nav-link" data-scroll-target="#lfa_set_flag"><span class="header-section-number">6.5.54</span> <code>lfa_set_flag</code></a></li>
  <li><a href="#lfa_visit_all_areas" id="toc-lfa_visit_all_areas" class="nav-link" data-scroll-target="#lfa_visit_all_areas"><span class="header-section-number">6.5.55</span> <code>lfa_visit_all_areas</code></a></li>
  <li><a href="#lfa_visualize_rf_metrics" id="toc-lfa_visualize_rf_metrics" class="nav-link" data-scroll-target="#lfa_visualize_rf_metrics"><span class="header-section-number">6.5.56</span> <code>lfa_visualize_rf_metrics</code></a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Evaluating Tree Species Diversity in Forest Ecosystems Using LiDAR Data: A Exploration in NRW</h1>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Authors</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://github.com/jakobdanel">Jakob Danel</a> </p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            Universität Münster
          </p>
      </div>
      <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://www.uni-muenster.de/Geoinformatics/institute/staff/index.php/351/Frederick_Bruch">Federick Bruch</a> </p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            Universität Münster
          </p>
      </div>
    </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 15, 2024</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="introduction" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Introduction</h1>
<p>Forests, referred as the “world’s air-conditioning system” or the “lungs of the planet,” play a critical role in maintaining the global environment, as noted by King Charles III. Delving into forests and their diverse tree species is essential for our comprehension of both local ecosystems and the broader globe. The combination of tree species significantly contributes to the functionality and biodiversity of a forest, as emphasized by Mori et al.&nbsp;in their work on forest tree biodiversity <span class="citation" data-cites="mori2017">(<a href="#ref-mori2017" role="doc-biblioref">Mori, Lertzman, and Gustafsson 2017</a>)</span>. Different tree species face distinct threats, posing challenges to both the environment and human populations. For instance, the Oak processionary moth, highlighted by Sobczyk et al, <span class="citation" data-cites="sobczyk2014eichenprozessionsspinner">(<a href="#ref-sobczyk2014eichenprozessionsspinner" role="doc-biblioref">Sobczyk 2014</a>)</span> in their study on the impact of the oak processionary moth, and the bark beetle, discussed in the work of Müller et al.&nbsp;on bark beetle infestations <span class="citation" data-cites="muller2019kaferkampfe">(<a href="#ref-muller2019kaferkampfe" role="doc-biblioref">Müller and Imhof 2019</a>)</span>, exemplify the varying risks associated with specific tree types. Forests play a pivotal role in climate regulation by absorbing carbon dioxide (CO<span class="math inline">\(_2\)</span>) and providing habitats for a multitude of animal and plant species. Approximately 80% of all terrestrial species find their home in forests, housing more than four-fifths of all known animal and plant species outside the oceans <span class="citation" data-cites="Naturefund">(<a href="#ref-Naturefund" role="doc-biblioref">n.d.</a>)</span>. Recognizing the importance of gaining a comprehensive understanding of forests is imperative, particularly in unraveling the (spatial) distribution patterns of different tree species. Such insights are crucial for informed conservation efforts, sustainable management, and addressing the diverse ecological challenges that forests face.</p>
<p>Often, forest monitoring relies on Sentinel data, offering valuable insights through various applications such as time series analysis of the deforestation process <span class="citation" data-cites="cremer2020potential">(<a href="#ref-cremer2020potential" role="doc-biblioref">Cremer et al. 2020</a>)</span>, forest classification <span class="citation" data-cites="dostalova2021european">(<a href="#ref-dostalova2021european" role="doc-biblioref">Dostálová et al. 2021</a>)</span>, and detection of forest succession <span class="citation" data-cites="szostak2018using">(<a href="#ref-szostak2018using" role="doc-biblioref">Szostak, Hawryło, and Piela 2018</a>)</span>. However, employing Sentinel data is not without challenges. Dependency on cloud coverage and limited resolution, such as the <span class="math inline">\(5\times5\)</span>m resolution on Sentinel-1 <span class="citation" data-cites="copernicusSentinel1Missions">(<a href="#ref-copernicusSentinel1Missions" role="doc-biblioref">Agency 2024</a>)</span>, can hinder the accurate identification of individual trees. To overcome these limitations, LiDAR data emerges as a promising alternative. LiDAR provides higher resolutions and utilizes the intensity of the returned signal as an indicator of the forest structure <span class="citation" data-cites="dlrRaumfahrtagenturLidar">(<a href="#ref-dlrRaumfahrtagenturLidar" role="doc-biblioref">Gonzalez</a>)</span>. The aim is to investigate whether LiDAR is an appropriate tool for distinguishing between tree species, with a focus on oak, beech, pine, and spruce – the most common species in North Rhine-Westphalia (NRW). This choice ensures comparable environmental conditions, including climate and altitude.</p>
<p>The methodology involves employing different statistical tools to analyze the distribution of LiDAR returns. Specifically, the study utilizes the random forest algorithm to predict the species of detected trees based on LiDAR data. This approach aims to enhance the precision and capabilities of forest monitoring, especially in regions like NRW, by leveraging LiDAR technology for species-specific insights. Our motivation and methodical approach leading to the following research question:</p>
<p><strong>Can LiDAR technology be effectively employed to distinguish between tree species in monocultural forests in North Rhine-Westphalia (NRW), and how do the distinctive characteristics of individual tree species contribute to the accuracy of LiDAR-based classification?</strong></p>
<p>We formulate the following hypothesis:</p>
<ol type="1">
<li>LiDAR data can effectively differentiate between tree species in monocultural forests in NRW, and there are statistically significant differences in the LiDAR-derived metrics among the various tree species.</li>
<li>Random Forest classification can be used to predict the tree species of trees in monoculteral forests.</li>
<li>Random Forest classification performance varies significantly depending on the set of LiDAR parameters used, indicating that certain combinations of parameters contribute more effectively to accurate tree species classification.</li>
</ol>
</section>
<section id="methods" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Methods</h1>
<p>We first describe the processes of data acquisition and data preprocessing. Then we will propose how we analyse the distributions and how we utilize random forest predictions for species distinguishing.</p>
<section id="data-acquisition" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="data-acquisition"><span class="header-section-number">2.1</span> Data acquisition</h2>
<p>Our primary objective is to identify patches where one tree species exhibits a high level of dominance, striving to capture monocultural stands within the diverse forests of Nordrhein-Westfalia (NRW). Recognizing the practical challenges of finding true monocultures, we aim to identify patches where one species is highly dominant, enabling meaningful comparisons across different species.</p>
<p>The study is framed within the NRW region due to the availability of an easily accessible dataset. Our focus includes four prominent tree species in NRW: oak, beech, spruce, and pine, representing the most prevalent species in the region. To ensure the validity of our findings, we derive three patches for each species, thereby confirming that observed variables are characteristic of a particular species rather than a specific patch. Each patch is carefully selected to encompass an area of approximately 10-50 hectares and contain between 500 and 5,000 trees. Striking a balance between relevance and manageability, these patches avoid excessive size to enhance the likelihood of capturing varied species mixes and ensure compatibility with local hardware.</p>
<p>Specific Goals:</p>
<ol type="1">
<li>Retrieve patches with highly dominant tree species.</li>
<li>Minimize or eliminate the presence of human-made structures within the selected patches.</li>
</ol>
<p>To achieve our goals, we utilized the waldmonitor dataset <span class="citation" data-cites="welle2014">(<a href="#ref-welle2014" role="doc-biblioref">Welle et al. 2022</a>)</span> and the map provided by <span class="citation" data-cites="Blickensdoerfer2022">(<a href="#ref-Blickensdoerfer2022" role="doc-biblioref">Blickensdoerfer 2022</a>)</span>, both indicating dominant tree species in NRW. We identified patches of feasible size where both sources predicted the presence of a specific species. Further validation involved examining sentinel images of these forest regions to assess the evenness of structures, leaf color distribution, and the absence of significant human-made structures such as roads or buildings. The subsequent preprocessing steps, detailed in the following subsection, involved refining our selected patches and deriving relevant variables, such as tree distribution and density, to ensure that the chosen areas align with the desired research domains.</p>
</section>
<section id="preprocessing" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="preprocessing"><span class="header-section-number">2.2</span> Preprocessing</h2>
<p>In this research study, the management and processing of a large dataset are crucial considerations. The dataset’s substantial size necessitates careful maintenance to ensure efficient handling. Furthermore, the data should be easily processable and editable to facilitate necessary corrections and precalculations within the context of our research objectives. To achieve our goals, we have implemented a framework that automatically derives data based on a shapefile, delineating areas of interest. The processed data and results of precalculations are stored in a straightforward manner to enhance accessibility. Additionally, we have designed functions that establish a user-friendly interface, enabling the execution of algorithms on subsets of the data, such as distinct species. These interfaces are not only directly callable by users but can also be integrated into other functions to automate processes. The overarching aim is to streamline the entire preprocessing workflow using a single script, leveraging only the shapefile as a basis. This subsection details the accomplishments of our R-package in realizing these goals, outlining the preprocessing steps undertaken and justifying their necessity in the context of our research.</p>
<p>The data are stored in a data subdirectory of the root directory in the format <code>species/location-name/tile-name</code>. To automate the matching of areas of interest with the catalog from the Land NRW<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>, we utilize the intersecting tool developed by Heisig<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>. This tool, allows for the automatic retrieval and placement of data downloaded from the Land NRW catalog. To enhance data accessibility, we have devised an object that incorporates species, location name, and tile name (the NRW internal identifier) for each area this object facilitates the specification of the area to be processed. Additionally, we have defined an initialization function that downloads all tiles, returning a list of tile location objects for subsequent processing. A pivotal component of the package’s preprocessing functionality is the map function, which iterates over a list of tile locations (effectively the entire dataset) and accepts a processing function as an argument. The subsequent paragraph outlines the specific preprocessing steps employed, all of which are implemented within the mapping function.</p>
<p>To facilitate memory-handling capabilities, each of the tiles, where one area can span multiple tiles, has been split into manageable chunks. We employed a <span class="math inline">\(50\times50\)</span>m size for each tile, resulting in the division of original <span class="math inline">\(1\times1\)</span>km files into 400 tiles. These tiles are stored in our directory structure, with each tile housed in a directory named after its tile name and assigned an id as the filename. Implementation-wise, the <code>lidr::catalog_retile</code> function was instrumental in achieving this segmentation. The resulting smaller chunks allow for efficient iteration during subsequent preprocessing steps.</p>
<p>The next phase involves reducing our data to the actual size by intersecting the tiles with the defined area of interest. Using the <code>lidR::merge_spatial</code> function, we intersect the area derived from the shapefile, removing all point cloud items outside this region. Due to our tile-wise approach, empty tiles may arise, and in such cases, those tiles are simply deleted.</p>
<p>Following the size reduction to our dataset, the next step involves correcting the <code>z</code> values. The <code>z</code> values in the data are originally relative to the ellipsoid used for referencing, but we require them to be relative to the ground. To achieve this, we utilize the <code>lidR::tin</code> function, which extrapolates a convex hull between all ground points (classified by the data provider) and calculates the z value based on this structure.</p>
<p>Subsequently, we aim to perform segmentation for each distinct tree, marking each item of the point cloud with a tree ID. We employ the algorithm described by <span class="citation" data-cites="li2012">Li et al. (<a href="#ref-li2012" role="doc-biblioref">2012</a>)</span>, using parameters <code>li2012(dt1 = 2, dt2 = 3, R = 2, Zu = 10, hmin = 5, speed_up = 12)</code>. The meanings of these parameters are elucidated in Li et al.’s work <span class="citation" data-cites="li2012">(<a href="#ref-li2012" role="doc-biblioref">Li et al. 2012</a>)</span>.</p>
<p>Finally, the last preprocessing step involves individual tree detection, seeking a single <code>POINT</code> object for each tree. The <code>lidR::lmf</code> function, an implementation of the tree data using a local maximum approach, is utilized for this purpose <span class="citation" data-cites="popescu2004">(<a href="#ref-popescu2004" role="doc-biblioref">Popescu and Wynne 2004</a>)</span>. The results are stored in GeoPackage files within our data structure.</p>
<p>See <a href="#sec-appendix-preprocessing">Section&nbsp;6.2</a> for the implementation of the preprocessing.</p>
</section>
<section id="analysis-of-different-distributions" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="analysis-of-different-distributions"><span class="header-section-number">2.3</span> Analysis of different distributions</h2>
<p>Analysis of data distributions is a critical aspect of our research, with a focus on comparing two or more distributions. Our objective extends beyond evaluating the disparities between species; we also aim to assess differences within a species. To gain a comprehensive understanding of the data, we employ various visualization techniques, including histograms, density functions, and box plots.</p>
<p>In tandem with visualizations, descriptive statistics, such as means, standard errors, and quantiles, are leveraged to provide key insights into the central tendency and variability of the data.</p>
<p>For a more quantitative analysis of distribution dissimilarity, statistical tests are employed. The Kullback-Leibler (KL) difference serves as a measure to compare the similarity of a set of distributions. This involves converting distributions into their density functions, with the standard error serving as the bandwidth. The KL difference is calculated for each pair of distributions, as it is asymmetric. For the two distributions the KL difference is defined as following <span class="citation" data-cites="kullback1951kullback">(<a href="#ref-kullback1951kullback" role="doc-biblioref">Kullback 1951</a>)</span>:</p>
<p><span class="math display">\[
D_{KL}(P \, \| \, Q) = \sum_i P(i) \log\left(\frac{P(i)}{Q(i)}\right)
\]</span></p>
<p>To obtain a symmetric score, the Jensen-Shannon Divergence (JSD) is utilized <span class="citation" data-cites="grosse2002analysis">(<a href="#ref-grosse2002analysis" role="doc-biblioref">Grosse et al. 2002</a>)</span>, expressed by the formula:</p>
<p><span class="math display">\[
JS(P || Q) = \frac{1}{2} * KL(P || M) + \frac{1}{2} * KL(Q || M)
\]</span> Here, <span class="math inline">\(M = \frac{1}{2} * (P + Q)\)</span>. The JSD provides a balanced measure of dissimilarity between distributions <span class="citation" data-cites="Brownlee2019Calculate">(<a href="#ref-Brownlee2019Calculate" role="doc-biblioref">Brownlee 2019</a>)</span>. For comparing the different scores to each other, we will use averages.</p>
<p>Additionally, the Kolmogorov-Smirnov Test is implemented to assess whether two distributions significantly differ from each other. This statistical test offers a formal evaluation of the dissimilarity between empirical distribution functions.</p>
</section>
<section id="random-forest-for-predicting-species" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="random-forest-for-predicting-species"><span class="header-section-number">2.4</span> Random Forest for predicting species</h2>
<p>The aim of our research is to investigate the feasibility of developing a predictive model for tree species classification based on derived parameters. To accomplish this objective, we employed the Random Forest algorithm, as introduced by <span class="citation" data-cites="breiman2001random">(<a href="#ref-breiman2001random" role="doc-biblioref">Breiman 2001</a>)</span>, which leverages multiple decision trees to predict the species of a given tree based on its characteristics.</p>
<p>The dataset utilized in our study consists of identified trees, and our primary goal is to predict the species of each tree. To establish ground truth for training and validation of the model, we assigned the dominant species to each distinct patch.</p>
<p>For validation purposes, we adopted a spatial cross-validation methodology, specifically implementing the leave-one-out principle. This technique systematically excludes a particular portion of the dataset in each iteration <span class="citation" data-cites="brovelli2008accuracy">(<a href="#ref-brovelli2008accuracy" role="doc-biblioref">Brovelli et al. 2008</a>)</span>. In our spatial cross-validation, the group left out is determined based on their spatial characteristics <span class="citation" data-cites="valavi2018blockcv">(<a href="#ref-valavi2018blockcv" role="doc-biblioref">Valavi et al. 2018</a>)</span>.</p>
<p>Our combined approach entails designating the patches with tree detections as the left-out group. The model is trained on the training split and validated using the left-out (validation) split of the data. Ground truth and predictions from each iteration are stored, and the results from all iterations are aggregated into a confusion matrix for a comprehensive analysis.</p>
<p>In our evaluation of results, we utilized various statistical measurements. The <em>accuracy</em>, defined as the ratio of correct predictions to all predictions <span class="citation" data-cites="alvarez2002exact">(<a href="#ref-alvarez2002exact" role="doc-biblioref">Alvarez 2002</a>)</span>, is calculated using the formula <span class="math display">\[\text{Accuracy} = \frac{\text{Correct Predictions}}{\text{All Predictions}}\]</span> This metric provides insight into the percentage of correct predictions.</p>
<p>To normalize accuracy in the context of a small number of classes (<span class="math inline">\(n = 4\)</span>), Kohen’s Kappa Index was employed <span class="citation" data-cites="fleiss1969large">(<a href="#ref-fleiss1969large" role="doc-biblioref">Fleiss, Cohen, and Everitt 1969</a>)</span>. The formula is given by <span class="math display">\[\kappa = \frac{P_o - P_e}{1 - P_e}\]</span> where <span class="math inline">\(P_o\)</span> represents the observed agreement between raters, and <span class="math inline">\(P_e\)</span> is the expected agreement, accounting for the probability of chance agreement.</p>
<p>Another crucial metric, <em>precision</em>, commonly used in binary classification, measures the accuracy of positive predictions <span class="citation" data-cites="cleverdon1967cranfield">(<a href="#ref-cleverdon1967cranfield" role="doc-biblioref">Cleverdon 1967</a>)</span>. It is calculated as <span class="math display">\[\text{Precision} = \frac{\text{True Positives}}{\text{True Positives + False Positives}}\]</span></p>
<p>Similarly, <em>recall</em>, also known as sensitivity or true positive rate, is essential in binary classification to assess the model’s ability to identify all relevant instances <span class="citation" data-cites="sparck1972statistical">(<a href="#ref-sparck1972statistical" role="doc-biblioref">Sparck Jones 1972</a>)</span>. The formula for recall is <span class="math display">\[\text{Recall} = \frac{\text{True Positives}}{\text{True Positives + False Negatives}}\]</span></p>
<p>These statistical measures collectively provide a comprehensive evaluation of the model’s performance, taking into account both overall accuracy and class-specific performance metrics.</p>
</section>
<section id="implementation" class="level2" data-number="2.5">
<h2 data-number="2.5" class="anchored" data-anchor-id="implementation"><span class="header-section-number">2.5</span> Implementation</h2>
<p>In the implementation of our project aimed at utilizing LiDAR data for distinguishing tree species in monocultural forests, we chose R as our programming language. This decision was driven by the robust and well-maintained package infrastructure offered by R, providing a solid foundation for our work. Additionally, R’s inherent capability to handle large data structures efficiently proved advantageous for our project, where processing LiDAR point clouds required a scalable and effective solution.</p>
<p>Our implementation relied on four key R packages, each serving a crucial role in the project’s success:</p>
<ul>
<li><p><strong>lidR:</strong> This package played a fundamental role in handling and processing LiDAR point clouds. Its capabilities were essential for the core management of LiDAR data throughout our analysis <span class="citation" data-cites="lidR">(<a href="#ref-lidR" role="doc-biblioref">Roussel et al. 2020</a>)</span>.</p></li>
<li><p><strong>sf:</strong> Used for managing spatial vector data and conducting spatial operations, the sf package enriched our analysis with essential spatial components, enhancing the geographical dimension of our research <span class="citation" data-cites="sf">(<a href="#ref-sf" role="doc-biblioref">Pebesma 2018</a>)</span>.</p></li>
<li><p><strong>dplyr:</strong> As a package specialized in data manipulation, dplyr provided indispensable tools for shaping and organizing our data to meet the specific requirements of our analysis <span class="citation" data-cites="dplyr">(<a href="#ref-dplyr" role="doc-biblioref">Wickham et al. 2023</a>)</span>.</p></li>
<li><p><strong>ggplot2:</strong> Renowned for its proficiency in creating high-quality visualizations, ggplot2 was employed to present our results in a manner that is not only informative but also reproducible <span class="citation" data-cites="ggplot2">(<a href="#ref-ggplot2" role="doc-biblioref">Wickham 2016</a>)</span>.</p></li>
</ul>
<p>To streamline the implementation process and make our work accessible to others, we encapsulated these functionalities into a comprehensive R package. This package handles various aspects of the project seamlessly:</p>
<ul>
<li><p><strong>Automated Data Management:</strong> The R package takes care of organizing and saving data in a predefined directory structure, ensuring a systematic and orderly approach.</p></li>
<li><p><strong>Meta Information Generation:</strong> It automatically generates meta-information based on the structure of the data, providing vital context and documentation for our analyses.</p></li>
<li><p><strong>Analytic Function Application:</strong> Analytic functions are applied automatically to specified data subsets, streamlining the analytical process and enhancing the reproducibility of our results.</p></li>
</ul>
<p>For those interested in exploring or contributing to our work, the GitHub repository containing the implementations can be accessed <a href="https://github.com/jakobdanel/lidar-forest-analysis">here</a>. This centralized repository serves as a collaborative hub, inviting further development and utilization of our LiDAR forest analysis package <span class="citation" data-cites="lfa">(<a href="#ref-lfa" role="doc-biblioref">Danel and Bruch 2024</a>)</span>. For further investigation you can find the package Documentation here: <a href="#sec-documentation">Section&nbsp;6.5</a></p>
</section>
</section>
<section id="results" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Results</h1>
<p>First we present the researched areas, after that we look into distribution characterics. Lastly we present a set of generated random forest models and their accuracy.</p>
<section id="researched-areas" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="researched-areas"><span class="header-section-number">3.1</span> Researched areas</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>sf<span class="sc">::</span><span class="fu">sf_use_s2</span>(<span class="cn">FALSE</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>patches <span class="ot">&lt;-</span> sf<span class="sc">::</span><span class="fu">read_sf</span>(<span class="st">"research_areas.shp"</span>) <span class="sc">|&gt;</span> sf<span class="sc">::</span><span class="fu">st_centroid</span>()</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>de <span class="ot">&lt;-</span> sf<span class="sc">::</span><span class="fu">read_sf</span>(<span class="st">"results/results/states_de/Bundesländer_2017_mit_Einwohnerzahl.shp"</span>) <span class="co"># Source: https://hub.arcgis.com/datasets/esri-de-content::bundesl%C3%A4nder-2017-mit-einwohnerzahl/explore?location=51.099647%2C10.454033%2C7.43</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>nrw <span class="ot">&lt;-</span> de[<span class="dv">5</span>,] <span class="sc">|&gt;</span> sf<span class="sc">::</span><span class="fu">st_geometry</span>()</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span> <span class="fu">geom_sf</span>(<span class="at">data =</span> nrw) <span class="sc">+</span> </span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_sf</span>(<span class="at">data =</span> patches, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">col =</span> species))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-patches-nrw" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-patches-nrw-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;1: Locations of the different patches with the dominant species for that patch. The patches centroids are displayed on a basemap describing the borders from NRW.</figcaption>
</figure>
</div>
</div>
</div>
<p>We draw three patches for each species from different regions (see <a href="#tbl-summary-researched-areas">Table&nbsp;1</a>). We download the LiDAR data for those patches and runned all preprocessing steps as described. We than checked with certain derived parameters (e.g.&nbsp;tree heights, tree distributions or tree density) that all patches contain valid forest data. In that step we discovered, that in one patch some forest clearance took place in the near past. This patch was removed from the dataset and was replaced with a new one.</p>
<p>In our research, drawing patches evenly distributed across Nordrhein-Westfalia is inherently constrained by natural factors. Consequently, the patches for oak and pine predominantly originate from the Münsterland region, as illustrated in <a href="#fig-patches-nrw">Figure&nbsp;1</a>. For spruce, the patches were derived from Sauerland, reflecting the prevalence of spruce forests in this specific region within NRW, as corroborated by Welle et al. <span class="citation" data-cites="welle2014">(<a href="#ref-welle2014" role="doc-biblioref">Welle et al. 2022</a>)</span> and Blickensdörfer et al. <span class="citation" data-cites="Blickensdoerfer2022">(<a href="#ref-Blickensdoerfer2022" role="doc-biblioref">Blickensdoerfer 2022</a>)</span>. Beech patches, on the other hand, were generated from diverse locations within NRW. Across all patches, no human-made objects were identified, with the exception of small paths for pedestrians and forestry vehicles.</p>
<p>The distribution of area and detections is notable for each four species. Beech covers 69,791.9 hectares with a total of 5,954 detections, oak spans 63,232.49 hectares with 5,354 detections, pine extends across 72,862.4 hectares with 8,912 detections, and spruce encompasses 57,940.02 hectares with 8,619 detections. Both the amount of detections and the corresponding area exhibit a relatively uniform distribution across the diverse patches, as summarized in <a href="#tbl-summary-researched-areas">Table&nbsp;1</a>.</p>
<p>With the selected dataset described, we intentionally chose three patches for each four species that exhibit a practical and usable size for our research objectives. These carefully chosen patches align with the conditions essential for our study, providing comprehensive and representative data for in-depth analysis and meaningful insights into the characteristics of each tree species within the specified areas.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>shp <span class="ot">&lt;-</span> sf<span class="sc">::</span><span class="fu">read_sf</span>(<span class="st">"research_areas.shp"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>table <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_all_areas</span>()</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>sf<span class="sc">::</span><span class="fu">sf_use_s2</span>(<span class="cn">FALSE</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (row <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(table)) {</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  area <span class="ot">&lt;-</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(shp, shp<span class="sc">$</span>species <span class="sc">==</span> table[row, <span class="st">"specie"</span>] <span class="sc">&amp;</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>                    shp<span class="sc">$</span>name <span class="sc">==</span> table[row, <span class="st">"area"</span>])</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  area_size <span class="ot">&lt;-</span> area <span class="sc">|&gt;</span> sf<span class="sc">::</span><span class="fu">st_area</span>()</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  point <span class="ot">&lt;-</span> area <span class="sc">|&gt;</span> sf<span class="sc">::</span><span class="fu">st_centroid</span>() <span class="sc">|&gt;</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>()</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  table[row,<span class="st">"point"</span>] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"("</span>,<span class="fu">round</span>(point[<span class="dv">1</span>], <span class="at">digits =</span> <span class="dv">4</span>),<span class="st">", "</span>,<span class="fu">round</span>(point[<span class="dv">2</span>],<span class="at">digits =</span> <span class="dv">4</span>),<span class="st">")"</span>)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  table[row, <span class="st">"area_size"</span>] <span class="ot">=</span> <span class="fu">round</span>(area_size,<span class="at">digits =</span> <span class="dv">2</span>) <span class="co">#paste0(round(area_size,digits = 2), " m²")</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  amount_det <span class="ot">&lt;-</span> <span class="fu">nrow</span>(lfa<span class="sc">::</span><span class="fu">lfa_get_detection_area</span>(table[row, <span class="st">"specie"</span>], table[row, <span class="st">"area"</span>]))</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(amount_det)){</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="fu">nrow</span>(lfa<span class="sc">::</span><span class="fu">lfa_get_detection_area</span>(table[row, <span class="st">"specie"</span>], table[row, <span class="st">"area"</span>])),table[row, <span class="st">"specie"</span>],table[row, <span class="st">"area"</span>])</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>  table[row, <span class="st">"amount_detections"</span>] <span class="ot">=</span> amount_det</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># table[row, "specie"] &lt;- lfa::lfa_capitalize_first_char(table[row,"specie"])</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  table[row, <span class="st">"area"</span>] <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_capitalize_first_char</span>(table[row,<span class="st">"area"</span>])</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>table<span class="sc">$</span>area <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, table<span class="sc">$</span>area)</span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>table<span class="sc">$</span>area <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"ue"</span>, <span class="st">"ü"</span>, table<span class="sc">$</span>area)</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>table <span class="ot">=</span> table[,<span class="sc">!</span><span class="fu">names</span>(table) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"specie"</span>)]</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(table, <span class="st">"html"</span>, <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">"Patch Name"</span>,<span class="st">"Location"</span>,<span class="st">"Area size (m²)"</span>,<span class="st">"Amount tree detections"</span> ), <span class="at">caption =</span> <span class="cn">NULL</span>, <span class="at">digits =</span> <span class="dv">2</span>, <span class="at">escape =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">kable_styling</span>(</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"striped"</span>, <span class="st">"hold_position"</span>, <span class="st">"bordered"</span>,<span class="st">"responsive"</span>),</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">stripe_index =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">7</span><span class="sc">:</span><span class="dv">9</span>),</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">full_width =</span> <span class="cn">FALSE</span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">pack_rows</span>(<span class="st">"Beech"</span>, <span class="dv">1</span>, <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">pack_rows</span>(<span class="st">"Oak"</span>, <span class="dv">4</span>, <span class="dv">6</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">pack_rows</span>(<span class="st">"Pine"</span>, <span class="dv">7</span>, <span class="dv">9</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">pack_rows</span>(<span class="st">"Spruce"</span>, <span class="dv">10</span>, <span class="dv">12</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>  kableExtra<span class="sc">::</span><span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-summary-researched-areas" class="anchored">
<div class="table-responsive">
<table class="table table-striped table-bordered table-sm small" data-quarto-postprocess="true">
<caption>Table&nbsp;1: Summary of researched patches grouped by species, with their location, area and the amount of detected trees.</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th">Patch Name</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Location</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Area size (m²)</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">Amount tree detections</th>
</tr>
</thead>
<tbody>
<tr class="odd" data-grouplength="3">
<td colspan="4" style="border-bottom: 1px solid"><strong>Beech</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em; font-weight: bold;" data-indentlevel="1">Bielefeld brackwede</td>
<td style="text-align: left;">(8.5244, 51.9902)</td>
<td style="text-align: right;">161410.57</td>
<td style="text-align: right;">1443</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em; font-weight: bold;" data-indentlevel="1">Billerbeck</td>
<td style="text-align: left;">(7.3273, 51.9987)</td>
<td style="text-align: right;">185887.25</td>
<td style="text-align: right;">1732</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em; font-weight: bold;" data-indentlevel="1">Wülfenrath</td>
<td style="text-align: left;">(7.0769, 51.2917)</td>
<td style="text-align: right;">350621.21</td>
<td style="text-align: right;">2779</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="4" style="border-bottom: 1px solid"><strong>Oak</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em; font-weight: bold;" data-indentlevel="1">Hamm</td>
<td style="text-align: left;">(7.8618, 51.6639)</td>
<td style="text-align: right;">269397.22</td>
<td style="text-align: right;">2441</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em; font-weight: bold;" data-indentlevel="1">Münster</td>
<td style="text-align: left;">(7.6187, 51.9174)</td>
<td style="text-align: right;">164116.61</td>
<td style="text-align: right;">1270</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em; font-weight: bold;" data-indentlevel="1">Rinkerode</td>
<td style="text-align: left;">(7.6744, 51.8598)</td>
<td style="text-align: right;">198811.09</td>
<td style="text-align: right;">1643</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="4" style="border-bottom: 1px solid"><strong>Pine</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em; font-weight: bold;" data-indentlevel="1">Greffen</td>
<td style="text-align: left;">(8.1697, 51.9913)</td>
<td style="text-align: right;">49418.81</td>
<td style="text-align: right;">513</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em; font-weight: bold;" data-indentlevel="1">Mesum</td>
<td style="text-align: left;">(7.5403, 52.2573)</td>
<td style="text-align: right;">405072.85</td>
<td style="text-align: right;">5031</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em; font-weight: bold;" data-indentlevel="1">Telgte</td>
<td style="text-align: left;">(7.7816, 52.0024)</td>
<td style="text-align: right;">274132.34</td>
<td style="text-align: right;">3368</td>
</tr>
<tr class="odd" data-grouplength="3">
<td colspan="4" style="border-bottom: 1px solid"><strong>Spruce</strong></td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em; font-weight: bold;" data-indentlevel="1">Brilon</td>
<td style="text-align: left;">(8.5352, 51.4084)</td>
<td style="text-align: right;">211478.20</td>
<td style="text-align: right;">3342</td>
</tr>
<tr class="odd">
<td style="text-align: left; padding-left: 2em; font-weight: bold;" data-indentlevel="1">Oberhundem</td>
<td style="text-align: left;">(8.1861, 51.0909)</td>
<td style="text-align: right;">151895.53</td>
<td style="text-align: right;">2471</td>
</tr>
<tr class="even">
<td style="text-align: left; padding-left: 2em; font-weight: bold;" data-indentlevel="1">Osterwald</td>
<td style="text-align: left;">(8.3721, 51.2151)</td>
<td style="text-align: right;">216026.43</td>
<td style="text-align: right;">2806</td>
</tr>
</tbody>
</table>
</div>
</div>


</div>
</div>
</section>
<section id="distribution-of-tree-characteristics" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="distribution-of-tree-characteristics"><span class="header-section-number">3.2</span> Distribution of tree characteristics</h2>
<p>The following subsections describe the distributions of different tree characterestics across the different species and patches.</p>
<section id="tree-heights" class="level3" data-number="3.2.1">
<h3 data-number="3.2.1" class="anchored" data-anchor-id="tree-heights"><span class="header-section-number">3.2.1</span> Tree Heights</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>detections <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_detections</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>In this study, we scrutinize the distribution of tree heights, focusing initially on the density distribution to unravel the nuances across various tree species. Notably, our examination reveals distinctive patterns, with Oak and Pine exhibiting significantly steeper peaks in their density curves compared to Beech and Spruce. While all species present unique density curves, a commonality emerges—each curve is characterized by a single peak, except for the intriguing exception observed in Telgte. Taking Beech as an illustrative example, our findings indicate a notable shift in the peak to a considerably higher extent. The varinace in the density curves indicating that an differentiation between species only with the help of tree height values could be difficult.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_density_plots</span>(detections, <span class="at">value_column =</span> <span class="st">"Z"</span>, <span class="at">category_column1 =</span> <span class="st">"area"</span>, <span class="at">category_column2 =</span> <span class="st">"specie"</span>, <span class="at">title =</span> <span class="st">"Density of the height distributions"</span>, <span class="at">xlims =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">50</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-density-z" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-density-z-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;2: Density of the height distribitions of the detectected trees. Splitted by the different researched areas and grouped by the dominant specie in this area.</figcaption>
</figure>
</div>
</div>
</div>
<p>To have a deeper look into the distributions of those <code>Z</code>-values we will now also have a look into the boxplots of the height distrubutions in the different areas. Noteworthy observations include the presence of outliers beyond the extended range of the Whisker Antennas (<span class="math inline">\(1.5*\text{IQR}\)</span>) in all datasets. Of particular interest is the Rinkerode dataset, which exhibits a higher prevalence of outliers in the upper domain. Anomalies in this dataset are attributed to potential inaccuracies, urging a critical examination of data integrity. A pairwise examination of Oak and Pine species indicates higher mean heights for Oak compared to Pine. This insight underscores the significance of species-specific attributes in shaping overall height distributions. Further exploration into the factors contributing to these mean differences enhances our understanding of the unique characteristics inherent to each species. Contrary to expectations, the spread within a particular species does not exhibit significant divergence from the spread observed between different species. This finding suggests that while species-specific traits play a crucial role in shaping height distributions, certain overarching factors may contribute to shared patterns across diverse tree populations.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_boxplot</span>(detections, <span class="at">value_column =</span> <span class="st">"Z"</span>, <span class="at">category_column1 =</span> <span class="st">"area"</span>, <span class="at">category_column2 =</span> <span class="st">"specie"</span>, <span class="at">title =</span> <span class="st">"Boxplots of the height distributions"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-boxplot-z" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-boxplot-z-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;3: Boxplots of the height distribitions of the detectected trees. Splitted by the different researched areas and grouped by the dominant specie in this area.</figcaption>
</figure>
</div>
</div>
</div>
<p>Our examination of Kullback-Leibler Divergence (KLD) and Jensen-Shannon Divergence (JSD) metrics reveals low mean values (KLD: 5.252696, JSD: 2.246663) across different species, indicating overall similarity in tree species height distributions. However, within specific species, particularly Pine, higher divergence values (see <a href="#tbl-z-values-kld-pine">Table&nbsp;5</a> and <a href="#tbl-z-values-jsd-pine">Table&nbsp;10</a>) suggest significant intraspecific differences.</p>
<p>Notably, the Spruce species consistently demonstrates low divergence values across all tested areas, implying a high level of explainability. This finding highlights tree height as a reliable indicator for detecting Spruce trees, indicating its potential for accurate species identification in diverse forest ecosystems.</p>
</section>
<section id="number-of-returns" class="level3" data-number="3.2.2">
<h3 data-number="3.2.2" class="anchored" data-anchor-id="number-of-returns"><span class="header-section-number">3.2.2</span> Number of returns</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> sf<span class="sc">::</span><span class="fu">st_read</span>(<span class="st">"data/tree_properties.gpkg"</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>neighbors <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_neighbor_paths</span>() <span class="sc">|&gt;</span> lfa<span class="sc">::</span><span class="fu">lfa_combine_sf_obj</span>(lfa<span class="sc">::</span><span class="fu">lfa_get_all_areas</span>())</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>data <span class="ot">=</span> sf<span class="sc">::</span><span class="fu">st_join</span>(data,neighbors, <span class="at">join =</span> sf<span class="sc">::</span>st_within)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Examining the distribution of LiDAR returns per tree is the focus of our current investigation. Initial analysis involves the study of density graphs representing the distribution of LiDAR returns. The density curves for each species exhibit distinct peaks corresponding to their respective species, providing a clear differentiation in LiDAR return patterns. Notably, there is an exception observed in the Brilon patch (Spruce), where the curve deviates, possibly indicative of variations in forest age. A noteworthy trend is the divergent shape of density curves between coniferous and deciduous trees. Conifers exhibit steeper curves, indicating lower density for higher return values compared to deciduous trees. This disparity underscores the potential of LiDAR data to distinguish between tree types based on return density characteristics. In the case of Beech trees, the peaks’ heights vary among different curves, suggesting nuanced variations within the species. Despite these differences, all species consistently peak in similar regions, emphasizing the overarching similarities in LiDAR return patterns across diverse tree species.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_density_plots</span>(data, <span class="at">value_column =</span> <span class="st">"number_of_returns"</span>, <span class="at">category_column1 =</span> <span class="st">"area"</span>, <span class="at">category_column2 =</span> <span class="st">"specie"</span>, <span class="at">title =</span> <span class="st">"Density of the distribution of LiDAR returns per individual tree"</span>, <span class="at">xlims =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">10000</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-density-number-returns" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-density-number-returns-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;4: Density of the amount of LiDAR returns per detectected tree. Splitted by the different researched areas and grouped by the dominant specie in this area.</figcaption>
</figure>
</div>
</div>
</div>
<p>Currently, our investigation focuses on boxplots representing each patch. We observe significant size variations among plots within the same species. Notably, numerous outliers are present above the box in each patch. For Pines, the boxes exhibit a notable similarity. However, the box for Brilon is entirely shifted from other boxes associated with patches featuring Spruce forest.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_boxplot</span>(data, <span class="at">value_column =</span> <span class="st">"number_of_returns"</span>, <span class="at">category_column1 =</span> <span class="st">"area"</span>, <span class="at">category_column2 =</span> <span class="st">"specie"</span>, <span class="at">title =</span> <span class="st">"Boxplots of the distribution of LiDAR returns per individual tree"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-boxplot-number-returns" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-boxplot-number-returns-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;5: Boxplots of the the amount of LiDAR returns per detectected tree. Splitted by the different researched areas and grouped by the dominant specie in this area.</figcaption>
</figure>
</div>
</div>
</div>
<p>Overall, our analysis reveals very low results for both Kullback-Leibler Divergence (KLD) and Jensen-Shannon Divergence (JSD) metrics across different species. Within species, there is high explainability observed for the different LiDAR return curves between patches.</p>
<p>This suggests that the number of returns alone may not be a robust predictor for identifying the dominant species in a forest. However, the curves indicate a clear potential for distinguishing between conifers (Pine and Spruce) and deciduous trees (Beech and Oak) based on the number of returns. This observation is further supported by the JSD scores, as detailed in <a href="#tbl-number-of-returns-jsd_specie">Table&nbsp;47</a>.</p>
</section>
<section id="n-nearest-neighbours" class="level3" data-number="3.2.3">
<h3 data-number="3.2.3" class="anchored" data-anchor-id="n-nearest-neighbours"><span class="header-section-number">3.2.3</span> n-nearest Neighbours</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>neighbors <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_combine_sf_obj</span>(lfa<span class="sc">::</span><span class="fu">lfa_get_neighbor_paths</span>(),lfa<span class="sc">::</span><span class="fu">lfa_get_all_areas</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="overview" class="level4">
<h4 class="anchored" data-anchor-id="overview">Overview</h4>
<p>To initiate our analysis, we first establish a framework for selecting neighbors by examining the distance development with different <span class="math inline">\(n\)</span>, as illustrated in <a href="#fig-n-nearest-overview">Figure&nbsp;6</a>. The curves share a similar design, but the actual values vary. Notably, as <span class="math inline">\(n\)</span> increases, the distance between all patches also increases, indicating a broader spatial context.</p>
<p>Considering this trend, we extend our investigation beyond the nearest neighbor to include the 100th nearest neighbor. The <span class="math inline">\(\Delta\)</span>distance shows a consistent decrease with each increment in <span class="math inline">\(n\)</span>, reinforcing our decision to limit exploration beyond <span class="math inline">\(n\)</span> of a hundred. Additionally, the constraint is driven by practical considerations, as our sample size occasionally lacks the capacity to explore larger <span class="math inline">\(n\)</span> values, resulting in inaccurate values due to the absence of the true nearest neighbor within the sample area.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_neighbor_mean_curves</span>(neighbors) <span class="sc">|&gt;</span> lfa<span class="sc">::</span><span class="fu">lfa_create_plot_per_area</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-n-nearest-overview" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-n-nearest-overview-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;6: Average Distance to n-nearest neighbor from each patch. For simplicity colored by the dominant specie of each tree.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="the-nearest-neighbour" class="level4">
<h4 class="anchored" data-anchor-id="the-nearest-neighbour">The Nearest Neighbour</h4>
<p>Our initial focus centers on examining the distance to the nearest neighbor for each tree. Notably, the curve representing Spruce exhibits distinct characteristics compared to the three other curves—displaying a steeper profile with less variance, as depicted in <a href="#fig-density-1-nearest">Figure&nbsp;7</a>.</p>
<p>Further analysis of all patches reveals similar distributions, as evident in the boxplot shown in Figure 2 (<a href="#fig-boxplot-1-nearest">Figure&nbsp;8</a>), where mean and variance demonstrate consistency across patches. However, these graphical statistics present challenges in effectively distinguishing between different tree species based on the distance to the nearest neighbor.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_density_plots</span>(neighbors,<span class="at">value_column =</span> <span class="st">"Neighbor_1"</span>,<span class="at">category_column1 =</span> <span class="st">"area"</span>,<span class="at">category_column2 =</span> <span class="st">"specie"</span>, <span class="at">title =</span> <span class="st">"Density plots for the nearest neighbor among species and areas"</span>, <span class="at">xlims =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">15</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-density-1-nearest" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-density-1-nearest-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;7: Density plot of the distance to the nearest neighbor distribution across all patches grouped by the dominant species.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_boxplot</span>(neighbors,<span class="at">value_column =</span> <span class="st">"Neighbor_1"</span>,<span class="at">category_column1 =</span> <span class="st">"area"</span>,<span class="at">category_column2 =</span> <span class="st">"specie"</span>, <span class="at">title =</span> <span class="st">"Box plots for the nearest neighbor among species and areas"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-boxplot-1-nearest" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-boxplot-1-nearest-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;8: Density plot of the distance to the nearest neighbor distribution across all patches grouped by the dominant species.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="the-100th-nearest-neighbor" class="level4">
<h4 class="anchored" data-anchor-id="the-100th-nearest-neighbor">The 100th nearest Neighbor</h4>
<p>Moving on to the analysis of the 100th nearest neighbor, intriguing patterns emerge. Peaks in the curves display varying heights and positions, with a notable example being the complete shift between Oak and Spruce, as illustrated in <a href="#fig-density-100-nearest">Figure&nbsp;9</a>.</p>
<p>However, it is essential to acknowledge the high variance observed between curves within a species, such as Pine or Beech. While this variance could serve as a potential indicator, it comes with the caveat that the sample size must be substantial for reliable conclusions.</p>
<p>Examining boxplots reveals numerous outliers above the boxes, hinting at potential edge effects on the sides of patches. This observation raises concerns about the adequacy of trees in these areas for a more in-depth analysis, posing challenges in deriving accurate insights.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_density_plots</span>(neighbors,<span class="at">value_column =</span> <span class="st">"Neighbor_100"</span>,<span class="at">category_column1 =</span> <span class="st">"area"</span>,<span class="at">category_column2 =</span> <span class="st">"specie"</span>, <span class="at">title =</span> <span class="st">"Density plots for the nearest neighbor along species and areas"</span>, <span class="at">xlims =</span> <span class="fu">c</span>(<span class="dv">35</span>,<span class="dv">100</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-density-100-nearest" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-density-100-nearest-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;9: Density plot of the distance to the nearest neighbor distribution across all patches grouped by the dominant species.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_boxplot</span>(neighbors,<span class="at">value_column =</span> <span class="st">"Neighbor_100"</span>,<span class="at">category_column1 =</span> <span class="st">"area"</span>,<span class="at">category_column2 =</span> <span class="st">"specie"</span>, <span class="at">title =</span> <span class="st">"Box plots for the nearest neighbor along species and areas"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-boxplot-100-nearest" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-boxplot-100-nearest-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;10: Density plot of the distance to the nearest neighbor distribution across all patches grouped by the dominant species.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="average-distance-to-100-nearest-neighbors" class="level4">
<h4 class="anchored" data-anchor-id="average-distance-to-100-nearest-neighbors">Average distance to 100 nearest neighbors</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Neighbor_"</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>neighbors<span class="sc">$</span>avg <span class="ot">=</span> <span class="fu">rowMeans</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">as.data.frame</span>(neighbors),names))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Turning our attention to the averages of the first 100 neighbors, our analysis indicates strikingly similar results. There is considerable variance observed between different species, as well as within individual species, as depicted in <a href="#fig-density-avg-nearest">Figure&nbsp;11</a>.</p>
<p>Despite the uniformity in average results, the issue of outliers persists, as evident in the boxplot representation shown in <a href="#fig-boxplot-avg-nearest">Figure&nbsp;12</a>. These outliers pose challenges and may be indicative of specific environmental conditions affecting tree distributions. Further exploration is required to better understand and mitigate the impact of outliers on our analysis.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_density_plots</span>(neighbors,<span class="at">value_column =</span> <span class="st">"avg"</span>,<span class="at">category_column1 =</span> <span class="st">"area"</span>,<span class="at">category_column2 =</span> <span class="st">"specie"</span>, <span class="at">title =</span> <span class="st">"Density plots for the average of 100 nearest neighbors"</span>, <span class="at">xlims =</span> <span class="fu">c</span>(<span class="dv">25</span>,<span class="dv">60</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-density-avg-nearest" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-density-avg-nearest-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;11: Density plot of the average distance to the nearest neighbor (n=100) distribution across all patches grouped by the dominant species.</figcaption>
</figure>
</div>
</div>
</div>
<p>The neighbor analysis proves potentially useful for distinguishing between tree species, yet the observed variances within each species suggest that relying solely on distance to neighbors may not suffice.</p>
<p>A critical consideration is the sample size problem, wherein more distinguishable patterns emerge with higher neighbor levels, but this necessitates a sufficiently large sample size. Unfortunately, deriving a clear relationship between sample size and the number of tree neighbors remains elusive in our current findings. This gap in understanding could be a pertinent subject for further research, delving into the intricate interplay between sample size and the effectiveness of neighbor analysis in species differentiation.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_boxplot</span>(neighbors,<span class="at">value_column =</span> <span class="st">"avg"</span>,<span class="at">category_column1 =</span> <span class="st">"area"</span>,<span class="at">category_column2 =</span> <span class="st">"specie"</span>, <span class="at">title =</span> <span class="st">"Box plots for the average to the nearest neighbor across all species and areas"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-boxplot-avg-nearest" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-boxplot-avg-nearest-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;12: Density plot of the average distance to the nearest neighbor (n = 100) distribution across all patches grouped by the dominant species.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="density-of-forest-patches" class="level3" data-number="3.2.4">
<h3 data-number="3.2.4" class="anchored" data-anchor-id="density-of-forest-patches"><span class="header-section-number">3.2.4</span> Density of forest patches</h3>
<p>Examining densities provides valuable insights into identifying the dominant species within patches. Spruce stands out as the densest species, surpassing all other patches. Following closely in density is Pine, as depicted in <a href="#fig-density-bar">Figure&nbsp;13</a>.</p>
<p>Beech and Oak exhibit similar density levels, with Beech consistently denser across all patches. When comparing the highest density patches for each species, Beech consistently outpaces Oak. While Oak is slightly less dense overall (<span class="math inline">\(8.354499 \times 10^{-3} \frac{1}{m^2}\)</span>) than Beech (<span class="math inline">\(8.727781 \times 10^{-3} \frac{1}{m^2}\)</span>), the distinction in density remains noticeable.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(units)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_calculate_patch_density</span>() <span class="sc">|&gt;</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  lfa<span class="sc">::</span><span class="fu">lfa_create_grouped_bar_plot</span>(<span class="at">grouping_var =</span> <span class="st">"species"</span>, <span class="at">value_col =</span> <span class="st">"density"</span>, <span class="at">label_col =</span> <span class="st">"name"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-density-bar" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-density-bar-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;13: Barplot of the densitys of all patches (#detected trees/area of patch). Colorized by the dominant tree species of each patch.</figcaption>
</figure>
</div>
</div>
</div>
<p>In summary, our findings indicate that the density of each patch proves highly effective in distinguishing dominant species. Furthermore, the differentiation between conifers (Pine and Spruce) and deciduous trees (Beech and Oak) based on density aligns with patterns observed in the number of return points per detected tree. While distinguishing within conifers is straightforward, discerning between the deciduous tree species Beech and Oak, is possible but poses a moderate challenge.</p>
</section>
<section id="canopy-height-model" class="level3" data-number="3.2.5">
<h3 data-number="3.2.5" class="anchored" data-anchor-id="canopy-height-model"><span class="header-section-number">3.2.5</span> Canopy Height Model</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>chms <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_visit_all_areas</span>(lfa<span class="sc">::</span>lfa_chm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>patches <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_all_areas</span>()</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>patches<span class="sc">$</span>chm_mean <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>patches<span class="sc">$</span>chm_var <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>patches<span class="sc">$</span>chm_median <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (area_key <span class="cf">in</span> <span class="fu">names</span>(chms)) {</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  area <span class="ot">&lt;-</span> chms[area_key]</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  area[[area_key]] <span class="sc">|&gt;</span> <span class="fu">as.vector</span>() <span class="ot">-&gt;</span> vec</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  patches[patches<span class="sc">$</span>area <span class="sc">==</span> area_key, <span class="st">"chm_mean"</span>] <span class="ot">&lt;-</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mean</span>(vec, <span class="at">na.rm =</span> T)</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>  patches[patches<span class="sc">$</span>area <span class="sc">==</span> area_key, <span class="st">"chm_var"</span>] <span class="ot">&lt;-</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">var</span>(vec, <span class="at">na.rm =</span> T)</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>  patches[patches<span class="sc">$</span>area <span class="sc">==</span> area_key, <span class="st">"chm_median"</span>] <span class="ot">&lt;-</span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">median</span>(vec, <span class="at">na.rm =</span> T)</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_grouped_bar_plot</span>(patches,<span class="st">"specie"</span>,<span class="st">"chm_mean"</span>,<span class="st">"area"</span>,<span class="at">ylab =</span> <span class="st">"Mean"</span>, <span class="at">title =</span> <span class="st">"Mean Canopy Height"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-chm-mean" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-chm-mean-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;14: Mean Canopy Height across all patches colored by the dominant specie of each patch.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_grouped_bar_plot</span>(patches,<span class="st">"specie"</span>,<span class="st">"chm_var"</span>,<span class="st">"area"</span>,<span class="at">ylab =</span> <span class="st">"Variance"</span>, <span class="at">title =</span> <span class="st">"Variance of Canopy Height"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-chm-var" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-chm-var-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;15: Variance Canopy Height across all patches colored by the dominant specie of each patch.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_create_grouped_bar_plot</span>(patches,<span class="st">"specie"</span>,<span class="st">"chm_median"</span>,<span class="st">"area"</span>,<span class="at">ylab =</span> <span class="st">"Median"</span>, <span class="at">title =</span> <span class="st">"Median Canopy Height"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-chm-median" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-chm-median-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;16: Median Canopy Height across all patches colored by the dominant specie of each patch.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="random-forest-predictions" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="random-forest-predictions"><span class="header-section-number">3.3</span> Random Forest Predictions</h2>
<p>This section outlines the presentation of Random Forest results, encompassing three models, each trained on different parameters. Each chapter provides an explanation for the selection of a particular model, offering insights into the reasoning behind our choices.</p>
<section id="use-neighbors-and-height" class="level3" data-number="3.3.1">
<h3 data-number="3.3.1" class="anchored" data-anchor-id="use-neighbors-and-height"><span class="header-section-number">3.3.1</span> Use neighbors and height</h3>
<p>A Random Forest model was trained on tree heights (z-values) and the distance to the nearest 100 neighbors. Despite achieving only medium results, this straightforward model clearly demonstrates the feasibility of distinguishing between tree species.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>detections <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_detections</span>()</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>neighbors <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_neighbor_paths</span>() <span class="sc">|&gt;</span> lfa<span class="sc">::</span><span class="fu">lfa_combine_sf_obj</span>(lfa<span class="sc">::</span><span class="fu">lfa_get_all_areas</span>())</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>neighbors <span class="ot">&lt;-</span> sf<span class="sc">::</span><span class="fu">st_join</span>(neighbors,detections, <span class="at">join =</span> sf<span class="sc">::</span>st_within)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(neighbors)[<span class="fu">names</span>(neighbors) <span class="sc">==</span> <span class="st">'specie.x'</span>] <span class="ot">&lt;-</span> <span class="st">'specie'</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(neighbors)[<span class="fu">names</span>(neighbors) <span class="sc">==</span> <span class="st">'area.x'</span>] <span class="ot">&lt;-</span> <span class="st">'area'</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>excluded_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"area.x"</span>,<span class="st">"specie.x"</span>,<span class="st">"treeID.y"</span>,<span class="st">"Z.y"</span>,<span class="st">"area.y"</span>,<span class="st">"specie.y"</span>,<span class="st">"geom"</span>,<span class="st">"treeID.x"</span>,<span class="st">"Z.x"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="report_cache/html/unnamed-chunk-27_a12ecf5a16a71e89cca40523771cd7b5">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_random_forest</span>(<span class="at">tree_data =</span> neighbors, <span class="at">excluded_input_columns =</span> excluded_cols,<span class="at">response_variable =</span> <span class="st">"specie"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>model.rf_neighbors <span class="ot">&lt;-</span> data</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(model.rf_neighbors, <span class="at">file =</span> <span class="st">"./models/neighbors.rData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>The classifier exhibits notable performance variations across different classes. Precision for the “Beech” class is high, indicating accurate positive predictions, but the lower recall suggests the possibility of missing some instances of “Beech.” In contrast, both precision and recall for the “Oak” and “Pine” classes are extremely low, highlighting challenges in accurately classifying instances. The “Spruce” class shows moderate precision and high recall, indicating comparatively better performance.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>cm <span class="ot">&lt;-</span> data<span class="sc">$</span>confusion_matrix</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_plot_confusion_matrix</span>(cm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-cm-neighbors" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-cm-neighbors-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;17: Confusion Matrix of randomForest on the distance to 100 nearest neighbors.</figcaption>
</figure>
</div>
</div>
</div>
<p>The model’s predictions for “Beech” are frequent but vary significantly, leading to substantial differences between Precision and Recall (see <a href="#fig-pr-neighbors">Figure&nbsp;18</a>). The model tends to make predictions that are either mostly correct or mostly incorrect, resulting in a homogenous prediction pattern, as illustrated in the confusion matrix (see <a href="#fig-cm-neighbors">Figure&nbsp;17</a>).</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>confusion_matrix <span class="sc">|&gt;</span> lfa<span class="sc">::</span><span class="fu">lfa_calculate_rf_metrics</span>() <span class="sc">|&gt;</span> lfa<span class="sc">::</span><span class="fu">lfa_visualize_rf_metrics</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-pr-neighbors" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-pr-neighbors-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;18: Class wise precision and recall for randomForest-Classification with distance to the 100 nearest neighbors.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="enrich-neighbors-with-segmentation-data" class="level3" data-number="3.3.2">
<h3 data-number="3.3.2" class="anchored" data-anchor-id="enrich-neighbors-with-segmentation-data"><span class="header-section-number">3.3.2</span> Enrich Neighbors with segmentation data</h3>
<p>This model employs the same parameters as the preceding one, while incorporating additional features derived from the segmentation of each tree in the point cloud. The expanded feature set includes density of returns per tree, mean and variance of Z values of return points, mean and variance of intensity of LiDAR returns, number of LiDAR returns per tree, and the area in square meters of each tree. The aim is to compare the results obtained with this augmented feature set against those based solely on tree detection characteristics.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> sf<span class="sc">::</span><span class="fu">st_read</span>(<span class="st">"./data/tree_properties.gpkg"</span>)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>detections <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_detections</span>()</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>neighbors <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_neighbor_paths</span>() <span class="sc">|&gt;</span> lfa<span class="sc">::</span><span class="fu">lfa_combine_sf_obj</span>(lfa<span class="sc">::</span><span class="fu">lfa_get_all_areas</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>combined <span class="ot">&lt;-</span> sf<span class="sc">::</span><span class="fu">st_join</span>(data,detections,<span class="at">join =</span> sf<span class="sc">::</span>st_within)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>combined<span class="sc">$</span>Z.x <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(combined)[<span class="fu">names</span>(combined) <span class="sc">==</span> <span class="st">'Z.y'</span>] <span class="ot">&lt;-</span> <span class="st">'Z'</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>combined<span class="sc">$</span>treeID.segmentation <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>combined[[<span class="st">"density"</span>]][<span class="fu">is.na</span>(combined[[<span class="st">"density"</span>]])] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>combined[[<span class="st">"Z.mean"</span>]][<span class="fu">is.na</span>(combined[[<span class="st">"Z.mean"</span>]])] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>combined[[<span class="st">"Z.var"</span>]][<span class="fu">is.na</span>(combined[[<span class="st">"Z.var"</span>]])] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>combined[[<span class="st">"Intensity.mean"</span>]][<span class="fu">is.na</span>(combined[[<span class="st">"Intensity.mean"</span>]])] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>combined[[<span class="st">"Intensity.var"</span>]][<span class="fu">is.na</span>(combined[[<span class="st">"Intensity.var"</span>]])] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>combined[[<span class="st">"number_of_returns"</span>]][<span class="fu">is.na</span>(combined[[<span class="st">"number_of_returns"</span>]])] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>combined[[<span class="st">"tree_area"</span>]][<span class="fu">is.na</span>(combined[[<span class="st">"tree_area"</span>]])] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">1</span></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>neighbors<span class="sc">$</span>treeID <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>neighbors<span class="sc">$</span>Z <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>neighbors<span class="sc">$</span>area <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>neighbors<span class="sc">$</span>specie <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>combined <span class="ot">=</span> sf<span class="sc">::</span><span class="fu">st_join</span>(combined, neighbors, sf<span class="sc">::</span>st_within)</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>excluded_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Z.x"</span>, <span class="st">"treeID.detection"</span>,<span class="st">"treeID.segmentation"</span>,<span class="st">"name_las_file"</span>,<span class="st">"treeID"</span>,<span class="st">"area"</span>,<span class="st">"specie"</span>,<span class="st">"geom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="report_cache/html/unnamed-chunk-34_3d9b048158d7d5ab3b43b2b9add6a717">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_random_forest</span>(<span class="at">tree_data =</span> combined, <span class="at">excluded_input_columns =</span> excluded_cols,<span class="at">response_variable =</span> <span class="st">"specie"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>model.rf_segmentation_detection_params <span class="ot">&lt;-</span> data</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(model.rf_segmentation_detection_params, <span class="at">file =</span> <span class="st">"./models/segmentation_detection_params.rData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>cm <span class="ot">&lt;-</span> data<span class="sc">$</span>confusion_matrix</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_plot_confusion_matrix</span>(cm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-cm-tree-level" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-cm-tree-level-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;19: Confusion Matrix of randomForest with all parameters derived from tree level.</figcaption>
</figure>
</div>
</div>
</div>
<p>The classifier exhibits suboptimal performance for the “Beech” and “Oak” classes, with low precision and recall values, indicative of challenges in accurately classifying instances for these categories. In contrast, the “Pine” class demonstrates higher precision and recall, suggesting improved performance in capturing true instances while minimizing false positives. The “Spruce” class displays moderate precision and high recall, indicating relatively better performance in capturing true instances with fewer false positives.</p>
<p>Notably, the overall accuracy (0.44) and Kappa (0.23) have significantly improved compared to the previous model (Accuracy = 0.35, Kappa = 0.14). The model demonstrates effectiveness for conifers but shows limitations for deciduous trees, as depicted in the confusion matrix at the tree-level (<a href="#fig-cm-tree-level">Figure&nbsp;19</a>). Specifically, “Oak” and “Beech” ground truth predictions are distributed among all classes, while “Pine” exhibits false negatives, especially for the “Spruce” class.</p>
<p>Additionally, this model achieves a more balanced distribution between Precision and Recall differences across classes, as illustrated in the Precision-Recall curve at the tree-level (<a href="#fig-pr-tree-level">Figure&nbsp;20</a>).</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>confusion_matrix <span class="sc">|&gt;</span> lfa<span class="sc">::</span><span class="fu">lfa_calculate_rf_metrics</span>() <span class="sc">|&gt;</span> lfa<span class="sc">::</span><span class="fu">lfa_visualize_rf_metrics</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-pr-tree-level" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-pr-tree-level-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;20: Precsion and Recall of randomForest with all parameters derived from tree level.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="train-with-patch-level-information" class="level3" data-number="3.3.3">
<h3 data-number="3.3.3" class="anchored" data-anchor-id="train-with-patch-level-information"><span class="header-section-number">3.3.3</span> Train with patch level information</h3>
<p>In this phase of our study, we examined the impact of patch-level information on the model. The model training incorporated Z-Values, distance to the 100 nearest neighbors, density of trees within a patch, as well as statistics related to the canopy height, including mean, variance, and median.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>chms <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_visit_all_areas</span>(lfa<span class="sc">::</span>lfa_chm)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>patches <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_all_areas</span>()</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>patches<span class="sc">$</span>chm_mean <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>patches<span class="sc">$</span>chm_var <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>patches<span class="sc">$</span>chm_median <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (area_key <span class="cf">in</span> <span class="fu">names</span>(chms)) {</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  area <span class="ot">&lt;-</span> chms[area_key]</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  area[[area_key]] <span class="sc">|&gt;</span> <span class="fu">as.vector</span>() <span class="ot">-&gt;</span> vec</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  patches[patches<span class="sc">$</span>area <span class="sc">==</span> area_key, <span class="st">"chm_mean"</span>] <span class="ot">&lt;-</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mean</span>(vec, <span class="at">na.rm =</span> T)</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>  patches[patches<span class="sc">$</span>area <span class="sc">==</span> area_key, <span class="st">"chm_var"</span>] <span class="ot">&lt;-</span></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">var</span>(vec, <span class="at">na.rm =</span> T)</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>  patches[patches<span class="sc">$</span>area <span class="sc">==</span> area_key, <span class="st">"chm_median"</span>] <span class="ot">&lt;-</span></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">median</span>(vec, <span class="at">na.rm =</span> T)</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>neighbors <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_neighbor_paths</span>() <span class="sc">|&gt;</span> lfa<span class="sc">::</span><span class="fu">lfa_combine_sf_obj</span>(lfa<span class="sc">::</span><span class="fu">lfa_get_all_areas</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>detections <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_detections</span>()</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>density <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_calculate_patch_density</span>(<span class="at">detections =</span> detections)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(density) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"id"</span>,<span class="st">"specie"</span>,<span class="st">"area"</span>,<span class="st">"geometry"</span>,<span class="st">"area_size"</span>,<span class="st">"detections"</span>,<span class="st">"density"</span>)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>detections <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(detections,density <span class="sc">|&gt;</span> <span class="fu">as.data.frame</span>(),<span class="at">by=</span><span class="fu">c</span>(<span class="st">"area"</span>,<span class="st">"specie"</span>))</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>detections <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(detections,patches, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"area"</span>,<span class="st">"specie"</span>))</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>detections <span class="ot">&lt;-</span> sf<span class="sc">::</span><span class="fu">st_join</span>(detections, neighbors, <span class="at">join =</span> sf<span class="sc">::</span>st_within)</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>detections<span class="sc">$</span>treeID.x <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(detections)[<span class="fu">names</span>(detections) <span class="sc">==</span> <span class="st">'treeID.y'</span>] <span class="ot">&lt;-</span> <span class="st">'treeID'</span></span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>detections<span class="sc">$</span>Z.x <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(detections)[<span class="fu">names</span>(detections) <span class="sc">==</span> <span class="st">'Z.y'</span>] <span class="ot">&lt;-</span> <span class="st">'Z'</span></span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a>detections<span class="sc">$</span>area.x <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb37-16"><a href="#cb37-16" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(detections)[<span class="fu">names</span>(detections) <span class="sc">==</span> <span class="st">'area.y'</span>] <span class="ot">&lt;-</span> <span class="st">'area'</span></span>
<span id="cb37-17"><a href="#cb37-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-18"><a href="#cb37-18" aria-hidden="true" tabindex="-1"></a>detections<span class="sc">$</span>specie.x <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb37-19"><a href="#cb37-19" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(detections)[<span class="fu">names</span>(detections) <span class="sc">==</span> <span class="st">'specie.y'</span>] <span class="ot">&lt;-</span> <span class="st">'specie'</span></span>
<span id="cb37-20"><a href="#cb37-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-21"><a href="#cb37-21" aria-hidden="true" tabindex="-1"></a>excluded_cols <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"treeID"</span>,<span class="st">"geom"</span>,<span class="st">"area"</span>,<span class="st">"specie"</span>,<span class="st">"id"</span>,<span class="st">"geometry"</span>,<span class="st">"area_size"</span>,<span class="st">"detections"</span>,<span class="st">"geometry"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell" data-hash="report_cache/html/unnamed-chunk-41_bf8f690bd341d180b3562c74b9060ef9">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_random_forest</span>(<span class="at">tree_data =</span> detections, <span class="at">excluded_input_columns =</span> excluded_cols,<span class="at">response_variable =</span> <span class="st">"specie"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>model.rf_patch <span class="ot">&lt;-</span> data</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(model.rf_patch, <span class="at">file =</span> <span class="st">"./models/patch.rData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>cm <span class="ot">&lt;-</span> data<span class="sc">$</span>confusion_matrix</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_plot_confusion_matrix</span>(cm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-cm-patch" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-cm-patch-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;21: Confusion Matrix of randomForest on returns per Tree.</figcaption>
</figure>
</div>
</div>
</div>
<p>The model yields favorable overall results, with an accuracy of 0.63 and a Kappa value of 0.49. Notably, predictions appear to be predominantly influenced by patch-level information, resulting in largely homogeneous predictions within most patches.</p>
<p>For the “Beech” and “Oak” classes, the classifier performs moderately well, exhibiting moderate precision and recall values. These values are within the same domain, with an equal balance between recall and precision for both classes.</p>
<p>In the case of the “Pine” class, high precision is accompanied by lower recall, indicating a conservative prediction approach. However, the classifier successfully captures a substantial portion of actual “Pine” instances. Noteworthy is the presence of outliers where “Pine” is falsely predicted as “Oak” (see <a href="#fig-cm-patch">Figure&nbsp;21</a>). These false predictions are concentrated in the Greffen patch, which contains a total of 513 detections. Notably, all significant Pine patches yield accurate predictions.</p>
<p>The “Spruce” class exhibits a good balance between precision and recall, with slightly better recall values. The deciduous vs.&nbsp;conifers comparison is notably successful, with conifers predicted mostly as true positives and false predictions for deciduous tree species predominantly leaning towards predicting deciduous trees.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>confusion_matrix <span class="sc">|&gt;</span> lfa<span class="sc">::</span><span class="fu">lfa_calculate_rf_metrics</span>() <span class="sc">|&gt;</span> lfa<span class="sc">::</span><span class="fu">lfa_visualize_rf_metrics</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-pr-patch" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="report_files/figure-html/fig-pr-patch-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;22: Class wise precision and recall for randomForest-Classification with LiDAR returns per tree.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="discussion" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Discussion</h1>
<section id="findings" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="findings"><span class="header-section-number">4.1</span> Findings</h2>
<p>Now, we will delve into the findings derived from the previously presented results, discussing them in the context of the hypotheses outlined in the Introduction. As we move forward, the discussion will delve into the alignment of our findings with the initial hypotheses, shedding light on the effectiveness of the employed methodology and the insights gained from the analysis.</p>
<p><strong>LiDAR data can effectively differentiate between tree species in monocultural forests in NRW, and there are statistically significant differences in the LiDAR-derived metrics among the various tree species.</strong></p>
<p>The statement provided offers a nuanced perspective on the effectiveness of tree species differentiation based on the analysis conducted. It is acknowledged that the effectiveness of differentiation is intricately tied to the chosen criteria and the specific species under consideration. Notably, certain patch-level statistics, such as tree density, emerge as robust indicators for distinguishing between tree species. However, when scrutinizing the distribution of characteristics at the individual tree level, the variance between the distributions of patches from the same species is often not significantly smaller than the variance between distributions of different species. This observation suggests that relying solely on characteristics at the tree level might not yield distinct species differentiation. An interesting finding is the effectiveness of distinguishing between conifers and deciduous trees, with notable differences, particularly concerning Spruce trees. The identified dissimilarities, especially when comparing Spruce trees to other species, underscore the potential for effective differentiation in certain instances. The conclusion drawn emphasizes the importance of generating a comprehensive set of statistics for a defined area of interest and comparing the results to sample patches. This approach provides a robust indicator of the species distribution within the specified area. Importantly, the recommendation is against solely relying on one selected parameter, given the relatively high variance observed within the same species. Instead, a multifaceted analysis involving various parameters proves more effective in characterizing and distinguishing tree species in a given region.</p>
<p><strong>Random Forest classification can be used to predict the tree species of trees in monoculteral forests.</strong></p>
<p>The presented statement accurately captures the key findings and insights derived from the analysis of the tree models. It is affirmed that all tree models can be partially utilized to distinguish between tree species. However, not all models demonstrated consistently good results across all classes, with oak scoring moderately to low on the tree models. The observation that utilizing patch-level information for training the model enhances performance aligns with the understanding that aggregating information at a higher level of abstraction can provide more robust insights. Furthermore, the notable success in distinguishing between conifers and deciduous trees, especially with the last model, highlights the efficacy of the approach, showcasing high accuracy in this particular differentiation. The conclusion drawn, emphasizing the effectiveness of looking into all tree parameters for distinguishing between inspected tree species using the random forest method, underscores the comprehensive nature of the analysis. It suggests that a holistic consideration of multiple parameters enhances the model’s ability to accurately classify and distinguish between different tree species.</p>
<p><strong>Random Forest classification performance varies significantly depending on the set of LiDAR parameters used, indicating that certain combinations of parameters contribute more effectively to accurate tree species classification.</strong></p>
<p>The provided statement accurately characterizes the findings regarding the quality of results obtained from the three models and highlights important nuances in the analysis. Indeed, the quality of results exhibits high variance across all three models, considering overall statistics such as accuracy and Kohen’s Kappa, as well as class-specific measures like precision and recall. The mention of presenting only some highlights of the researched parameter combinations emphasizes that the models’ performance is influenced by a multitude of factors. The observation that incorporating segmentation parameters into the last model leads to a decrease in performance underscores the sensitivity of the model to specific parameter combinations. The statement further notes the high dependency of model performance on the utilized tree characteristics, with significant variations across different classes. For instance, the predictability of beech is highlighted, showcasing that certain parameters, such as tree heights and distance to neighbors, play a more crucial role in accurate predictions compared to others. The insight that another machine learning approach that can recognize the importance of specific parameters for the prediction of a certain class might be beneficial suggests avenues for further exploration and model refinement. Additionally, the variation in the power of differentiation between conifers and deciduous trees among the three models, with the last model outperforming others, reinforces the significance of the approach that incorporates patch-level information. This observation hints at the importance of considering a holistic view of the forest rather than relying solely on individual tree characteristics for certain tasks. In conclusion, the statement provides a comprehensive overview of the complexities and considerations involved in interpreting and optimizing the machine learning models for tree species prediction based on the dataset and analysis conducted.</p>
</section>
<section id="limitations" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="limitations"><span class="header-section-number">4.2</span> Limitations</h2>
<p>Our approach, while yielding valuable insights, is not without its set of assumptions and limitations, each deserving careful consideration to grasp the nuances of our findings.</p>
<p>Firstly, the assumption that all trees within the chosen areas of interest share the same species is integral to our methodology. While this assumption facilitates the development of predictive models, we acknowledge that 100% monocultural forests are rare in reality. Our validation process, primarily relying on Sentinel images, raises potential concerns regarding the inadvertent selection of mixed forests. Such a scenario could significantly impact the distribution of tree characteristics and introduce inaccuracies in the ground truth used for training the random forest algorithm. The temporal aspect of LiDAR data collection is a crucial factor that can significantly impact various LiDAR characteristics. LiDAR data may be captured at different seasons of the year, and these seasonal variations can introduce substantial differences in the derived information. For instance, the number of returns and intensities in LiDAR data can be highly influenced by the season during which the data is collected. Seasonal changes in vegetation, such as leaf growth or shedding, can impact the LiDAR returns. Another key assumption revolves around the representation of naturally grown forests in our areas of interest. Given the absence of true natural forests in Germany and North Rhine-Westphalia (NRW) due to forestry practices, our study areas may consist of newly planted or managed forests with differing characteristics. While this could potentially influence our analysis, we consider the impact relatively minor given the prevalent managed nature of forests in the region.</p>
<p>We assumed the independence of each forest in our study from its spatial environment. However, forests are inherently influenced by various factors such as neighborhood type, altitude, climate, and the presence of fauna. Despite our efforts to draw samples from diverse locations in NRW, these environmental variations persist, introducing a layer of complexity to our analysis. The relatively small sample size, with only three areas of interest per species, is another limitation. Increasing the sample size would undoubtedly enhance the robustness of our analysis and mitigate potential drawbacks associated with the limited number of samples. The correctness of our detection process, reliant on the accuracy of the LiDAR point cloud, is a crucial aspect. While the algorithm employed is established and widely used, the precision of our results hinges on the accuracy of the LiDAR detection process. Lastly, we acknowledge technical limitations, notably the tile-based segmentation approach used for processing efficiency. The choice of <span class="math inline">\(50\times50\)</span>m tiles, while suitable for practical hardware constraints, introduces potential challenges near tile borders. Trees located at these borders may be counted as two separate trees during the segmentation process, impacting our results.</p>
<p>In conclusion, understanding these assumptions and limitations is paramount for a nuanced interpretation of our results. Each aspect informs the intricacies of our methodology and underscores potential areas for refinement in future studies.</p>
</section>
<section id="further-work" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="further-work"><span class="header-section-number">4.3</span> Further Work</h2>
<p>Our work has laid the foundation for further exploration and refinement in the field of tree species prediction and analysis. As we reflect on our findings, several avenues for future research and optimization strategies emerge.</p>
<p>One promising direction involves delving into the predictability of other tree species. While our study focused on Beech, Oak, Pine, and Spruce, primarily due to their regional relevance, investigating the performance of our methods on a broader spectrum of species could provide valuable insights into the generalizability of our models.</p>
<p>Expanding the geographical scope of our approach is another intriguing prospect. Our study was confined to a localized area in North Rhine-Westphalia (NRW). Testing the project’s performance in diverse regions with similar species could unravel variations influenced by different environmental conditions. Additionally, exploring potential spatial trends in derived parameters and model performance could enhance our understanding of broader patterns.</p>
<p>If tree-level datasets are available, future research could test the derived parameters without the assumption that all trees within a patch share the same species. This would offer a more nuanced representation of individual tree characteristics and potentially enhance the accuracy of the random forest models.</p>
<p>Quantitative parameter selection methods present an opportunity for optimization. Moving beyond a trial-and-error approach, incorporating metrics like Kullback-Leibler Divergence (KLD) and Jensen-Shannon Divergence (JSD) into the selection process could provide a systematic and data-driven way to identify relevant parameters. Advanced techniques, such as reinforced learning methods, may also hold promise in this context.</p>
<p>Dataset quality optimization, particularly the removal of outliers, is another avenue for exploration. Many character distributions in our dataset exhibited outliers, and assessing the impact of outlier removal on random forest accuracy could be a valuable optimization strategy.</p>
<p>Finally, exploring additional tree statistics beyond those considered in our current study could enrich our understanding. This might involve investigating intensity distribution, the number of returns distribution on the z-axis per tree, and more sophisticated spatial point pattern analyses <span class="citation" data-cites="pebesma2023spatial">(<a href="#ref-pebesma2023spatial" role="doc-biblioref">Pebesma and Bivand 2023</a>)</span>. Techniques like Ripley’s reduced second moment function <span class="citation" data-cites="ripley1977modelling">(<a href="#ref-ripley1977modelling" role="doc-biblioref">Ripley 1977</a>)</span> could offer deeper insights into spatial patterns.</p>
<p>Incorporating these avenues into future research endeavors has the potential to advance the field, refining methodologies and expanding the applicability of tree species prediction and analysis. Our work, while a significant step forward, lays the groundwork for a more comprehensive exploration of the intricacies within this domain.</p>
</section>
<section id="conclusion" class="level2" data-number="4.4">
<h2 data-number="4.4" class="anchored" data-anchor-id="conclusion"><span class="header-section-number">4.4</span> Conclusion</h2>
<p>Our research aimed to investigate the usability of LiDAR data in distinguishing between tree species in monocultural forests, contributing to the field of forest monitoring. The overall success of our approach suggests promising outcomes. However, it’s important to acknowledge and further address certain limitations identified in our work. Future endeavors in this line of research could delve deeper into these limitations to enhance the robustness and applicability of the methodology. The potential utility of our approach in forest monitoring applications is a significant finding. The ability to distinguish between tree species in monocultural forests using LiDAR data offers valuable insights for forest management and ecological studies. The success of our methodology lays a foundation for continued research and application in real-world scenarios, potentially influencing forest monitoring practices.</p>
<p>In summary, our research has demonstrated the viability of using LiDAR data for distinguishing tree species in monocultural forests, contributing to the broader field of forest monitoring. By addressing and expanding upon the identified limitations, our approach has the potential to become a valuable tool in practical forest management and environmental research.</p>
</section>
</section>
<section id="references" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> References</h1>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-Naturefund" class="csl-entry" role="listitem">
n.d. <em>Naturefund</em>. Naturefund e. V. <a href="https://www.naturefund.de/en/information/afforestation/forests_importance_and_function/the_importance_of_forests">https://www.naturefund.de/en/information/afforestation/forests_importance_and_function/the_importance_of_forests</a>.
</div>
<div id="ref-copernicusSentinel1Missions" class="csl-entry" role="listitem">
Agency, European Space. 2024. <span>“<span>S</span>entinel-1 - <span>M</span>issions - <span>S</span>entinel <span>O</span>nline - <span>S</span>entinel <span>O</span>nline — Sentinels.copernicus.eu.”</span> <a href="https://sentinels.copernicus.eu/web/sentinel/missions/sentinel-1" class="uri">https://sentinels.copernicus.eu/web/sentinel/missions/sentinel-1</a>.
</div>
<div id="ref-alvarez2002exact" class="csl-entry" role="listitem">
Alvarez, Sergio A. 2002. <span>“An Exact Analytical Relation Among Recall, Precision, and Classification Accuracy in Information Retrieval.”</span> <em>Boston College, Boston, Technical Report BCCS-02-01</em>, 1–22.
</div>
<div id="ref-Blickensdoerfer2022" class="csl-entry" role="listitem">
Blickensdoerfer, Lukas. 2022. <span>“Dominant Tree Species for Germany (2017/2018).”</span> <em>Waldatlas- Wald Und Waldnutzung</em>. Thünen Atlas. <a href="https://atlas.thuenen.de/layers/geonode:Dominant_Species_Class">https://atlas.thuenen.de/layers/geonode:Dominant_Species_Class</a>.
</div>
<div id="ref-breiman2001random" class="csl-entry" role="listitem">
Breiman, Leo. 2001. <span>“Random Forests.”</span> <em>Machine Learning</em> 45: 5–32.
</div>
<div id="ref-brovelli2008accuracy" class="csl-entry" role="listitem">
Brovelli, Maria Antonia, Mattia Crespi, Francesca Fratarcangeli, Francesca Giannone, and Eugenio Realini. 2008. <span>“Accuracy Assessment of High Resolution Satellite Imagery Orientation by Leave-One-Out Method.”</span> <em>ISPRS Journal of Photogrammetry and Remote Sensing</em> 63 (4): 427–40.
</div>
<div id="ref-Brownlee2019Calculate" class="csl-entry" role="listitem">
Brownlee, Jason. 2019. <span>“How to Calculate the KL Divergence for Machine Learning.”</span> <em>MachineLearningMastery.com</em>. <a href="https://machinelearningmastery.com/divergence-between-probability-distributions/">https://machinelearningmastery.com/divergence-between-probability-distributions/</a>.
</div>
<div id="ref-cleverdon1967cranfield" class="csl-entry" role="listitem">
Cleverdon, Cyril. 1967. <span>“The Cranfield Tests on Index Language Devices.”</span> In <em>Aslib Proceedings</em>, 19:173–94. 6. MCB UP Ltd.
</div>
<div id="ref-cremer2020potential" class="csl-entry" role="listitem">
Cremer, Felix, Mikhail Urbazaev, José Cortés, John Truckenbrodt, Christiane Schmullius, and Christian Thiel. 2020. <span>“Potential of Recurrence Metrics from Sentinel-1 Time Series for Deforestation Mapping.”</span> <em>IEEE Journal of Selected Topics in Applied Earth Observations and Remote Sensing</em> 13: 5233–40.
</div>
<div id="ref-lfa" class="csl-entry" role="listitem">
Danel, Jakob, and Frederick Bruch. 2024. <em>Lfa: LiDAR Forest Analysis: Diversity of Tree Species in an Oecosystem</em>. <a href="https://github.com/jakobdanel/lidar-forest-analysis">https://github.com/jakobdanel/lidar-forest-analysis</a>.
</div>
<div id="ref-dostalova2021european" class="csl-entry" role="listitem">
Dostálová, Alena, Mait Lang, Janis Ivanovs, Lars T Waser, and Wolfgang Wagner. 2021. <span>“European Wide Forest Classification Based on Sentinel-1 Data.”</span> <em>Remote Sensing</em> 13 (3): 337.
</div>
<div id="ref-fleiss1969large" class="csl-entry" role="listitem">
Fleiss, Joseph L, Jacob Cohen, and Brian S Everitt. 1969. <span>“Large Sample Standard Errors of Kappa and Weighted Kappa.”</span> <em>Psychological Bulletin</em> 72 (5): 323.
</div>
<div id="ref-dlrRaumfahrtagenturLidar" class="csl-entry" role="listitem">
Gonzalez, Diana. <span>“<span>D</span><span>L</span><span>R</span> - <span>R</span>aumfahrtagentur - <span>L</span>idar — Dlr.de.”</span> <a href="https://www.dlr.de/rd/desktopdefault.aspx/tabid-5626/9178_read-17527/" class="uri">https://www.dlr.de/rd/desktopdefault.aspx/tabid-5626/9178_read-17527/</a>.
</div>
<div id="ref-grosse2002analysis" class="csl-entry" role="listitem">
Grosse, Ivo, Pedro Bernaola-Galván, Pedro Carpena, Ramón Román-Roldán, Jose Oliver, and H Eugene Stanley. 2002. <span>“Analysis of Symbolic Sequences Using the Jensen-Shannon Divergence.”</span> <em>Physical Review E</em> 65 (4): 041905.
</div>
<div id="ref-kullback1951kullback" class="csl-entry" role="listitem">
Kullback, Solomon. 1951. <span>“Kullback-Leibler Divergence.”</span>
</div>
<div id="ref-li2012" class="csl-entry" role="listitem">
Li, Wenkai, Qinghua Guo, Marek Jakubowski, and Maggi Kelly. 2012. <span>“A New Method for Segmenting Individual Trees from the Lidar Point Cloud.”</span> <em>Photogrammetric Engineering and Remote Sensing</em> 78 (January): 75–84. <a href="https://doi.org/10.14358/PERS.78.1.75">https://doi.org/10.14358/PERS.78.1.75</a>.
</div>
<div id="ref-mori2017" class="csl-entry" role="listitem">
Mori, Akira S., Kenneth P. Lertzman, and Lena Gustafsson. 2017. <span>“Biodiversity and Ecosystem Services in Forest Ecosystems: A Research Agenda for Applied Forest Ecology.”</span> <em>Journal of Applied Ecology</em> 54 (1): 12–27. https://doi.org/<a href="https://doi.org/10.1111/1365-2664.12669">https://doi.org/10.1111/1365-2664.12669</a>.
</div>
<div id="ref-muller2019kaferkampfe" class="csl-entry" role="listitem">
Müller, Martin, and Nadja Imhof. 2019. <span>“K<span>ä</span>ferk<span>ä</span>mpfe: Borkenk<span>ä</span>fer Und Landschaftskonflikte Im Nationalpark Bayerischer Wald.”</span> <em>Landschaftskonflikte</em>, 313–29.
</div>
<div id="ref-sf" class="csl-entry" role="listitem">
Pebesma, Edzer. 2018. <span>“<span class="nocase">Simple Features for R: Standardized Support for Spatial Vector Data</span>.”</span> <em><span>The R Journal</span></em> 10 (1): 439–46. <a href="https://doi.org/10.32614/RJ-2018-009">https://doi.org/10.32614/RJ-2018-009</a>.
</div>
<div id="ref-pebesma2023spatial" class="csl-entry" role="listitem">
Pebesma, Edzer, and Roger Bivand. 2023. <em>Spatial Data Science: With Applications in r</em>. CRC Press.
</div>
<div id="ref-popescu2004" class="csl-entry" role="listitem">
Popescu, Sorin, and Randolph Wynne. 2004. <span>“Seeing the Trees in the Forest: Using Lidar and Multispectral Data Fusion with Local Filtering and Variable Window Size for Estimating Tree Height.”</span> <em>Photogrammetric Engineering and Remote Sensing</em> 70 (May): 589–604. <a href="https://doi.org/10.14358/PERS.70.5.589">https://doi.org/10.14358/PERS.70.5.589</a>.
</div>
<div id="ref-ripley1977modelling" class="csl-entry" role="listitem">
Ripley, Brian D. 1977. <span>“Modelling Spatial Patterns.”</span> <em>Journal of the Royal Statistical Society: Series B (Methodological)</em> 39 (2): 172–92.
</div>
<div id="ref-lidR" class="csl-entry" role="listitem">
Roussel, Jean-Romain, David Auty, Nicholas C. Coops, Piotr Tompalski, Tristan R. H. Goodbody, Andrew Sánchez Meador, Jean-François Bourdon, Florian de Boissieu, and Alexis Achim. 2020. <span>“lidR: An r Package for Analysis of Airborne Laser Scanning (ALS) Data.”</span> <em>Remote Sensing of Environment</em> 251: 112061. <a href="https://doi.org/10.1016/j.rse.2020.112061">https://doi.org/10.1016/j.rse.2020.112061</a>.
</div>
<div id="ref-sobczyk2014eichenprozessionsspinner" class="csl-entry" role="listitem">
Sobczyk, Thomas. 2014. <em>Der Eichenprozessionsspinner in Deutschland: Historie, Biologie, Gefahren, Bek<span>ä</span>mpfung</em>. Deutschland/Bundesamt f<span>ü</span>r Naturschutz.
</div>
<div id="ref-sparck1972statistical" class="csl-entry" role="listitem">
Sparck Jones, Karen. 1972. <span>“A Statistical Interpretation of Term Specificity and Its Application in Retrieval.”</span> <em>Journal of Documentation</em> 28 (1): 11–21.
</div>
<div id="ref-szostak2018using" class="csl-entry" role="listitem">
Szostak, Marta, Paweł Hawryło, and Dobrosława Piela. 2018. <span>“Using of Sentinel-2 Images for Automation of the Forest Succession Detection.”</span> <em>European Journal of Remote Sensing</em> 51 (1): 142–49.
</div>
<div id="ref-valavi2018blockcv" class="csl-entry" role="listitem">
Valavi, Roozbeh, Jane Elith, José J Lahoz-Monfort, and Gurutzeta Guillera-Arroita. 2018. <span>“blockCV: An r Package for Generating Spatially or Environmentally Separated Folds for k-Fold Cross-Validation of Species Distribution Models.”</span> <em>Biorxiv</em>, 357798.
</div>
<div id="ref-welle2014" class="csl-entry" role="listitem">
Welle, Torsten, Lukas Aschenbrenner, Kevin Kuonath, Stefan Kirmaier, and Jonas Franke. 2022. <span>“Mapping Dominant Tree Species of German Forests.”</span> <em>Remote Sensing</em> 14 (14). <a href="https://doi.org/10.3390/rs14143330">https://doi.org/10.3390/rs14143330</a>.
</div>
<div id="ref-ggplot2" class="csl-entry" role="listitem">
Wickham, Hadley. 2016. <em>Ggplot2: Elegant Graphics for Data Analysis</em>. Springer-Verlag New York. <a href="https://ggplot2.tidyverse.org">https://ggplot2.tidyverse.org</a>.
</div>
<div id="ref-dplyr" class="csl-entry" role="listitem">
Wickham, Hadley, Romain François, Lionel Henry, Kirill Müller, and Davis Vaughan. 2023. <em>Dplyr: A Grammar of Data Manipulation</em>. <a href="https://CRAN.R-project.org/package=dplyr">https://CRAN.R-project.org/package=dplyr</a>.
</div>
</div>
</section>
<section id="appendix" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Appendix</h1>
<section id="contributions" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="contributions"><span class="header-section-number">6.1</span> Contributions</h2>
<p>Most of the work was done in pair programming. In the following sections we highlight, our key contributions to the project.</p>
<section id="frederick" class="level3" data-number="6.1.1">
<h3 data-number="6.1.1" class="anchored" data-anchor-id="frederick"><span class="header-section-number">6.1.1</span> Frederick</h3>
<p>My contributions to the project encompassed several key areas: data acquisition, preprocessing, visualization, random forest implementation, and canopy height modeling. Each of these tasks presented unique challenges and learning opportunities.</p>
<section id="data-acquisition-1" class="level4">
<h4 class="anchored" data-anchor-id="data-acquisition-1">Data Acquisition</h4>
<p>My role in data acquisition involved sourcing relevant datasets for our study, particularly focusing on obtaining patches where one tree species exhibited dominance. This process involved navigating through existing datasets such as the <em>Waldmonitor</em> dataset and utilizing maps indicating dominant tree species in Nordrhein-Westfalia (NRW). One of the challenges encountered was ensuring the validity of the selected patches, considering factors such as patch size, species diversity, and the absence of human-made structures. Through this process, I learned the importance of meticulous data selection and validation to ensure the integrity of subsequent analyses.</p>
</section>
<section id="preprocessing-tilingheight-normalization" class="level4">
<h4 class="anchored" data-anchor-id="preprocessing-tilingheight-normalization">Preprocessing: Tiling/Height Normalization</h4>
<p>Preprocessing the acquired data was essential to derive meaningful variables for analysis. Tiling and height normalization were particularly crucial steps to standardize the data and facilitate further analysis. Tiling involved partitioning the patches into smaller, manageable units, while height normalization ensured consistency in height measurements across different patches. This process taught me the importance of data standardization and the impact it can have on downstream analyses.</p>
</section>
<section id="visualization-density-and-boxplot" class="level4">
<h4 class="anchored" data-anchor-id="visualization-density-and-boxplot">Visualization: Density and Boxplot</h4>
<p>Visualizing the data through density plots and boxplots was instrumental in gaining insights into the distribution and characteristics of the variables under study. Creating informative visualizations required careful selection of plotting parameters and consideration of the audience’s interpretation. Balancing clarity and complexity posed a challenge, but it also provided an opportunity to enhance my visualization skills and effectively communicate key findings.</p>
</section>
<section id="random-forest-scores-and-confusion-matrix" class="level4">
<h4 class="anchored" data-anchor-id="random-forest-scores-and-confusion-matrix">Random Forest: Scores and Confusion Matrix</h4>
<p>Implementing the Random Forest algorithm for species classification involved several steps, including model training, validation, and performance evaluation. Generating scores and confusion matrices allowed us to assess the model’s accuracy and identify areas for improvement.</p>
</section>
<section id="canopy-height-model-1" class="level4">
<h4 class="anchored" data-anchor-id="canopy-height-model-1">Canopy Height Model</h4>
<p>Developing a canopy height model involved integrating data from various sources to estimate the height of forest canopies. Overcoming technical challenges and refining the model parameters provided valuable insights into remote sensing techniques and their applicability in forestry research.</p>
<p>In summary, my contributions to the project encompassed a range of tasks, each presenting its own set of challenges and learning opportunities. From data acquisition to model implementation, I gained valuable experience in ecological research methods and data analysis techniques.</p>
</section>
</section>
<section id="jakob" class="level3" data-number="6.1.2">
<h3 data-number="6.1.2" class="anchored" data-anchor-id="jakob"><span class="header-section-number">6.1.2</span> Jakob</h3>
<p>In the broader context of a comprehensive lidar data analysis project, my contributions were instrumental in advancing key components. The preprocessing phase saw enhancements in Intersection, Detection, and Segmentation processes. I played a pivotal role in refining Intersection by overlaying lidar data onto shapefiles, ensuring a coherent file structure. Additionally, my efforts in Detection focused on optimizing tree identification within the intersected data. I developed the tree Segmentation too.</p>
<p>In terms of visualization, my contributions elevated the understanding of lidar data. The presentation of the confusion matrix as a colored table provided a comprehensive view of classification performance. Through the implementation of bar plots for patch-level data, I contributed to insights into the distribution of features. Precision and recall metrics were effectively communicated via bar plots, offering a nuanced evaluation of the model’s performance. The integration of Canopy Height Plots further enriched the visual representation of canopy structures.</p>
<p>Within the tree-level distribution analysis, my role extended to Data Preparation and the application of KLD and JSD. I ensured that the dataset was well-structured and prepared for analysis. Moreover, my involvement in researching methods, implementing calculations, and dynamically building result tables for KLD and JSD significantly contributed to the depth of the analysis.</p>
<p>In the development of an interface for Lidar data, I played a crucial role in designing a structured file system for efficient data storage. The implementation of various functions, including <code>lfa::map_tile_locations</code>, <code>lfa::visit_all_areas</code>, and <code>lfa::combine_sf_object</code>, showcased my commitment to enhancing data iteration and combination processes.</p>
<p>Throughout the project, my contributions led to valuable learnings for the team. I actively addressed challenges associated with handling big data in R and underscored the importance of structured data storage. The development of an R-package was a testament to my commitment to enhancing functionality and promoting reusability. Additionally, my insights deepened the team’s understanding of package workflows in R.</p>
<p>Despite encountering challenges with lidR mapping functions, my proactive approach to debugging, including the development of a custom mapping solution, showcased my resilience and commitment to overcoming obstacles. In summary, my contributions significantly shaped the success and advancement of the broader lidar data analysis project.</p>
</section>
</section>
<section id="sec-appendix-preprocessing" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="sec-appendix-preprocessing"><span class="header-section-number">6.2</span> Script which can be used to do all preprocessing</h2>
<p>Load the file with the research areas</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>sf <span class="ot">&lt;-</span> sf<span class="sc">::</span><span class="fu">read_sf</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"research_areas.shp"</span>), <span class="at">quiet =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Init the project</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lfa)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>sf<span class="sc">::</span><span class="fu">sf_use_s2</span>(<span class="cn">FALSE</span>)</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>locations <span class="ot">&lt;-</span> <span class="fu">lfa_init</span>(<span class="st">"research_areas.shp"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Do all of the preprocessing steps</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co">#lfa::lfa_map_tile_locations(locations, lfa::retile,check_flag = "retile")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_map_tile_locations</span>(locations, lfa<span class="sc">::</span>lfa_intersect_areas, <span class="at">ctg =</span> <span class="cn">NULL</span>, <span class="at">areas_sf =</span> sf,<span class="at">check_flag =</span> <span class="st">"intersect"</span>)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_map_tile_locations</span>(locations, lfa<span class="sc">::</span>lfa_ground_correction, <span class="at">ctg =</span> <span class="cn">NULL</span>,<span class="at">check_flag =</span> <span class="st">"z_correction"</span>)</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_map_tile_locations</span>(locations, lfa<span class="sc">::</span>lfa_segmentation, <span class="at">ctg =</span> <span class="cn">NULL</span>,<span class="at">check_flag =</span> <span class="st">"segmentation"</span>)</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_map_tile_locations</span>(locations, lfa<span class="sc">::</span>lfa_detection, <span class="at">catalog =</span> <span class="cn">NULL</span>, <span class="at">write_to_file =</span> <span class="cn">TRUE</span>,<span class="at">check_flag =</span> <span class="st">"detection"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="sec-appendix-chms" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="sec-appendix-chms"><span class="header-section-number">6.3</span> Canopy Height Models</h2>
<p>The following section contains all canopy height models of the researched patches, plotted.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(raster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: sp</code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>chms <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_visit_all_areas</span>(lfa<span class="sc">::</span>lfa_chm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Load /home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/beech/bielefeld_brackwede/chm.tif"
[1] "Load /home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/beech/billerbeck/chm.tif"
[1] "Load /home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/beech/wuelfenrath/chm.tif"
[1] "Load /home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/oak/hamm/chm.tif"
[1] "Load /home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/oak/muenster/chm.tif"
[1] "Load /home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/oak/rinkerode/chm.tif"
[1] "Load /home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/pine/greffen/chm.tif"
[1] "Load /home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/pine/mesum/chm.tif"
[1] "Load /home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/pine/telgte/chm.tif"
[1] "Load /home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/spruce/brilon/chm.tif"
[1] "Load /home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/spruce/oberhundem/chm.tif"
[1] "Load /home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/spruce/osterwald/chm.tif"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>rinkerode[chms<span class="sc">$</span>rinkerode <span class="sc">&gt;</span> <span class="dv">50</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>bielefeld_brackwede <span class="sc">|&gt;</span> <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">"Canopy Height Model: Bielefeld Brackwede (Beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-51-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Canopy Height Model of the patch Bielefeld Brackwede (Beech) colorized by the canopy height.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>billerbeck <span class="sc">|&gt;</span> <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">"Canopy Height Model: Billerbeck (Beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-52-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Canopy Height Model of the patch Billerbeck (Beech) colorized by the canopy height.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>wuelfenrath <span class="sc">|&gt;</span> <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">"Canopy Height Model: Wülfenrath (Beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-53-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Canopy Height Model of the patch Wülfenrath (Beech) colorized by the canopy height.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>hamm <span class="sc">|&gt;</span> <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">"Canopy Height Model: Hamm (Oak)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-54-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Canopy Height Model of the patch Hamm (Oak) colorized by the canopy height.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>muenster <span class="sc">|&gt;</span> <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">"Canopy Height Model: Münster (Oak)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-55-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Canopy Height Model of the patch Münster (Oak) colorized by the canopy height.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>rinkerode <span class="sc">|&gt;</span> <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">"Canopy Height Model: Rinkerode (Oak)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-56-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Canopy Height Model of the patch Rinkerode (Oak) colorized by the canopy height.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>greffen <span class="sc">|&gt;</span> <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">"Canopy Height Model: Greffen (Pine)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-57-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Canopy Height Model of the patch Greffen (Pine) colorized by the canopy height.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>mesum <span class="sc">|&gt;</span> <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">"Canopy Height Model: Mesum (Pine)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-58-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Canopy Height Model of the patch Mesum (Pine) colorized by the canopy height.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>telgte <span class="sc">|&gt;</span> <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">"Canopy Height Model: Telgte (Pine)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-59-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Canopy Height Model of the patch Telgte (Pine) colorized by the canopy height.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>brilon <span class="sc">|&gt;</span> <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">"Canopy Height Model: Brilon (Spruce)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-60-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Canopy Height Model of the patch Brilon (Spruce) colorized by the canopy height.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>oberhundem <span class="sc">|&gt;</span> <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">"Canopy Height Model: Oberhundem (Spruce)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-61-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Canopy Height Model of the patch Oberhundem (Spruce) colorized by the canopy height.</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>chms<span class="sc">$</span>osterwald <span class="sc">|&gt;</span> <span class="fu">plot</span>(<span class="at">main =</span> <span class="st">"Canopy Height Model: Osterwald (Spruce)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="report_files/figure-html/unnamed-chunk-62-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Canopy Height Model of the patch Osterwald (Spruce) colorized by the canopy height.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="quantitative-results" class="level2" data-number="6.4">
<h2 data-number="6.4" class="anchored" data-anchor-id="quantitative-results"><span class="header-section-number">6.4</span> Quantitative Results</h2>
<section id="distribution-of-z-values" class="level3" data-number="6.4.1">
<h3 data-number="6.4.1" class="anchored" data-anchor-id="distribution-of-z-values"><span class="header-section-number">6.4.1</span> Distribution of Z-Values</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_detections</span>()</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>value_column <span class="ot">&lt;-</span> <span class="st">"Z"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="kullback-leibler-divergence" class="level4">
<h4 class="anchored" data-anchor-id="kullback-leibler-divergence">Kullback-Leibler-Divergence</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>kld_results_specie <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(data,value_column,<span class="st">"specie"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_specie,<span class="st">"Kullback-Leibler-Divergence between species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-z-values-kld_specie" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;2: Kullback-Leibler-Divergence between the researched species Beech, Oak, Pine and Spruce for the atrribute z-values</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Beech</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oak</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Pine</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Spruce</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Beech</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">13.2</td>
<td style="text-align: left;">12.5</td>
<td style="text-align: left;">0.76</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oak</td>
<td style="text-align: left;">4.2</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">3.4</td>
<td style="text-align: left;">5.02</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Pine</td>
<td style="text-align: left;">2.3</td>
<td style="text-align: left;">5.6</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">3.95</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Spruce</td>
<td style="text-align: left;">2.4</td>
<td style="text-align: left;">14.7</td>
<td style="text-align: left;">16.1</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_specie, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5.252696</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"beech"</span>,]</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>kld_results_beech <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_beech,<span class="st">"Kullback-Leibler-Divergence between areas with beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-z-values-kld-beech" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;3: Kullback-Leibler-Divergence between the researched areas which have the dominante specie beech for the atrribute z-values</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Bielefeld_brackwede</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Billerbeck</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Wuelfenrath</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Bielefeld_brackwede</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.4</td>
<td style="text-align: left;">3.1</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Billerbeck</td>
<td style="text-align: left;">0.27</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">6.0</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Wuelfenrath</td>
<td style="text-align: left;">1.13</td>
<td style="text-align: left;">2.4</td>
<td style="text-align: left;">0.0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_beech, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.473353</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"oak"</span>,]</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>kld_results_oak <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_oak,<span class="st">"Kullback-Leibler-Divergence between areas with oak"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-z-values-kld-oak" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;4: Kullback-Leibler-Divergence between the researched areas which have the dominante specie oak for the atrribute z-values</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Hamm</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Muenster</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Rinkerode</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Hamm</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">2.1</td>
<td style="text-align: left;">16</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Muenster</td>
<td style="text-align: left;">0.4</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">17</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Rinkerode</td>
<td style="text-align: left;">7.6</td>
<td style="text-align: left;">17.8</td>
<td style="text-align: left;">0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_oak, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 6.779863</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"pine"</span>,]</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>kld_results_pine <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_pine,<span class="st">"Kullback-Leibler-Divergence between areas with pine"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-z-values-kld-pine" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;5: Kullback-Leibler-Divergence between the researched areas which have the dominante specie pine for the atrribute z-values</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Greffen</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Mesum</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Telgte</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Greffen</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.74</td>
<td style="text-align: left;">16</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Mesum</td>
<td style="text-align: left;">0.43</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">18</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Telgte</td>
<td style="text-align: left;">3.87</td>
<td style="text-align: left;">6.82</td>
<td style="text-align: left;">0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_pine, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5.129383</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"spruce"</span>,]</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>kld_results_spruce <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_spruce,<span class="st">"Kullback-Leibler-Divergence between areas with spruce"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-z-values-kld-spruce" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;6: Kullback-Leibler-Divergence between the researched areas which have the dominante specie spruce for the atrribute z-values</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Brilon</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oberhundem</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Osterwald</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Brilon</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.092</td>
<td style="text-align: left;">1.7</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oberhundem</td>
<td style="text-align: left;">0.081</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">2.1</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Osterwald</td>
<td style="text-align: left;">1.521</td>
<td style="text-align: left;">2.178</td>
<td style="text-align: left;">0.0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_spruce, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.8509258</code></pre>
</div>
</div>
</section>
<section id="jensen-shannon-divergence" class="level4">
<h4 class="anchored" data-anchor-id="jensen-shannon-divergence">Jensen-Shannon Divergence</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>jsd_results_specie <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(data,value_column,<span class="st">"specie"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_specie,<span class="st">"Jensen-Shannon Divergence between species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-z-values-jsd_specie" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;7: Jensen-Shannon Divergence between the researched species Beech, Oak, Pine and Spruce for the atrribute z-values</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Beech</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oak</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Pine</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Spruce</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Beech</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">4.5</td>
<td style="text-align: left;">4.6</td>
<td style="text-align: left;">2.4</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oak</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">3.9</td>
<td style="text-align: left;">6.1</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Pine</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">7.1</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Spruce</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_specie, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2.246663</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"beech"</span>,]</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>jsd_results_beech <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_beech,<span class="st">"Jensen-Shannon Divergence between areas with beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-z-values-jsd-beech" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;8: Jensen-Shannon Divergence between the researched areas which have the dominante specie beech for the atrribute z-values</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Bielefeld_brackwede</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Billerbeck</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Wuelfenrath</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Bielefeld_brackwede</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">1.1</td>
<td style="text-align: left;">3.3</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Billerbeck</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">4.9</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Wuelfenrath</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_beech, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.10555</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"oak"</span>,]</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>jsd_results_oak <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_oak,<span class="st">"Jensen-Shannon Divergence between areas with oak"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-z-values-jsd-oak" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;9: Jensen-Shannon Divergence between the researched areas which have the dominante specie oak for the atrribute z-values</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Hamm</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Muenster</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Rinkerode</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Hamm</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">1.6</td>
<td style="text-align: left;">6.5</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Muenster</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">6.4</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Rinkerode</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_oak, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.692942</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"pine"</span>,]</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>jsd_results_pine <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_pine,<span class="st">"Jensen-Shannon Divergence between areas with pine"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-z-values-jsd-pine" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;10: Jensen-Shannon Divergence between the researched areas which have the dominante specie pine for the atrribute z-values</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Greffen</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Mesum</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Telgte</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Greffen</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">3.1</td>
<td style="text-align: left;">12</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Mesum</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">10</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Telgte</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_pine, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2.956354</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"spruce"</span>,]</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>jsd_results_spruce <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_spruce,<span class="st">"Jensen-Shannon Divergence between areas with spruce"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-z-values-jsd-spruce" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;11: Jensen-Shannon Divergence between the researched areas which have the dominante specie spruce for the atrribute z-values</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Brilon</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oberhundem</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Osterwald</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Brilon</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.31</td>
<td style="text-align: left;">4.0</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oberhundem</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">5.5</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Osterwald</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_spruce, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.100383</code></pre>
</div>
</div>
</section>
</section>
<section id="nearest-neighbours" class="level3" data-number="6.4.2">
<h3 data-number="6.4.2" class="anchored" data-anchor-id="nearest-neighbours"><span class="header-section-number">6.4.2</span> Nearest Neighbours</h3>
<section id="distribution-of-nearest-neighbor-distances" class="level4">
<h4 class="anchored" data-anchor-id="distribution-of-nearest-neighbor-distances">Distribution of nearest neighbor distances</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_combine_sf_obj</span>(lfa<span class="sc">::</span><span class="fu">lfa_get_neighbor_paths</span>(),lfa<span class="sc">::</span><span class="fu">lfa_get_all_areas</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/beech/bielefeld_brackwede/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 1443 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 466999.8 ymin: 5759839 xmax: 467617.1 ymax: 5760261
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/beech/billerbeck/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 1732 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 384890.8 ymin: 5761918 xmax: 385590.9 ymax: 5762478
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/beech/wuelfenrath/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 2779 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 365546.3 ymin: 5683711 xmax: 366356.1 ymax: 5684321
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/oak/hamm/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 2441 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 420953.3 ymin: 5723884 xmax: 421596 ymax: 5724609
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/oak/muenster/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 1270 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 404615.6 ymin: 5752535 xmax: 405396.8 ymax: 5752971
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/oak/rinkerode/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 1643 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 408428.2 ymin: 5746021 xmax: 409014.8 ymax: 5746511
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/pine/greffen/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 513 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 442816.1 ymin: 5760217 xmax: 443148.9 ymax: 5760567
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/pine/mesum/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 5031 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 399930.6 ymin: 5790412 xmax: 400969.7 ymax: 5790950
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/pine/telgte/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 3368 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 416135.1 ymin: 5761663 xmax: 416697.1 ymax: 5762477
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/spruce/brilon/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 3342 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 467305.7 ymin: 5695055 xmax: 467996.9 ymax: 5695593
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/spruce/oberhundem/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 2471 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 442631.7 ymin: 5660096 xmax: 443309.5 ymax: 5660502
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/spruce/osterwald/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 2806 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 455822 ymin: 5673761 xmax: 456483.2 ymax: 5674162
Projected CRS: ETRS89 / UTM zone 32N</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>value_column <span class="ot">&lt;-</span> <span class="st">"Neighbor_1"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="kullback-leibler-divergence-1" class="level5">
<h5 class="anchored" data-anchor-id="kullback-leibler-divergence-1">Kullback-Leibler-Divergence</h5>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>kld_results_specie <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(data,value_column,<span class="st">"specie"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_specie,<span class="st">"Kullback-Leibler-Divergence between species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-1-kld_specie" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;12: Kullback-Leibler-Divergence between the researched species Beech, Oak, Pine and Spruce for the atrribute nearest-neighbor-1</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Beech</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oak</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Pine</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Spruce</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Beech</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.029</td>
<td style="text-align: left;">0.40</td>
<td style="text-align: left;">3.3</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oak</td>
<td style="text-align: left;">0.031</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.25</td>
<td style="text-align: left;">3.9</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Pine</td>
<td style="text-align: left;">0.213</td>
<td style="text-align: left;">0.128</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">4.9</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Spruce</td>
<td style="text-align: left;">2.735</td>
<td style="text-align: left;">3.199</td>
<td style="text-align: left;">4.52</td>
<td style="text-align: left;">0.0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_specie, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1.477983</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"beech"</span>,]</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>kld_results_beech <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_beech,<span class="st">"Kullback-Leibler-Divergence between areas with beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-1-kld-beech" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;13: Kullback-Leibler-Divergence between the researched areas which have the dominante specie beech for the atrribute nearest-neighbor-1</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Bielefeld_brackwede</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Billerbeck</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Wuelfenrath</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Bielefeld_brackwede</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.35</td>
<td style="text-align: left;">0.051</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Billerbeck</td>
<td style="text-align: left;">0.380</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.138</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Wuelfenrath</td>
<td style="text-align: left;">0.059</td>
<td style="text-align: left;">0.15</td>
<td style="text-align: left;">0.000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_beech, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1249588</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"oak"</span>,]</span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>kld_results_oak <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_oak,<span class="st">"Kullback-Leibler-Divergence between areas with oak"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-1-kld-oak" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;14: Kullback-Leibler-Divergence between the researched areas which have the dominante specie oak for the atrribute nearest-neighbor-1</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Hamm</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Muenster</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Rinkerode</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Hamm</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.079</td>
<td style="text-align: left;">0.078</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Muenster</td>
<td style="text-align: left;">0.092</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.019</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Rinkerode</td>
<td style="text-align: left;">0.086</td>
<td style="text-align: left;">0.020</td>
<td style="text-align: left;">0.000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_oak, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.04167636</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"pine"</span>,]</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>kld_results_pine <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_pine,<span class="st">"Kullback-Leibler-Divergence between areas with pine"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-1-kld-pine" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;15: Kullback-Leibler-Divergence between the researched areas which have the dominante specie pine for the atrribute nearest-neighbor-1</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Greffen</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Mesum</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Telgte</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Greffen</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.495</td>
<td style="text-align: left;">0.258</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Mesum</td>
<td style="text-align: left;">0.48</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.098</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Telgte</td>
<td style="text-align: left;">0.22</td>
<td style="text-align: left;">0.076</td>
<td style="text-align: left;">0.000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_pine, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1812239</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"spruce"</span>,]</span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>kld_results_spruce <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_spruce,<span class="st">"Kullback-Leibler-Divergence between areas with spruce"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-1-kld-spruce" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;16: Kullback-Leibler-Divergence between the researched areas which have the dominante specie spruce for the atrribute nearest-neighbor-1</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Brilon</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oberhundem</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Osterwald</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Brilon</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.67</td>
<td style="text-align: left;">5.1</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oberhundem</td>
<td style="text-align: left;">0.41</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">7.2</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Osterwald</td>
<td style="text-align: left;">6.09</td>
<td style="text-align: left;">6.23</td>
<td style="text-align: left;">0.0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_spruce, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2.863587</code></pre>
</div>
</div>
</section>
<section id="jensen-shannon-divergence-1" class="level5">
<h5 class="anchored" data-anchor-id="jensen-shannon-divergence-1">Jensen-Shannon Divergence</h5>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a>jsd_results_specie <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(data,value_column,<span class="st">"specie"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_specie,<span class="st">"Jensen-Shannon Divergence between species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-1-jsd_specie" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;17: Jensen-Shannon Divergence between the researched species Beech, Oak, Pine and Spruce for the atrribute nearest-neighbor-1</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Beech</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oak</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Pine</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Spruce</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Beech</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.22</td>
<td style="text-align: left;">2.1</td>
<td style="text-align: left;">9.3</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oak</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">1.3</td>
<td style="text-align: left;">10.6</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Pine</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">14.7</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Spruce</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_specie, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2.470051</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"beech"</span>,]</span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>jsd_results_beech <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_beech,<span class="st">"Jensen-Shannon Divergence between areas with beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-1-jsd-beech" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;18: Jensen-Shannon Divergence between the researched areas which have the dominante specie beech for the atrribute nearest-neighbor-1</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Bielefeld_brackwede</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Billerbeck</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Wuelfenrath</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Bielefeld_brackwede</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">2.2</td>
<td style="text-align: left;">0.39</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Billerbeck</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">0.85</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Wuelfenrath</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_beech, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.5042359</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"oak"</span>,]</span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>jsd_results_oak <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_oak,<span class="st">"Jensen-Shannon Divergence between areas with oak"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-1-jsd-oak" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;19: Jensen-Shannon Divergence between the researched areas which have the dominante specie oak for the atrribute nearest-neighbor-1</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Hamm</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Muenster</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Rinkerode</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Hamm</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.57</td>
<td style="text-align: left;">0.61</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Muenster</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.17</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Rinkerode</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_oak, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1803836</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"pine"</span>,]</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>jsd_results_pine <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_pine,<span class="st">"Jensen-Shannon Divergence between areas with pine"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-1-jsd-pine" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;20: Jensen-Shannon Divergence between the researched areas which have the dominante specie pine for the atrribute nearest-neighbor-1</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Greffen</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Mesum</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Telgte</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Greffen</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">3.6</td>
<td style="text-align: left;">1.89</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Mesum</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">0.68</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Telgte</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_pine, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.891592</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"spruce"</span>,]</span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a>jsd_results_spruce <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_spruce,<span class="st">"Jensen-Shannon Divergence between areas with spruce"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-1-jsd-spruce" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;21: Jensen-Shannon Divergence between the researched areas which have the dominante specie spruce for the atrribute nearest-neighbor-1</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Brilon</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oberhundem</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Osterwald</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Brilon</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">4.1</td>
<td style="text-align: left;">16</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oberhundem</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">18</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Osterwald</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_spruce, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4.471632</code></pre>
</div>
</div>
</section>
</section>
<section id="distribution-of-distances-to-100th-nearest-neighbor" class="level4">
<h4 class="anchored" data-anchor-id="distribution-of-distances-to-100th-nearest-neighbor">Distribution of distances to 100th nearest neighbor</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_combine_sf_obj</span>(lfa<span class="sc">::</span><span class="fu">lfa_get_neighbor_paths</span>(),lfa<span class="sc">::</span><span class="fu">lfa_get_all_areas</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/beech/bielefeld_brackwede/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 1443 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 466999.8 ymin: 5759839 xmax: 467617.1 ymax: 5760261
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/beech/billerbeck/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 1732 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 384890.8 ymin: 5761918 xmax: 385590.9 ymax: 5762478
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/beech/wuelfenrath/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 2779 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 365546.3 ymin: 5683711 xmax: 366356.1 ymax: 5684321
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/oak/hamm/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 2441 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 420953.3 ymin: 5723884 xmax: 421596 ymax: 5724609
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/oak/muenster/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 1270 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 404615.6 ymin: 5752535 xmax: 405396.8 ymax: 5752971
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/oak/rinkerode/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 1643 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 408428.2 ymin: 5746021 xmax: 409014.8 ymax: 5746511
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/pine/greffen/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 513 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 442816.1 ymin: 5760217 xmax: 443148.9 ymax: 5760567
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/pine/mesum/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 5031 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 399930.6 ymin: 5790412 xmax: 400969.7 ymax: 5790950
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/pine/telgte/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 3368 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 416135.1 ymin: 5761663 xmax: 416697.1 ymax: 5762477
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/spruce/brilon/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 3342 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 467305.7 ymin: 5695055 xmax: 467996.9 ymax: 5695593
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/spruce/oberhundem/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 2471 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 442631.7 ymin: 5660096 xmax: 443309.5 ymax: 5660502
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/spruce/osterwald/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 2806 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 455822 ymin: 5673761 xmax: 456483.2 ymax: 5674162
Projected CRS: ETRS89 / UTM zone 32N</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a>value_column <span class="ot">&lt;-</span> <span class="st">"Neighbor_100"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="kullback-leibler-divergence-2" class="level5">
<h5 class="anchored" data-anchor-id="kullback-leibler-divergence-2">Kullback-Leibler-Divergence</h5>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>kld_results_specie <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(data,value_column,<span class="st">"specie"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_specie,<span class="st">"Kullback-Leibler-Divergence between species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-100-kld_specie" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;22: Kullback-Leibler-Divergence between the researched species Beech, Oak, Pine and Spruce for the atrribute nearest-neighbor-100</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Beech</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oak</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Pine</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Spruce</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Beech</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.194</td>
<td style="text-align: left;">0.082</td>
<td style="text-align: left;">0.89</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oak</td>
<td style="text-align: left;">0.183</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.063</td>
<td style="text-align: left;">0.67</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Pine</td>
<td style="text-align: left;">0.084</td>
<td style="text-align: left;">0.069</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.86</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Spruce</td>
<td style="text-align: left;">1.083</td>
<td style="text-align: left;">0.809</td>
<td style="text-align: left;">1.200</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_specie, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.3862841</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"beech"</span>,]</span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a>kld_results_beech <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_beech,<span class="st">"Kullback-Leibler-Divergence between areas with beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-100-kld-beech" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;23: Kullback-Leibler-Divergence between the researched areas which have the dominante specie beech for the atrribute nearest-neighbor-100</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Bielefeld_brackwede</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Billerbeck</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Wuelfenrath</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Bielefeld_brackwede</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.12</td>
<td style="text-align: left;">0.12</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Billerbeck</td>
<td style="text-align: left;">0.14</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.40</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Wuelfenrath</td>
<td style="text-align: left;">0.12</td>
<td style="text-align: left;">0.31</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_beech, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1338066</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"oak"</span>,]</span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a>kld_results_oak <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb136-3"><a href="#cb136-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_oak,<span class="st">"Kullback-Leibler-Divergence between areas with oak"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-100-kld-oak" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;24: Kullback-Leibler-Divergence between the researched areas which have the dominante specie oak for the atrribute nearest-neighbor-100</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Hamm</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Muenster</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Rinkerode</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Hamm</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.19</td>
<td style="text-align: left;">0.11</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Muenster</td>
<td style="text-align: left;">0.20</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.06</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Rinkerode</td>
<td style="text-align: left;">0.11</td>
<td style="text-align: left;">0.07</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_oak, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.08182597</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"pine"</span>,]</span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a>kld_results_pine <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb139-3"><a href="#cb139-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_pine,<span class="st">"Kullback-Leibler-Divergence between areas with pine"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-100-kld-pine" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;25: Kullback-Leibler-Divergence between the researched areas which have the dominante specie pine for the atrribute nearest-neighbor-100</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Greffen</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Mesum</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Telgte</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Greffen</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.25</td>
<td style="text-align: left;">0.51</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Mesum</td>
<td style="text-align: left;">0.20</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.25</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Telgte</td>
<td style="text-align: left;">0.54</td>
<td style="text-align: left;">0.26</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_pine, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.22229</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"spruce"</span>,]</span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a>kld_results_spruce <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb142-3"><a href="#cb142-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_spruce,<span class="st">"Kullback-Leibler-Divergence between areas with spruce"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-100-kld-spruce" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;26: Kullback-Leibler-Divergence between the researched areas which have the dominante specie spruce for the atrribute nearest-neighbor-100</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Brilon</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oberhundem</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Osterwald</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Brilon</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.05</td>
<td style="text-align: left;">0.23</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oberhundem</td>
<td style="text-align: left;">0.046</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.37</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Osterwald</td>
<td style="text-align: left;">0.276</td>
<td style="text-align: left;">0.46</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_spruce, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1591879</code></pre>
</div>
</div>
</section>
<section id="jensen-shannon-divergence-2" class="level5">
<h5 class="anchored" data-anchor-id="jensen-shannon-divergence-2">Jensen-Shannon Divergence</h5>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a>jsd_results_specie <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(data,value_column,<span class="st">"specie"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_specie,<span class="st">"Jensen-Shannon Divergence between species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-100-jsd_specie" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;27: Jensen-Shannon Divergence between the researched species Beech, Oak, Pine and Spruce for the atrribute nearest-neighbor-100</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Beech</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oak</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Pine</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Spruce</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Beech</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.38</td>
<td style="text-align: left;">0.14</td>
<td style="text-align: left;">1.27</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oak</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.30</td>
<td style="text-align: left;">0.78</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Pine</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">1.39</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Spruce</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_specie, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.2997233</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"beech"</span>,]</span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a>jsd_results_beech <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb148-3"><a href="#cb148-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_beech,<span class="st">"Jensen-Shannon Divergence between areas with beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-100-jsd-beech" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;28: Jensen-Shannon Divergence between the researched areas which have the dominante specie beech for the atrribute nearest-neighbor-100</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Bielefeld_brackwede</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Billerbeck</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Wuelfenrath</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Bielefeld_brackwede</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.22</td>
<td style="text-align: left;">0.21</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Billerbeck</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.57</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Wuelfenrath</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_beech, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.124106</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"oak"</span>,]</span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a>jsd_results_oak <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_oak,<span class="st">"Jensen-Shannon Divergence between areas with oak"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-100-jsd-oak" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;29: Jensen-Shannon Divergence between the researched areas which have the dominante specie oak for the atrribute nearest-neighbor-100</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Hamm</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Muenster</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Rinkerode</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Hamm</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.34</td>
<td style="text-align: left;">0.17</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Muenster</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.23</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Rinkerode</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_oak, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1007612</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"pine"</span>,]</span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a>jsd_results_pine <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_pine,<span class="st">"Jensen-Shannon Divergence between areas with pine"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-100-jsd-pine" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;30: Jensen-Shannon Divergence between the researched areas which have the dominante specie pine for the atrribute nearest-neighbor-100</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Greffen</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Mesum</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Telgte</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Greffen</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.45</td>
<td style="text-align: left;">0.86</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Mesum</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.50</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Telgte</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_pine, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.2265055</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"spruce"</span>,]</span>
<span id="cb157-2"><a href="#cb157-2" aria-hidden="true" tabindex="-1"></a>jsd_results_spruce <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb157-3"><a href="#cb157-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_spruce,<span class="st">"Jensen-Shannon Divergence between areas with spruce"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-100-jsd-spruce" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;31: Jensen-Shannon Divergence between the researched areas which have the dominante specie spruce for the atrribute nearest-neighbor-100</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Brilon</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oberhundem</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Osterwald</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Brilon</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.1</td>
<td style="text-align: left;">0.57</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oberhundem</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
<td style="text-align: left;">0.73</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Osterwald</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb158"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_spruce, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1613747</code></pre>
</div>
</div>
</section>
</section>
<section id="distribution-of-average-nearest-neighbor-distances" class="level4">
<h4 class="anchored" data-anchor-id="distribution-of-average-nearest-neighbor-distances">Distribution of average nearest neighbor distances</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_combine_sf_obj</span>(lfa<span class="sc">::</span><span class="fu">lfa_get_neighbor_paths</span>(),lfa<span class="sc">::</span><span class="fu">lfa_get_all_areas</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/beech/bielefeld_brackwede/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 1443 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 466999.8 ymin: 5759839 xmax: 467617.1 ymax: 5760261
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/beech/billerbeck/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 1732 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 384890.8 ymin: 5761918 xmax: 385590.9 ymax: 5762478
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/beech/wuelfenrath/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 2779 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 365546.3 ymin: 5683711 xmax: 366356.1 ymax: 5684321
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/oak/hamm/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 2441 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 420953.3 ymin: 5723884 xmax: 421596 ymax: 5724609
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/oak/muenster/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 1270 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 404615.6 ymin: 5752535 xmax: 405396.8 ymax: 5752971
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/oak/rinkerode/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 1643 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 408428.2 ymin: 5746021 xmax: 409014.8 ymax: 5746511
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/pine/greffen/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 513 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 442816.1 ymin: 5760217 xmax: 443148.9 ymax: 5760567
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/pine/mesum/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 5031 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 399930.6 ymin: 5790412 xmax: 400969.7 ymax: 5790950
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/pine/telgte/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 3368 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 416135.1 ymin: 5761663 xmax: 416697.1 ymax: 5762477
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/spruce/brilon/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 3342 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 467305.7 ymin: 5695055 xmax: 467996.9 ymax: 5695593
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/spruce/oberhundem/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 2471 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 442631.7 ymin: 5660096 xmax: 443309.5 ymax: 5660502
Projected CRS: ETRS89 / UTM zone 32N
Reading layer `neighbours' from data source 
  `/home/jakob/gi-master/project-courses/lidar-forest-analysis/src/data/spruce/osterwald/neighbours.gpkg' 
  using driver `GPKG'
Simple feature collection with 2806 features and 102 fields
Geometry type: POINT
Dimension:     XY
Bounding box:  xmin: 455822 ymin: 5673761 xmax: 456483.2 ymax: 5674162
Projected CRS: ETRS89 / UTM zone 32N</code></pre>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb162"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Neighbor_"</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>)</span>
<span id="cb162-2"><a href="#cb162-2" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>avg <span class="ot">=</span> <span class="fu">rowMeans</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">as.data.frame</span>(data),names))</span>
<span id="cb162-3"><a href="#cb162-3" aria-hidden="true" tabindex="-1"></a>value_column <span class="ot">&lt;-</span> <span class="st">"avg"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="kullback-leibler-divergence-3" class="level5">
<h5 class="anchored" data-anchor-id="kullback-leibler-divergence-3">Kullback-Leibler-Divergence</h5>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb163"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a>kld_results_specie <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(data,value_column,<span class="st">"specie"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb163-2"><a href="#cb163-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_specie,<span class="st">"Kullback-Leibler-Divergence between species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-avg-kld_specie" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;32: Kullback-Leibler-Divergence between the researched species Beech, Oak, Pine and Spruce for the atrribute nearest-neighbor-avg</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Beech</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oak</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Pine</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Spruce</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Beech</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.31</td>
<td style="text-align: left;">0.065</td>
<td style="text-align: left;">1.28</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oak</td>
<td style="text-align: left;">0.302</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.178</td>
<td style="text-align: left;">0.83</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Pine</td>
<td style="text-align: left;">0.067</td>
<td style="text-align: left;">0.17</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">1.23</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Spruce</td>
<td style="text-align: left;">1.660</td>
<td style="text-align: left;">0.92</td>
<td style="text-align: left;">1.869</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb164"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_specie, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.5552882</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb166"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"beech"</span>,]</span>
<span id="cb166-2"><a href="#cb166-2" aria-hidden="true" tabindex="-1"></a>kld_results_beech <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb166-3"><a href="#cb166-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_beech,<span class="st">"Kullback-Leibler-Divergence between areas with beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-avg-kld-beech" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;33: Kullback-Leibler-Divergence between the researched areas which have the dominante specie beech for the atrribute nearest-neighbor-avg</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Bielefeld_brackwede</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Billerbeck</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Wuelfenrath</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Bielefeld_brackwede</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.052</td>
<td style="text-align: left;">0.50</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Billerbeck</td>
<td style="text-align: left;">0.052</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.91</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Wuelfenrath</td>
<td style="text-align: left;">0.348</td>
<td style="text-align: left;">0.612</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb167"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb167-1"><a href="#cb167-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_beech, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.27574</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb169"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"oak"</span>,]</span>
<span id="cb169-2"><a href="#cb169-2" aria-hidden="true" tabindex="-1"></a>kld_results_oak <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb169-3"><a href="#cb169-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_oak,<span class="st">"Kullback-Leibler-Divergence between areas with oak"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-avg-kld-oak" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;34: Kullback-Leibler-Divergence between the researched areas which have the dominante specie oak for the atrribute nearest-neighbor-avg</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Hamm</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Muenster</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Rinkerode</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Hamm</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.166</td>
<td style="text-align: left;">0.217</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Muenster</td>
<td style="text-align: left;">0.16</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.031</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Rinkerode</td>
<td style="text-align: left;">0.21</td>
<td style="text-align: left;">0.037</td>
<td style="text-align: left;">0.000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb170"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb170-1"><a href="#cb170-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_oak, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.09154318</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb172"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb172-1"><a href="#cb172-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"pine"</span>,]</span>
<span id="cb172-2"><a href="#cb172-2" aria-hidden="true" tabindex="-1"></a>kld_results_pine <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb172-3"><a href="#cb172-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_pine,<span class="st">"Kullback-Leibler-Divergence between areas with pine"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-avg-kld-pine" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;35: Kullback-Leibler-Divergence between the researched areas which have the dominante specie pine for the atrribute nearest-neighbor-avg</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Greffen</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Mesum</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Telgte</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Greffen</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.17</td>
<td style="text-align: left;">0.29</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Mesum</td>
<td style="text-align: left;">0.14</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.30</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Telgte</td>
<td style="text-align: left;">0.26</td>
<td style="text-align: left;">0.32</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb173"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb173-1"><a href="#cb173-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_pine, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1637513</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb175"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb175-1"><a href="#cb175-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"spruce"</span>,]</span>
<span id="cb175-2"><a href="#cb175-2" aria-hidden="true" tabindex="-1"></a>kld_results_spruce <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb175-3"><a href="#cb175-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_spruce,<span class="st">"Kullback-Leibler-Divergence between areas with spruce"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-avg-kld-spruce" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;36: Kullback-Leibler-Divergence between the researched areas which have the dominante specie spruce for the atrribute nearest-neighbor-avg</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Brilon</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oberhundem</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Osterwald</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Brilon</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.11</td>
<td style="text-align: left;">0.29</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oberhundem</td>
<td style="text-align: left;">0.097</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.59</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Osterwald</td>
<td style="text-align: left;">0.341</td>
<td style="text-align: left;">0.75</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb176"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb176-1"><a href="#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_spruce, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.2404004</code></pre>
</div>
</div>
</section>
<section id="jensen-shannon-divergence-3" class="level5">
<h5 class="anchored" data-anchor-id="jensen-shannon-divergence-3">Jensen-Shannon Divergence</h5>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb178"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a>jsd_results_specie <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(data,value_column,<span class="st">"specie"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb178-2"><a href="#cb178-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_specie,<span class="st">"Jensen-Shannon Divergence between species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-avg-jsd_specie" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;37: Jensen-Shannon Divergence between the researched species Beech, Oak, Pine and Spruce for the atrribute nearest-neighbor-avg</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Beech</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oak</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Pine</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Spruce</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Beech</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.73</td>
<td style="text-align: left;">0.19</td>
<td style="text-align: left;">2.6</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oak</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.64</td>
<td style="text-align: left;">1.4</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Pine</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">3.0</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Spruce</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb179"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb179-1"><a href="#cb179-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_specie, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.5999417</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb181"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb181-1"><a href="#cb181-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"beech"</span>,]</span>
<span id="cb181-2"><a href="#cb181-2" aria-hidden="true" tabindex="-1"></a>jsd_results_beech <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb181-3"><a href="#cb181-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_beech,<span class="st">"Jensen-Shannon Divergence between areas with beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-avg-jsd-beech" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;38: Jensen-Shannon Divergence between the researched areas which have the dominante specie beech for the atrribute nearest-neighbor-avg</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Bielefeld_brackwede</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Billerbeck</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Wuelfenrath</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Bielefeld_brackwede</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.14</td>
<td style="text-align: left;">1.0</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Billerbeck</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">1.7</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Wuelfenrath</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb182"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb182-1"><a href="#cb182-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_beech, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.3215991</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb184"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb184-1"><a href="#cb184-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"oak"</span>,]</span>
<span id="cb184-2"><a href="#cb184-2" aria-hidden="true" tabindex="-1"></a>jsd_results_oak <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb184-3"><a href="#cb184-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_oak,<span class="st">"Jensen-Shannon Divergence between areas with oak"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-avg-jsd-oak" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;39: Jensen-Shannon Divergence between the researched areas which have the dominante specie oak for the atrribute nearest-neighbor-avg</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Hamm</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Muenster</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Rinkerode</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Hamm</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.41</td>
<td style="text-align: left;">0.53</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Muenster</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.26</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Rinkerode</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb185"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_oak, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1558436</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb187"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"pine"</span>,]</span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a>jsd_results_pine <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb187-3"><a href="#cb187-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_pine,<span class="st">"Jensen-Shannon Divergence between areas with pine"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-avg-jsd-pine" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;40: Jensen-Shannon Divergence between the researched areas which have the dominante specie pine for the atrribute nearest-neighbor-avg</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Greffen</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Mesum</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Telgte</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Greffen</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.44</td>
<td style="text-align: left;">0.76</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Mesum</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.89</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Telgte</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb188"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb188-1"><a href="#cb188-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_pine, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.2560143</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb190"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb190-1"><a href="#cb190-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"spruce"</span>,]</span>
<span id="cb190-2"><a href="#cb190-2" aria-hidden="true" tabindex="-1"></a>jsd_results_spruce <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb190-3"><a href="#cb190-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_spruce,<span class="st">"Jensen-Shannon Divergence between areas with spruce"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-nearest-neighbor-avg-jsd-spruce" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;41: Jensen-Shannon Divergence between the researched areas which have the dominante specie spruce for the atrribute nearest-neighbor-avg</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Brilon</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oberhundem</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Osterwald</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Brilon</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.32</td>
<td style="text-align: left;">1.1</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oberhundem</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">1.8</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Osterwald</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb191"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb191-1"><a href="#cb191-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_spruce, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.3713411</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="distribution-of-the-number-of-returns" class="level3" data-number="6.4.3">
<h3 data-number="6.4.3" class="anchored" data-anchor-id="distribution-of-the-number-of-returns"><span class="header-section-number">6.4.3</span> Distribution of the number of returns</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb193"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb193-1"><a href="#cb193-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> sf<span class="sc">::</span><span class="fu">st_read</span>(<span class="st">"data/tree_properties.gpkg"</span>)</span>
<span id="cb193-2"><a href="#cb193-2" aria-hidden="true" tabindex="-1"></a>neighbors <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_get_neighbor_paths</span>() <span class="sc">|&gt;</span> lfa<span class="sc">::</span><span class="fu">lfa_combine_sf_obj</span>(lfa<span class="sc">::</span><span class="fu">lfa_get_all_areas</span>())</span>
<span id="cb193-3"><a href="#cb193-3" aria-hidden="true" tabindex="-1"></a>data <span class="ot">=</span> sf<span class="sc">::</span><span class="fu">st_join</span>(data,neighbors, <span class="at">join =</span> sf<span class="sc">::</span>st_within)</span>
<span id="cb193-4"><a href="#cb193-4" aria-hidden="true" tabindex="-1"></a>value_column <span class="ot">&lt;-</span> <span class="st">"number_of_returns"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="kullback-leibler-divergence-4" class="level4">
<h4 class="anchored" data-anchor-id="kullback-leibler-divergence-4">Kullback-Leibler-Divergence</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb194"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb194-1"><a href="#cb194-1" aria-hidden="true" tabindex="-1"></a>kld_results_specie <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(data,value_column,<span class="st">"specie"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb194-2"><a href="#cb194-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_specie,<span class="st">"Kullback-Leibler-Divergence between species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-number-of-returns-kld_specie" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;42: Kullback-Leibler-Divergence between the researched species Beech, Oak, Pine and Spruce for the atrribute number-of-returns</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Beech</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oak</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Pine</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Spruce</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Beech</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.083</td>
<td style="text-align: left;">0.57</td>
<td style="text-align: left;">0.049</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oak</td>
<td style="text-align: left;">0.051</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.84</td>
<td style="text-align: left;">0.059</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Pine</td>
<td style="text-align: left;">0.432</td>
<td style="text-align: left;">0.833</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.526</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Spruce</td>
<td style="text-align: left;">0.036</td>
<td style="text-align: left;">0.059</td>
<td style="text-align: left;">0.54</td>
<td style="text-align: left;">0.000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb195"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb195-1"><a href="#cb195-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_specie, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.2550987</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb197"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb197-1"><a href="#cb197-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"beech"</span>,]</span>
<span id="cb197-2"><a href="#cb197-2" aria-hidden="true" tabindex="-1"></a>kld_results_beech <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb197-3"><a href="#cb197-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_beech,<span class="st">"Kullback-Leibler-Divergence between areas with beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-number-of-returns-kld-beech" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;43: Kullback-Leibler-Divergence between the researched areas which have the dominante specie beech for the atrribute number-of-returns</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Bielefeld_brackwede</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Billerbeck</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Wuelfenrath</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Bielefeld_brackwede</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.15</td>
<td style="text-align: left;">0.082</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Billerbeck</td>
<td style="text-align: left;">0.21</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.136</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Wuelfenrath</td>
<td style="text-align: left;">0.13</td>
<td style="text-align: left;">0.19</td>
<td style="text-align: left;">0.000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_beech, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.09985223</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb200"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb200-1"><a href="#cb200-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"oak"</span>,]</span>
<span id="cb200-2"><a href="#cb200-2" aria-hidden="true" tabindex="-1"></a>kld_results_oak <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb200-3"><a href="#cb200-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_oak,<span class="st">"Kullback-Leibler-Divergence between areas with oak"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-number-of-returns-kld-oak" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;44: Kullback-Leibler-Divergence between the researched areas which have the dominante specie oak for the atrribute number-of-returns</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Hamm</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Muenster</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Rinkerode</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Hamm</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.46</td>
<td style="text-align: left;">0.846</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Muenster</td>
<td style="text-align: left;">0.41</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.077</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Rinkerode</td>
<td style="text-align: left;">0.81</td>
<td style="text-align: left;">0.09</td>
<td style="text-align: left;">0.000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb201"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb201-1"><a href="#cb201-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_oak, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.2994815</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb203"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb203-1"><a href="#cb203-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"pine"</span>,]</span>
<span id="cb203-2"><a href="#cb203-2" aria-hidden="true" tabindex="-1"></a>kld_results_pine <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb203-3"><a href="#cb203-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_pine,<span class="st">"Kullback-Leibler-Divergence between areas with pine"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-number-of-returns-kld-pine" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;45: Kullback-Leibler-Divergence between the researched areas which have the dominante specie pine for the atrribute number-of-returns</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Greffen</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Mesum</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Telgte</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Greffen</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.1444</td>
<td style="text-align: left;">0.1773</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Mesum</td>
<td style="text-align: left;">0.14</td>
<td style="text-align: left;">0.0000</td>
<td style="text-align: left;">0.0047</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Telgte</td>
<td style="text-align: left;">0.16</td>
<td style="text-align: left;">0.0045</td>
<td style="text-align: left;">0.0000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb204"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb204-1"><a href="#cb204-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_pine, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.07005788</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb206"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb206-1"><a href="#cb206-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"spruce"</span>,]</span>
<span id="cb206-2"><a href="#cb206-2" aria-hidden="true" tabindex="-1"></a>kld_results_spruce <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_asymmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_kld_from_vec)</span>
<span id="cb206-3"><a href="#cb206-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(kld_results_spruce,<span class="st">"Kullback-Leibler-Divergence between areas with spruce"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-number-of-returns-kld-spruce" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;46: Kullback-Leibler-Divergence between the researched areas which have the dominante specie spruce for the atrribute number-of-returns</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Brilon</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oberhundem</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Osterwald</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Brilon</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.04</td>
<td style="text-align: left;">0.034</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oberhundem</td>
<td style="text-align: left;">0.041</td>
<td style="text-align: left;">0.00</td>
<td style="text-align: left;">0.079</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Osterwald</td>
<td style="text-align: left;">0.045</td>
<td style="text-align: left;">0.10</td>
<td style="text-align: left;">0.000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb207"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(kld_results_spruce, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.03779495</code></pre>
</div>
</div>
</section>
<section id="jensen-shannon-divergence-4" class="level4">
<h4 class="anchored" data-anchor-id="jensen-shannon-divergence-4">Jensen-Shannon Divergence</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb209"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb209-1"><a href="#cb209-1" aria-hidden="true" tabindex="-1"></a>jsd_results_specie <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(data,value_column,<span class="st">"specie"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb209-2"><a href="#cb209-2" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_specie,<span class="st">"Jensen-Shannon Divergence between species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-number-of-returns-jsd_specie" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;47: Jensen-Shannon Divergence between the researched species Beech, Oak, Pine and Spruce for the atrribute number-of-returns</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Beech</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oak</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Pine</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Spruce</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Beech</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">3e-04</td>
<td style="text-align: left;">0.019</td>
<td style="text-align: left;">0.0014</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oak</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0e+00</td>
<td style="text-align: left;">0.021</td>
<td style="text-align: left;">0.0016</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Pine</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.000</td>
<td style="text-align: left;">0.0143</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Spruce</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb210"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb210-1"><a href="#cb210-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_specie, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.004419638</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb212"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"beech"</span>,]</span>
<span id="cb212-2"><a href="#cb212-2" aria-hidden="true" tabindex="-1"></a>jsd_results_beech <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb212-3"><a href="#cb212-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_beech,<span class="st">"Jensen-Shannon Divergence between areas with beech"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-number-of-returns-jsd-beech" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;48: Jensen-Shannon Divergence between the researched areas which have the dominante specie beech for the atrribute number-of-returns</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Bielefeld_brackwede</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Billerbeck</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Wuelfenrath</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Bielefeld_brackwede</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.0035</td>
<td style="text-align: left;">0.00099</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Billerbeck</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0000</td>
<td style="text-align: left;">0.00554</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Wuelfenrath</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb213"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_beech, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.001314268</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb215"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb215-1"><a href="#cb215-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"oak"</span>,]</span>
<span id="cb215-2"><a href="#cb215-2" aria-hidden="true" tabindex="-1"></a>jsd_results_oak <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb215-3"><a href="#cb215-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_oak,<span class="st">"Jensen-Shannon Divergence between areas with oak"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-number-of-returns-jsd-oak" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;49: Jensen-Shannon Divergence between the researched areas which have the dominante specie oak for the atrribute number-of-returns</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Hamm</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Muenster</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Rinkerode</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Hamm</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.0068</td>
<td style="text-align: left;">0.0128</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Muenster</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0000</td>
<td style="text-align: left;">0.0017</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Rinkerode</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb216"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb216-1"><a href="#cb216-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_oak, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.002747351</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb218"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb218-1"><a href="#cb218-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"pine"</span>,]</span>
<span id="cb218-2"><a href="#cb218-2" aria-hidden="true" tabindex="-1"></a>jsd_results_pine <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb218-3"><a href="#cb218-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_pine,<span class="st">"Jensen-Shannon Divergence between areas with pine"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-number-of-returns-jsd-pine" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;50: Jensen-Shannon Divergence between the researched areas which have the dominante specie pine for the atrribute number-of-returns</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Greffen</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Mesum</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Telgte</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Greffen</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.0035</td>
<td style="text-align: left;">0.00458</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Mesum</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0000</td>
<td style="text-align: left;">0.00037</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Telgte</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.00000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb219"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb219-1"><a href="#cb219-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_pine, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.001130537</code></pre>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb221"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb221-1"><a href="#cb221-1" aria-hidden="true" tabindex="-1"></a>specie <span class="ot">&lt;-</span> data[data<span class="sc">$</span>specie<span class="sc">==</span><span class="st">"spruce"</span>,]</span>
<span id="cb221-2"><a href="#cb221-2" aria-hidden="true" tabindex="-1"></a>jsd_results_spruce <span class="ot">&lt;-</span> lfa<span class="sc">::</span><span class="fu">lfa_run_test_symmetric</span>(specie,value_column,<span class="st">"area"</span>,lfa<span class="sc">::</span>lfa_jsd_from_vec)</span>
<span id="cb221-3"><a href="#cb221-3" aria-hidden="true" tabindex="-1"></a>lfa<span class="sc">::</span><span class="fu">lfa_generate_result_table_tests</span>(jsd_results_spruce,<span class="st">"Jensen-Shannon Divergence between areas with spruce"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-number-of-returns-jsd-spruce" class="anchored">
<table class="table table-bordered table-sm table-striped small" data-quarto-postprocess="true">
<caption>Table&nbsp;51: Jensen-Shannon Divergence between the researched areas which have the dominante specie spruce for the atrribute number-of-returns</caption>
<thead>
<tr class="header">
<th style="text-align: left;" data-quarto-table-cell-role="th"></th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Brilon</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Oberhundem</th>
<th style="text-align: left;" data-quarto-table-cell-role="th">Osterwald</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Brilon</td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">0.0069</td>
<td style="text-align: left;">0.005</td>
</tr>
<tr class="even">
<td style="text-align: left; font-weight: bold;">Oberhundem</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.0000</td>
<td style="text-align: left;">0.002</td>
</tr>
<tr class="odd">
<td style="text-align: left; font-weight: bold;">Osterwald</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">0.000</td>
</tr>
</tbody>
</table>
</div>


</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb222"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb222-1"><a href="#cb222-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(jsd_results_spruce, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span> <span class="fu">mean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.001939104</code></pre>
</div>
</div>
</section>
</section>
</section>
<section id="sec-documentation" class="level2" data-number="6.5">
<h2 data-number="6.5" class="anchored" data-anchor-id="sec-documentation"><span class="header-section-number">6.5</span> Documentation</h2>
<section id="lfa_calculate_patch_density" class="level3" data-number="6.5.1">
<h3 data-number="6.5.1" class="anchored" data-anchor-id="lfa_calculate_patch_density"><span class="header-section-number">6.5.1</span> <code>lfa_calculate_patch_density</code></h3>
<p>Calculate patch density for specified areas based on detection data</p>
<section id="arguments" class="level4">
<h4 class="anchored" data-anchor-id="arguments">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>areas_location</code></td>
<td>The file path to a shapefile containing spatial polygons representing the areas for which patch density needs to be calculated. Default is “research_areas.shp”.</td>
</tr>
<tr class="even">
<td><code>detections</code></td>
<td>A data frame containing detection information, where each row represents a detection and includes the ‘area’ column specifying the corresponding area. Default is obtained using lfa_get_detections().</td>
</tr>
</tbody>
</table>
</section>
<section id="description" class="level4">
<h4 class="anchored" data-anchor-id="description">Description</h4>
<p>This function calculates patch density for specified areas using detection data. It reads the spatial polygons from a shapefile, computes the area size for each patch, counts the number of detections in each patch, and calculates the patch density.</p>
</section>
<section id="value" class="level4">
<h4 class="anchored" data-anchor-id="value">Value</h4>
<p>A data frame with patch density information for each specified area. Columns include ‘name’ (area name), ‘geometry’ (polygon geometry), ‘area_size’ (patch area size), ‘detections’ (number of detections in the patch), and ‘density’ (computed patch density).</p>
</section>
<section id="examples" class="level4">
<h4 class="anchored" data-anchor-id="examples">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb224"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb224-1"><a href="#cb224-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming you have a shapefile 'your_research_areas.shp' and detection data</span></span>
<span id="cb224-2"><a href="#cb224-2" aria-hidden="true" tabindex="-1"></a><span class="co"># from lfa_get_detections()</span></span>
<span id="cb224-3"><a href="#cb224-3" aria-hidden="true" tabindex="-1"></a>density_data <span class="ot">&lt;-</span> <span class="fu">lfa_calculate_patch_density</span>(<span class="at">areas_location =</span> <span class="st">"your_research_areas.shp"</span>)</span>
<span id="cb224-4"><a href="#cb224-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(density_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage" class="level4">
<h4 class="anchored" data-anchor-id="usage">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb225"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb225-1"><a href="#cb225-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_calculate_patch_density</span>(</span>
<span id="cb225-2"><a href="#cb225-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">areas_location =</span> <span class="st">"research_areas.shp"</span>,</span>
<span id="cb225-3"><a href="#cb225-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">detections =</span> lfa<span class="sc">::</span><span class="fu">lfa_get_detections</span>()</span>
<span id="cb225-4"><a href="#cb225-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_calculate_rf_metrics" class="level3" data-number="6.5.2">
<h3 data-number="6.5.2" class="anchored" data-anchor-id="lfa_calculate_rf_metrics"><span class="header-section-number">6.5.2</span> <code>lfa_calculate_rf_metrics</code></h3>
<p>Calculate Precision and Recall Metrics for Random Forest Classification</p>
<section id="arguments-1" class="level4">
<h4 class="anchored" data-anchor-id="arguments-1">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>conf_matrix</code></td>
<td>Confusion matrix obtained from a Random Forest classification.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-1" class="level4">
<h4 class="anchored" data-anchor-id="description-1">Description</h4>
<p>This function calculates precision and recall metrics for each class based on the provided confusion matrix from a Random Forest classification.</p>
</section>
<section id="details" class="level4">
<h4 class="anchored" data-anchor-id="details">Details</h4>
<p>The function calculates precision and recall metrics for each class based on the confusion matrix obtained from a Random Forest classification.</p>
</section>
<section id="seealso" class="level4">
<h4 class="anchored" data-anchor-id="seealso">Seealso</h4>
<p><a href="#lfaprecisionperclass"><code>lfa_precision_per_class</code></a> , <a href="#lfarecallperclass"><code>lfa_recall_per_class</code></a></p>
</section>
<section id="value-1" class="level4">
<h4 class="anchored" data-anchor-id="value-1">Value</h4>
<p>A data frame containing precision and recall metrics for each class.</p>
</section>
<section id="examples-1" class="level4">
<h4 class="anchored" data-anchor-id="examples-1">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb226"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb226-1"><a href="#cb226-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example confusion matrix from a Random Forest classification</span></span>
<span id="cb226-2"><a href="#cb226-2" aria-hidden="true" tabindex="-1"></a>rf_cm <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="at">predicted =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>), <span class="at">actual =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"B"</span>))</span>
<span id="cb226-3"><a href="#cb226-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate precision and recall metrics</span></span>
<span id="cb226-4"><a href="#cb226-4" aria-hidden="true" tabindex="-1"></a>rf_metrics_df <span class="ot">&lt;-</span> <span class="fu">lfa_calculate_rf_metrics</span>(rf_cm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-1" class="level4">
<h4 class="anchored" data-anchor-id="usage-1">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb227"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb227-1"><a href="#cb227-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_calculate_rf_metrics</span>(conf_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_capitalize_first_char" class="level3" data-number="6.5.3">
<h3 data-number="6.5.3" class="anchored" data-anchor-id="lfa_capitalize_first_char"><span class="header-section-number">6.5.3</span> <code>lfa_capitalize_first_char</code></h3>
<p>Capitalize First Character of a String</p>
<section id="arguments-2" class="level4">
<h4 class="anchored" data-anchor-id="arguments-2">Arguments</h4>
<table class="table">
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>input_string</code></td>
<td>A single-character string to be processed.</td>
</tr>
</tbody>
</table>
</section>
<section id="concept" class="level4">
<h4 class="anchored" data-anchor-id="concept">Concept</h4>
<p>String Manipulation</p>
</section>
<section id="description-2" class="level4">
<h4 class="anchored" data-anchor-id="description-2">Description</h4>
<p>This function takes a string as input and returns the same string with the first character capitalized. If the first character is already capitalized, the function does nothing. If the first character is not from the alphabet, an error is thrown.</p>
</section>
<section id="details-1" class="level4">
<h4 class="anchored" data-anchor-id="details-1">Details</h4>
<p>This function performs the following steps:</p>
<ul>
<li><p>Checks if the input is a single-character string.</p></li>
<li><p>Verifies if the first character is from the alphabet (A-Z or a-z).</p></li>
<li><p>If the first character is not already capitalized, it capitalizes it.</p></li>
<li><p>Returns the modified string.</p></li>
</ul>
</section>
<section id="keyword" class="level4">
<h4 class="anchored" data-anchor-id="keyword">Keyword</h4>
<p>alphabet</p>
</section>
<section id="note" class="level4">
<h4 class="anchored" data-anchor-id="note">Note</h4>
<p>This function is case-sensitive and assumes ASCII characters.</p>
</section>
<section id="references-1" class="level4">
<h4 class="anchored" data-anchor-id="references-1">References</h4>
<p>None</p>
</section>
<section id="seealso-1" class="level4">
<h4 class="anchored" data-anchor-id="seealso-1">Seealso</h4>
<p>This function is related to the basic string manipulation functions in base R.</p>
</section>
<section id="value-2" class="level4">
<h4 class="anchored" data-anchor-id="value-2">Value</h4>
<p>A modified string with the first character capitalized if it is not already. If the first character is already capitalized, the original string is returned.</p>
</section>
<section id="examples-2" class="level4">
<h4 class="anchored" data-anchor-id="examples-2">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb228"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb228-1"><a href="#cb228-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Capitalize the first character of a string</span></span>
<span id="cb228-2"><a href="#cb228-2" aria-hidden="true" tabindex="-1"></a><span class="fu">capitalize_first_char</span>(<span class="st">"hello"</span>) <span class="co"># Returns "Hello"</span></span>
<span id="cb228-3"><a href="#cb228-3" aria-hidden="true" tabindex="-1"></a><span class="fu">capitalize_first_char</span>(<span class="st">"World"</span>) <span class="co"># Returns "World"</span></span>
<span id="cb228-4"><a href="#cb228-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb228-5"><a href="#cb228-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Error example (non-alphabetic first character)</span></span>
<span id="cb228-6"><a href="#cb228-6" aria-hidden="true" tabindex="-1"></a><span class="fu">capitalize_first_char</span>(<span class="st">"123abc"</span>) <span class="co"># Throws an error</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-2" class="level4">
<h4 class="anchored" data-anchor-id="usage-2">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb229"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb229-1"><a href="#cb229-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_capitalize_first_char</span>(input_string)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_check_flag" class="level3" data-number="6.5.4">
<h3 data-number="6.5.4" class="anchored" data-anchor-id="lfa_check_flag"><span class="header-section-number">6.5.4</span> <code>lfa_check_flag</code></h3>
<p>Check if a flag is set, indicating the completion of a specific process.</p>
<section id="arguments-3" class="level4">
<h4 class="anchored" data-anchor-id="arguments-3">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>flag_name</code></td>
<td>A character string specifying the name of the flag file. It should be a descriptive and unique identifier for the process being checked.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-3" class="level4">
<h4 class="anchored" data-anchor-id="description-3">Description</h4>
<p>This function checks for the existence of a hidden flag file at a specified location within the working directory. If the flag file is found, a message is printed, and the function returns <code>TRUE</code> to indicate that the associated processing step has already been completed. If the flag file is not found, the function returns <code>FALSE</code> , indicating that further processing can proceed.</p>
</section>
<section id="value-3" class="level4">
<h4 class="anchored" data-anchor-id="value-3">Value</h4>
<p>A logical value indicating whether the flag is set ( <code>TRUE</code> ) or not ( <code>FALSE</code> ).</p>
</section>
<section id="examples-3" class="level4">
<h4 class="anchored" data-anchor-id="examples-3">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb230"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb230-1"><a href="#cb230-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check if the flag for a process named "data_processing" is set</span></span>
<span id="cb230-2"><a href="#cb230-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_check_flag</span>(<span class="st">"data_processing"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-3" class="level4">
<h4 class="anchored" data-anchor-id="usage-3">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb231"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb231-1"><a href="#cb231-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_check_flag</span>(flag_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_chm" class="level3" data-number="6.5.5">
<h3 data-number="6.5.5" class="anchored" data-anchor-id="lfa_chm"><span class="header-section-number">6.5.5</span> <code>lfa_chm</code></h3>
<p>Create Canopy Height Model (CHM) from Lidar Data</p>
<section id="arguments-4" class="level4">
<h4 class="anchored" data-anchor-id="arguments-4">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>specie</code></td>
<td>Character string indicating the species name.</td>
</tr>
<tr class="even">
<td><code>area</code></td>
<td>Character string indicating the specific area or location.</td>
</tr>
<tr class="odd">
<td><code>res</code></td>
<td>Numeric value indicating the spatial resolution of the CHM. Default is 0.5.</td>
</tr>
<tr class="even">
<td><code>save_to_file</code></td>
<td>Logical. If TRUE, the generated CHM will be saved to a GeoTIFF file. Default is TRUE.</td>
</tr>
<tr class="odd">
<td><code>overwrite</code></td>
<td>Logical. If TRUE, existing CHM file will be overwritten. Default is FALSE.</td>
</tr>
<tr class="even">
<td><code>...</code></td>
<td>Additional arguments to be passed to the underlying functions, such as lidR::catalog_map.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-4" class="level4">
<h4 class="anchored" data-anchor-id="description-4">Description</h4>
<p>This function generates a Canopy Height Model (CHM) from Lidar data using the lidR package.</p>
</section>
<section id="details-2" class="level4">
<h4 class="anchored" data-anchor-id="details-2">Details</h4>
<p>The behavior of the function with different input parameters is as follows:</p>
<ul>
<li><p>When a CHM file already exists at the specified path and <code>overwrite</code> is FALSE, the function loads the existing CHM and returns it.</p></li>
<li><p>If the CHM file does not exist or <code>overwrite</code> is TRUE, the function processes Lidar data using <code>lfa_rasterize_chunk</code> and creates a CHM.</p></li>
<li><p>The spatial resolution of the CHM can be controlled with the <code>res</code> parameter.</p></li>
<li><p>If <code>save_to_file</code> is TRUE, the generated CHM will be saved to a GeoTIFF file.</p></li>
</ul>
</section>
<section id="seealso-2" class="level4">
<h4 class="anchored" data-anchor-id="seealso-2">Seealso</h4>
<p><a href="#lfareadareaascatalog"><code>lfa_read_area_as_catalog</code></a> , <a href="#catalogmap"><code>catalog_map</code></a> , <a href="#terra::rast"><code>terra::rast</code></a> , <a href="#terra::writeraster"><code>terra::writeRaster</code></a></p>
</section>
<section id="value-4" class="level4">
<h4 class="anchored" data-anchor-id="value-4">Value</h4>
<p>A raster layer representing the Canopy Height Model (CHM).</p>
</section>
<section id="examples-4" class="level4">
<h4 class="anchored" data-anchor-id="examples-4">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb232"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb232-1"><a href="#cb232-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate CHM for a specific species and area</span></span>
<span id="cb232-2"><a href="#cb232-2" aria-hidden="true" tabindex="-1"></a>chm <span class="ot">&lt;-</span> <span class="fu">lfa_chm</span>(<span class="at">specie =</span> <span class="st">"ExampleSpecies"</span>, <span class="at">area =</span> <span class="st">"ExampleArea"</span>, <span class="at">res =</span> <span class="fl">1.0</span>)</span>
<span id="cb232-3"><a href="#cb232-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb232-4"><a href="#cb232-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate CHM and save it to a file</span></span>
<span id="cb232-5"><a href="#cb232-5" aria-hidden="true" tabindex="-1"></a>chm <span class="ot">&lt;-</span> <span class="fu">lfa_chm</span>(<span class="at">specie =</span> <span class="st">"ExampleSpecies"</span>, <span class="at">area =</span> <span class="st">"ExampleArea"</span>, <span class="at">res =</span> <span class="fl">1.0</span>, <span class="at">save_to_file =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-4" class="level4">
<h4 class="anchored" data-anchor-id="usage-4">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb233"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb233-1"><a href="#cb233-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_chm</span>(specie, area, <span class="at">res =</span> <span class="fl">0.5</span>, <span class="at">save_to_file =</span> <span class="cn">TRUE</span>, <span class="at">overwrite =</span> <span class="cn">FALSE</span>, ...)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_combine_sf_obj" class="level3" data-number="6.5.6">
<h3 data-number="6.5.6" class="anchored" data-anchor-id="lfa_combine_sf_obj"><span class="header-section-number">6.5.6</span> <code>lfa_combine_sf_obj</code></h3>
<p>Combine Spatial Feature Objects from Multiple GeoPackage Files</p>
<section id="arguments-5" class="level4">
<h4 class="anchored" data-anchor-id="arguments-5">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>paths</code></td>
<td>A character vector containing file paths to GeoPackage files with neighbor information.</td>
</tr>
<tr class="even">
<td><code>area_infos</code></td>
<td>A data frame or list containing information about the corresponding detection areas, including “area” and “specie” columns.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-5" class="level4">
<h4 class="anchored" data-anchor-id="description-5">Description</h4>
<p>This function reads spatial feature objects (sf) from multiple GeoPackage files and combines them into a single sf object. Each GeoPackage file is assumed to contain neighbor information for a specific detection area, and the resulting sf object includes additional columns indicating the corresponding area and species information.</p>
</section>
<section id="value-5" class="level4">
<h4 class="anchored" data-anchor-id="value-5">Value</h4>
<p>A combined sf object with additional columns for area and specie information.</p>
</section>
<section id="examples-5" class="level4">
<h4 class="anchored" data-anchor-id="examples-5">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb234"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb234-1"><a href="#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming paths and area_infos are defined</span></span>
<span id="cb234-2"><a href="#cb234-2" aria-hidden="true" tabindex="-1"></a>combined_sf <span class="ot">&lt;-</span> <span class="fu">lfa_combine_sf_obj</span>(paths, area_infos)</span>
<span id="cb234-3"><a href="#cb234-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb234-4"><a href="#cb234-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the combined sf object</span></span>
<span id="cb234-5"><a href="#cb234-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(combined_sf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-5" class="level4">
<h4 class="anchored" data-anchor-id="usage-5">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb235"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb235-1"><a href="#cb235-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_combine_sf_obj</span>(paths, area_infos)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_count_returns_all_areas" class="level3" data-number="6.5.7">
<h3 data-number="6.5.7" class="anchored" data-anchor-id="lfa_count_returns_all_areas"><span class="header-section-number">6.5.7</span> <code>lfa_count_returns_all_areas</code></h3>
<p>Count tree returns for all species and areas, returning a consolidated data frame.</p>
<section id="description-6" class="level4">
<h4 class="anchored" data-anchor-id="description-6">Description</h4>
<p>This function iterates through all species and areas obtained from the function <a href="#lfagetallareas"><code>lfa_get_all_areas</code></a> . For each combination of species and area, it reads the corresponding area as a catalog, counts the returns per tree using <a href="#lfacountreturnspertree"><code>lfa_count_returns_per_tree</code></a> , and consolidates the results into a data frame. The resulting data frame includes columns for the species, area, and return counts per tree.</p>
</section>
<section id="keyword-1" class="level4">
<h4 class="anchored" data-anchor-id="keyword-1">Keyword</h4>
<p>counting</p>
</section>
<section id="seealso-3" class="level4">
<h4 class="anchored" data-anchor-id="seealso-3">Seealso</h4>
<p><a href="#lfagetallareas"><code>lfa_get_all_areas</code></a> , <a href="#lfareadareaascatalog"><code>lfa_read_area_as_catalog</code></a> , <a href="#lfacountreturnspertree"><code>lfa_count_returns_per_tree</code></a></p>
</section>
<section id="value-6" class="level4">
<h4 class="anchored" data-anchor-id="value-6">Value</h4>
<p>A data frame with columns for species, area, and return counts per tree.</p>
</section>
<section id="examples-6" class="level4">
<h4 class="anchored" data-anchor-id="examples-6">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb236"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb236-1"><a href="#cb236-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Count tree returns for all species and areas</span></span>
<span id="cb236-2"><a href="#cb236-2" aria-hidden="true" tabindex="-1"></a>returns_counts <span class="ot">&lt;-</span> <span class="fu">lfa_count_returns_all_areas</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-6" class="level4">
<h4 class="anchored" data-anchor-id="usage-6">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb237"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb237-1"><a href="#cb237-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_count_returns_all_areas</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_count_returns_per_tree" class="level3" data-number="6.5.8">
<h3 data-number="6.5.8" class="anchored" data-anchor-id="lfa_count_returns_per_tree"><span class="header-section-number">6.5.8</span> <code>lfa_count_returns_per_tree</code></h3>
<p>Count returns per tree for a given lidR catalog.</p>
<section id="arguments-6" class="level4">
<h4 class="anchored" data-anchor-id="arguments-6">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>ctg</code></td>
<td>A lidR catalog object containing LAS files to be processed.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-7" class="level4">
<h4 class="anchored" data-anchor-id="description-7">Description</h4>
<p>This function takes a lidR catalog as input and counts the returns per tree. It uses the lidR package to read LAS files from the catalog and performs the counting operation on each tree. The result is a data frame containing the counts of returns for each unique tree ID within the lidR catalog.</p>
</section>
<section id="keyword-2" class="level4">
<h4 class="anchored" data-anchor-id="keyword-2">Keyword</h4>
<p>counting</p>
</section>
<section id="seealso-4" class="level4">
<h4 class="anchored" data-anchor-id="seealso-4">Seealso</h4>
<p><a href="#lidr::readlas"><code>lidR::readLAS</code></a> , <a href="#lidr::is.empty"><code>lidR::is.empty</code></a> , <a href="#base::table"><code>base::table</code></a> , <a href="#dplyr::bindrows"><code>dplyr::bind_rows</code></a></p>
</section>
<section id="value-7" class="level4">
<h4 class="anchored" data-anchor-id="value-7">Value</h4>
<p>A data frame with columns for tree ID and the corresponding count of returns.</p>
</section>
<section id="examples-7" class="level4">
<h4 class="anchored" data-anchor-id="examples-7">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb238"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb238-1"><a href="#cb238-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Count returns per tree for a lidR catalog</span></span>
<span id="cb238-2"><a href="#cb238-2" aria-hidden="true" tabindex="-1"></a>ctg <span class="ot">&lt;-</span> <span class="fu">lfa_read_area_as_catalog</span>(<span class="st">"SpeciesA"</span>, <span class="st">"Area1"</span>)</span>
<span id="cb238-3"><a href="#cb238-3" aria-hidden="true" tabindex="-1"></a>returns_counts_per_tree <span class="ot">&lt;-</span> <span class="fu">lfa_count_returns_per_tree</span>(ctg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-7" class="level4">
<h4 class="anchored" data-anchor-id="usage-7">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb239"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb239-1"><a href="#cb239-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_count_returns_per_tree</span>(ctg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_create_boxplot" class="level3" data-number="6.5.9">
<h3 data-number="6.5.9" class="anchored" data-anchor-id="lfa_create_boxplot"><span class="header-section-number">6.5.9</span> <code>lfa_create_boxplot</code></h3>
<p>Create a box plot from a data frame</p>
<section id="arguments-7" class="level4">
<h4 class="anchored" data-anchor-id="arguments-7">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>data</code></td>
<td>A data frame containing the data.</td>
</tr>
<tr class="even">
<td><code>value_column</code></td>
<td>The name of the column containing the values for the box plot.</td>
</tr>
<tr class="odd">
<td><code>category_column1</code></td>
<td>The name of the column containing the first categorical variable.</td>
</tr>
<tr class="even">
<td><code>category_column2</code></td>
<td>The name of the column containing the second categorical variable.</td>
</tr>
<tr class="odd">
<td><code>title</code></td>
<td>An optional title for the plot. If not provided, a default title is generated based on the data frame name.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-8" class="level4">
<h4 class="anchored" data-anchor-id="description-8">Description</h4>
<p>This function generates a box plot using ggplot2 based on the specified data frame and columns.</p>
</section>
<section id="details-3" class="level4">
<h4 class="anchored" data-anchor-id="details-3">Details</h4>
<p>The function creates a box plot where the x-axis is based on the second categorical variable, the y-axis is based on the specified value column, and the box plots are colored based on the first categorical variable. The grouping of box plots is done based on the unique values in the second categorical variable.</p>
</section>
<section id="value-8" class="level4">
<h4 class="anchored" data-anchor-id="value-8">Value</h4>
<p>A ggplot object representing the box plot.</p>
</section>
<section id="examples-8" class="level4">
<h4 class="anchored" data-anchor-id="examples-8">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb240"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming you have a data frame 'your_data' with columns 'value', 'category1', and 'category2'</span></span>
<span id="cb240-2"><a href="#cb240-2" aria-hidden="true" tabindex="-1"></a><span class="fu">create_boxplot</span>(your_data, <span class="st">"value"</span>, <span class="st">"category1"</span>, <span class="st">"category2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-8" class="level4">
<h4 class="anchored" data-anchor-id="usage-8">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb241"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb241-1"><a href="#cb241-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_boxplot</span>(</span>
<span id="cb241-2"><a href="#cb241-2" aria-hidden="true" tabindex="-1"></a>  data,</span>
<span id="cb241-3"><a href="#cb241-3" aria-hidden="true" tabindex="-1"></a>  value_column,</span>
<span id="cb241-4"><a href="#cb241-4" aria-hidden="true" tabindex="-1"></a>  category_column1,</span>
<span id="cb241-5"><a href="#cb241-5" aria-hidden="true" tabindex="-1"></a>  category_column2,</span>
<span id="cb241-6"><a href="#cb241-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">title =</span> <span class="cn">NULL</span></span>
<span id="cb241-7"><a href="#cb241-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_create_density_plots" class="level3" data-number="6.5.10">
<h3 data-number="6.5.10" class="anchored" data-anchor-id="lfa_create_density_plots"><span class="header-section-number">6.5.10</span> <code>lfa_create_density_plots</code></h3>
<p>Create density plots for groups in a data frame</p>
<section id="arguments-8" class="level4">
<h4 class="anchored" data-anchor-id="arguments-8">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>data</code></td>
<td>A data frame containing the data.</td>
</tr>
<tr class="even">
<td><code>value_column</code></td>
<td>The name of the column containing the values for the density plot.</td>
</tr>
<tr class="odd">
<td><code>category_column1</code></td>
<td>The name of the column containing the categorical variable for grouping.</td>
</tr>
<tr class="even">
<td><code>category_column2</code></td>
<td>The name of the column containing the categorical variable for arranging plots.</td>
</tr>
<tr class="odd">
<td><code>title</code></td>
<td>An optional title for the plot. If not provided, a default title is generated based on the data frame name.</td>
</tr>
<tr class="even">
<td><code>xlims</code></td>
<td>Optional limits for the x-axis. Should be a numeric vector with two elements (lower and upper bounds).</td>
</tr>
<tr class="odd">
<td><code>ylims</code></td>
<td>Optional limits for the y-axis. Should be a numeric vector with two elements (lower and upper bounds).</td>
</tr>
</tbody>
</table>
</section>
<section id="description-9" class="level4">
<h4 class="anchored" data-anchor-id="description-9">Description</h4>
<p>This function generates density plots using ggplot2 based on the specified data frame and columns.</p>
</section>
<section id="details-4" class="level4">
<h4 class="anchored" data-anchor-id="details-4">Details</h4>
<p>The function creates density plots where the x-axis is based on the specified value column, and the density plots are colored based on the first categorical variable. The arrangement of plots is done based on the unique values in the second categorical variable. The plots are arranged in a 2x2 grid.</p>
</section>
<section id="value-9" class="level4">
<h4 class="anchored" data-anchor-id="value-9">Value</h4>
<p>A ggplot object representing the density plots arranged in a 2x2 grid.</p>
</section>
<section id="examples-9" class="level4">
<h4 class="anchored" data-anchor-id="examples-9">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb242"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb242-1"><a href="#cb242-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming you have a data frame 'your_data' with columns 'value', 'category1', and 'category2'</span></span>
<span id="cb242-2"><a href="#cb242-2" aria-hidden="true" tabindex="-1"></a><span class="fu">create_density_plots</span>(your_data, <span class="st">"value"</span>, <span class="st">"category1"</span>, <span class="st">"category2"</span>, <span class="at">title =</span> <span class="st">"Density Plots"</span>, <span class="at">xlims =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">10</span>), <span class="at">ylims =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-9" class="level4">
<h4 class="anchored" data-anchor-id="usage-9">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb243"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb243-1"><a href="#cb243-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_density_plots</span>(</span>
<span id="cb243-2"><a href="#cb243-2" aria-hidden="true" tabindex="-1"></a>  data,</span>
<span id="cb243-3"><a href="#cb243-3" aria-hidden="true" tabindex="-1"></a>  value_column,</span>
<span id="cb243-4"><a href="#cb243-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">category_column1 =</span> <span class="st">"area"</span>,</span>
<span id="cb243-5"><a href="#cb243-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">category_column2 =</span> <span class="st">"specie"</span>,</span>
<span id="cb243-6"><a href="#cb243-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">title =</span> <span class="cn">NULL</span>,</span>
<span id="cb243-7"><a href="#cb243-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlims =</span> <span class="cn">NULL</span>,</span>
<span id="cb243-8"><a href="#cb243-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylims =</span> <span class="cn">NULL</span></span>
<span id="cb243-9"><a href="#cb243-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_create_grouped_bar_plot" class="level3" data-number="6.5.11">
<h3 data-number="6.5.11" class="anchored" data-anchor-id="lfa_create_grouped_bar_plot"><span class="header-section-number">6.5.11</span> <code>lfa_create_grouped_bar_plot</code></h3>
<p>Create a barplot using ggplot2</p>
<section id="arguments-9" class="level4">
<h4 class="anchored" data-anchor-id="arguments-9">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>xlab</code></td>
<td>Label for x-Axis.</td>
</tr>
<tr class="even">
<td><code>ylab</code></td>
<td>Label for y-Axis.</td>
</tr>
<tr class="odd">
<td><code>title</code></td>
<td>Title of the plot.</td>
</tr>
<tr class="even">
<td><code>df</code></td>
<td>A data frame containing the relevant columns for the barplot.</td>
</tr>
<tr class="odd">
<td><code>value_column</code></td>
<td>The column containing the values to be plotted.</td>
</tr>
<tr class="even">
<td><code>label_column</code></td>
<td>The column used for labeling the bars on the x-axis. Default is “name”.</td>
</tr>
<tr class="odd">
<td><code>grouping_column</code></td>
<td>The column used for grouping the bars. Default is “species”.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-10" class="level4">
<h4 class="anchored" data-anchor-id="description-10">Description</h4>
<p>This function generates a barplot using ggplot2 based on the specified data frame columns. The barplot displays the values from the specified column, grouped by another column. The grouping can be further differentiated by color if desired.</p>
</section>
<section id="value-10" class="level4">
<h4 class="anchored" data-anchor-id="value-10">Value</h4>
<p>A ggplot2 barplot.</p>
</section>
<section id="examples-10" class="level4">
<h4 class="anchored" data-anchor-id="examples-10">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb244"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb244-1"><a href="#cb244-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming you have a data frame 'your_data_frame' with columns "name", "species", and "value"</span></span>
<span id="cb244-2"><a href="#cb244-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_barplot</span>(your_data_frame, <span class="at">value_column =</span> <span class="st">"value"</span>, <span class="at">label_column =</span> <span class="st">"name"</span>, <span class="at">grouping_column =</span> <span class="st">"species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-10" class="level4">
<h4 class="anchored" data-anchor-id="usage-10">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb245"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb245-1"><a href="#cb245-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_grouped_bar_plot</span>(</span>
<span id="cb245-2"><a href="#cb245-2" aria-hidden="true" tabindex="-1"></a>  data,</span>
<span id="cb245-3"><a href="#cb245-3" aria-hidden="true" tabindex="-1"></a>  grouping_var,</span>
<span id="cb245-4"><a href="#cb245-4" aria-hidden="true" tabindex="-1"></a>  value_col,</span>
<span id="cb245-5"><a href="#cb245-5" aria-hidden="true" tabindex="-1"></a>  label_col,</span>
<span id="cb245-6"><a href="#cb245-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">"Name of Patch"</span>,</span>
<span id="cb245-7"><a href="#cb245-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylab =</span> <span class="st">"Density"</span>,</span>
<span id="cb245-8"><a href="#cb245-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">title =</span> <span class="st">"Tree density across the different patches, grouped by specie"</span></span>
<span id="cb245-9"><a href="#cb245-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_create_neighbor_mean_curves" class="level3" data-number="6.5.12">
<h3 data-number="6.5.12" class="anchored" data-anchor-id="lfa_create_neighbor_mean_curves"><span class="header-section-number">6.5.12</span> <code>lfa_create_neighbor_mean_curves</code></h3>
<p>Create neighbor mean curves for specified areas</p>
<section id="arguments-10" class="level4">
<h4 class="anchored" data-anchor-id="arguments-10">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>neighbors</code></td>
<td>A data frame containing information about neighbors, where each column represents a specific neighbor, and each row corresponds to an area.</td>
</tr>
<tr class="even">
<td><code>use_avg</code></td>
<td>Logical. If TRUE, the function computes average curves across all neighbors. If FALSE, it computes curves for individual neighbors.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-11" class="level4">
<h4 class="anchored" data-anchor-id="description-11">Description</h4>
<p>This function generates mean curves for a specified set of areas based on neighbor data. The user can choose to compute mean curves for individual neighbors or averages across neighbors.</p>
</section>
<section id="value-11" class="level4">
<h4 class="anchored" data-anchor-id="value-11">Value</h4>
<p>A data frame with mean curves for each specified area. Columns represent areas, and rows represent index values.</p>
</section>
<section id="examples-11" class="level4">
<h4 class="anchored" data-anchor-id="examples-11">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb246"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb246-1"><a href="#cb246-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming you have a data frame 'your_neighbors_data' with neighbor information</span></span>
<span id="cb246-2"><a href="#cb246-2" aria-hidden="true" tabindex="-1"></a>mean_curves <span class="ot">&lt;-</span> <span class="fu">lfa_create_neighbor_mean_curves</span>(your_neighbors_data, <span class="at">use_avg =</span> <span class="cn">TRUE</span>)</span>
<span id="cb246-3"><a href="#cb246-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(mean_curves)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-11" class="level4">
<h4 class="anchored" data-anchor-id="usage-11">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb247"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb247-1"><a href="#cb247-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_neighbor_mean_curves</span>(neighbors, <span class="at">use_avg =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_create_plot_per_area" class="level3" data-number="6.5.13">
<h3 data-number="6.5.13" class="anchored" data-anchor-id="lfa_create_plot_per_area"><span class="header-section-number">6.5.13</span> <code>lfa_create_plot_per_area</code></h3>
<p>Create a line plot per area with one color per specie</p>
<section id="arguments-11" class="level4">
<h4 class="anchored" data-anchor-id="arguments-11">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>data</code></td>
<td>A data frame with numeric columns and a column named ‘specie’ for species information.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-12" class="level4">
<h4 class="anchored" data-anchor-id="description-12">Description</h4>
<p>This function takes a data frame containing numeric columns and creates a line plot using ggplot2. Each line in the plot represents a different area, with one color per specie.</p>
</section>
<section id="value-12" class="level4">
<h4 class="anchored" data-anchor-id="value-12">Value</h4>
<p>A ggplot2 line plot.</p>
</section>
<section id="examples-12" class="level4">
<h4 class="anchored" data-anchor-id="examples-12">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb248"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb248-1"><a href="#cb248-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb248-2"><a href="#cb248-2" aria-hidden="true" tabindex="-1"></a><span class="at">specie =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">"Species1"</span>, <span class="st">"Species2"</span>, <span class="st">"Species3"</span>), <span class="at">each =</span> <span class="dv">10</span>),</span>
<span id="cb248-3"><a href="#cb248-3" aria-hidden="true" tabindex="-1"></a><span class="at">column1 =</span> <span class="fu">rnorm</span>(<span class="dv">30</span>),</span>
<span id="cb248-4"><a href="#cb248-4" aria-hidden="true" tabindex="-1"></a><span class="at">column2 =</span> <span class="fu">rnorm</span>(<span class="dv">30</span>),</span>
<span id="cb248-5"><a href="#cb248-5" aria-hidden="true" tabindex="-1"></a><span class="at">column3 =</span> <span class="fu">rnorm</span>(<span class="dv">30</span>)</span>
<span id="cb248-6"><a href="#cb248-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb248-7"><a href="#cb248-7" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_plot_per_area</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-12" class="level4">
<h4 class="anchored" data-anchor-id="usage-12">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb249"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb249-1"><a href="#cb249-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_plot_per_area</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_create_ppp_from_area" class="level3" data-number="6.5.14">
<h3 data-number="6.5.14" class="anchored" data-anchor-id="lfa_create_ppp_from_area"><span class="header-section-number">6.5.14</span> <code>lfa_create_ppp_from_area</code></h3>
<p>Create a point pattern from tree detections in a specified area for a given species.</p>
<section id="arguments-12" class="level4">
<h4 class="anchored" data-anchor-id="arguments-12">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>species_identifier</code></td>
<td>A character string specifying the target species for which the point pattern is to be generated.</td>
</tr>
<tr class="even">
<td><code>area_identifier</code></td>
<td>A character string specifying the target area for which the point pattern is to be generated.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-13" class="level4">
<h4 class="anchored" data-anchor-id="description-13">Description</h4>
<p>This function generates a point pattern from tree detections for a specific species within a defined area. It filters the detections using the provided <code>species_identifier</code> and <code>area_identifier</code> parameters. The area is defined by a shapefile named “research_areas.shp,” and the resulting point pattern is created within the specified area.</p>
</section>
<section id="keyword-3" class="level4">
<h4 class="anchored" data-anchor-id="keyword-3">Keyword</h4>
<p>data</p>
</section>
<section id="seealso-5" class="level4">
<h4 class="anchored" data-anchor-id="seealso-5">Seealso</h4>
<p><a href="#lfagetdetections"><code>lfa_get_detections</code></a> , <a href="#sf::sttransform"><code>sf::st_transform</code></a> , <a href="#sf::stunion"><code>sf::st_union</code></a> , <a href="#spatstat.geom::as.owin"><code>spatstat.geom::as.owin</code></a> , <a href="#spatstat.geom::as.ppp"><code>spatstat.geom::as.ppp</code></a></p>
</section>
<section id="value-13" class="level4">
<h4 class="anchored" data-anchor-id="value-13">Value</h4>
<p>A point pattern representing tree detections for the specified species within the defined area.</p>
</section>
<section id="examples-13" class="level4">
<h4 class="anchored" data-anchor-id="examples-13">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb250"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb250-1"><a href="#cb250-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_ppp_from_area</span>(<span class="at">species_identifier =</span> <span class="st">"SpeciesA"</span>, <span class="at">area_identifier =</span> <span class="st">"Area1"</span>)</span>
<span id="cb250-2"><a href="#cb250-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb250-3"><a href="#cb250-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a point pattern for a specific species in a given area</span></span>
<span id="cb250-4"><a href="#cb250-4" aria-hidden="true" tabindex="-1"></a>pp <span class="ot">&lt;-</span> <span class="fu">lfa_create_ppp_from_area</span>(<span class="at">species_identifier =</span> <span class="st">"SpeciesA"</span>, <span class="at">area_identifier =</span> <span class="st">"Area1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-13" class="level4">
<h4 class="anchored" data-anchor-id="usage-13">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb251"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb251-1"><a href="#cb251-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_ppp_from_area</span>(species_identifier, area_identifier)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_create_stacked_distributions_plot" class="level3" data-number="6.5.15">
<h3 data-number="6.5.15" class="anchored" data-anchor-id="lfa_create_stacked_distributions_plot"><span class="header-section-number">6.5.15</span> <code>lfa_create_stacked_distributions_plot</code></h3>
<p>Create a stacked distribution plot for tree detections, visualizing the distribution of a specified variable on the x-axis, differentiated by another variable.</p>
<section id="arguments-13" class="level4">
<h4 class="anchored" data-anchor-id="arguments-13">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>trees</code></td>
<td>A data frame containing tree detection data.</td>
</tr>
<tr class="even">
<td><code>x_value</code></td>
<td>A character string specifying the column name used for finding the values on the x-axis of the histogram.</td>
</tr>
<tr class="odd">
<td><code>fill_value</code></td>
<td>A character string specifying the column name by which the data are differentiated in the plot.</td>
</tr>
<tr class="even">
<td><code>bin</code></td>
<td>An integer specifying the number of bins for the histogram. Default is 100.</td>
</tr>
<tr class="odd">
<td><code>ylab</code></td>
<td>A character string specifying the y-axis label. Default is “Amount trees.”</td>
</tr>
<tr class="even">
<td><code>xlim</code></td>
<td>A numeric vector of length 2 specifying the x-axis limits. Default is c(0, 100).</td>
</tr>
<tr class="odd">
<td><code>ylim</code></td>
<td>A numeric vector of length 2 specifying the y-axis limits. Default is c(0, 1000).</td>
</tr>
<tr class="even">
<td><code>title</code></td>
<td>The title of the plot.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-14" class="level4">
<h4 class="anchored" data-anchor-id="description-14">Description</h4>
<p>This function generates a stacked distribution plot using the ggplot2 package, providing a visual representation of the distribution of a specified variable ( <code>x_value</code> ) on the x-axis, with differentiation based on another variable ( <code>fill_value</code> ). The data for the plot are derived from the provided <code>trees</code> data frame.</p>
</section>
<section id="keyword-4" class="level4">
<h4 class="anchored" data-anchor-id="keyword-4">Keyword</h4>
<p>data</p>
</section>
<section id="seealso-6" class="level4">
<h4 class="anchored" data-anchor-id="seealso-6">Seealso</h4>
<p><a href="#ggplot2::geomhistogram"><code>ggplot2::geom_histogram</code></a> , <a href="#ggplot2::facetwrap"><code>ggplot2::facet_wrap</code></a> , <a href="#ggplot2::ylab"><code>ggplot2::ylab</code></a> , <a href="#ggplot2::scalefillbrewer"><code>ggplot2::scale_fill_brewer</code></a> , <a href="#ggplot2::coordcartesian"><code>ggplot2::coord_cartesian</code></a></p>
</section>
<section id="value-14" class="level4">
<h4 class="anchored" data-anchor-id="value-14">Value</h4>
<p>A ggplot object representing the stacked distribution plot.</p>
</section>
<section id="examples-14" class="level4">
<h4 class="anchored" data-anchor-id="examples-14">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb252"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb252-1"><a href="#cb252-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a stacked distribution plot for variable "Z," differentiated by "area"</span></span>
<span id="cb252-2"><a href="#cb252-2" aria-hidden="true" tabindex="-1"></a>trees <span class="ot">&lt;-</span> <span class="fu">lfa_get_detections</span>()</span>
<span id="cb252-3"><a href="#cb252-3" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_stacked_distributions_plot</span>(trees, <span class="st">"Z"</span>, <span class="st">"area"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-14" class="level4">
<h4 class="anchored" data-anchor-id="usage-14">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb253"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb253-1"><a href="#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_stacked_distributions_plot</span>(</span>
<span id="cb253-2"><a href="#cb253-2" aria-hidden="true" tabindex="-1"></a>  trees,</span>
<span id="cb253-3"><a href="#cb253-3" aria-hidden="true" tabindex="-1"></a>  x_value,</span>
<span id="cb253-4"><a href="#cb253-4" aria-hidden="true" tabindex="-1"></a>  fill_value,</span>
<span id="cb253-5"><a href="#cb253-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">bin =</span> <span class="dv">100</span>,</span>
<span id="cb253-6"><a href="#cb253-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylab =</span> <span class="st">"Amount trees"</span>,</span>
<span id="cb253-7"><a href="#cb253-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>),</span>
<span id="cb253-8"><a href="#cb253-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1000</span>),</span>
<span id="cb253-9"><a href="#cb253-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">title =</span></span>
<span id="cb253-10"><a href="#cb253-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Histograms of height distributions between species 'beech', 'oak', 'pine' and 'spruce' divided by the different areas of Interest"</span></span>
<span id="cb253-11"><a href="#cb253-11" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_create_stacked_histogram" class="level3" data-number="6.5.16">
<h3 data-number="6.5.16" class="anchored" data-anchor-id="lfa_create_stacked_histogram"><span class="header-section-number">6.5.16</span> <code>lfa_create_stacked_histogram</code></h3>
<p>Create a stacked histogram for tree detections, summing up the values for each species.</p>
<section id="arguments-14" class="level4">
<h4 class="anchored" data-anchor-id="arguments-14">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>trees</code></td>
<td>A data frame containing tree detection data.</td>
</tr>
<tr class="even">
<td><code>x_value</code></td>
<td>A character string specifying the column name used for finding the values on the x-axis of the histogram.</td>
</tr>
<tr class="odd">
<td><code>fill_value</code></td>
<td>A character string specifying the column name by which the data are differentiated in the plot.</td>
</tr>
<tr class="even">
<td><code>bin</code></td>
<td>An integer specifying the number of bins for the histogram. Default is 30.</td>
</tr>
<tr class="odd">
<td><code>ylab</code></td>
<td>A character string specifying the y-axis label. Default is “Frequency.”</td>
</tr>
<tr class="even">
<td><code>xlim</code></td>
<td>A numeric vector of length 2 specifying the x-axis limits. Default is c(0, 100).</td>
</tr>
<tr class="odd">
<td><code>ylim</code></td>
<td>A numeric vector of length 2 specifying the y-axis limits. Default is NULL.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-15" class="level4">
<h4 class="anchored" data-anchor-id="description-15">Description</h4>
<p>This function generates a stacked histogram using the ggplot2 package, summing up the values for each species and visualizing the distribution of a specified variable ( <code>x_value</code> ) on the x-axis, differentiated by another variable ( <code>fill_value</code> ). The data for the plot are derived from the provided <code>trees</code> data frame.</p>
</section>
<section id="keyword-5" class="level4">
<h4 class="anchored" data-anchor-id="keyword-5">Keyword</h4>
<p>data</p>
</section>
<section id="seealso-7" class="level4">
<h4 class="anchored" data-anchor-id="seealso-7">Seealso</h4>
<p><a href="#ggplot2::geomhistogram"><code>ggplot2::geom_histogram</code></a> , <a href="#ggplot2::ylab"><code>ggplot2::ylab</code></a> , <a href="#ggplot2::scalefillbrewer"><code>ggplot2::scale_fill_brewer</code></a> , <a href="#ggplot2::coordcartesian"><code>ggplot2::coord_cartesian</code></a></p>
</section>
<section id="value-15" class="level4">
<h4 class="anchored" data-anchor-id="value-15">Value</h4>
<p>A ggplot object representing the stacked histogram.</p>
</section>
<section id="examples-15" class="level4">
<h4 class="anchored" data-anchor-id="examples-15">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb254"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb254-1"><a href="#cb254-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a stacked histogram for variable "Z," differentiated by "area"</span></span>
<span id="cb254-2"><a href="#cb254-2" aria-hidden="true" tabindex="-1"></a>trees <span class="ot">&lt;-</span> <span class="fu">lfa_get_detections</span>()</span>
<span id="cb254-3"><a href="#cb254-3" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_stacked_histogram</span>(trees, <span class="st">"Z"</span>, <span class="st">"area"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-15" class="level4">
<h4 class="anchored" data-anchor-id="usage-15">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb255"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb255-1"><a href="#cb255-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_stacked_histogram</span>(</span>
<span id="cb255-2"><a href="#cb255-2" aria-hidden="true" tabindex="-1"></a>  trees,</span>
<span id="cb255-3"><a href="#cb255-3" aria-hidden="true" tabindex="-1"></a>  x_value,</span>
<span id="cb255-4"><a href="#cb255-4" aria-hidden="true" tabindex="-1"></a>  fill_value,</span>
<span id="cb255-5"><a href="#cb255-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">bin =</span> <span class="dv">30</span>,</span>
<span id="cb255-6"><a href="#cb255-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylab =</span> <span class="st">"Frequency"</span>,</span>
<span id="cb255-7"><a href="#cb255-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">100</span>),</span>
<span id="cb255-8"><a href="#cb255-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylim =</span> <span class="cn">NULL</span></span>
<span id="cb255-9"><a href="#cb255-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_create_tile_location_objects" class="level3" data-number="6.5.17">
<h3 data-number="6.5.17" class="anchored" data-anchor-id="lfa_create_tile_location_objects"><span class="header-section-number">6.5.17</span> <code>lfa_create_tile_location_objects</code></h3>
<p>Create tile location objects</p>
<section id="author" class="level4">
<h4 class="anchored" data-anchor-id="author">Author</h4>
<p>Jakob Danel</p>
</section>
<section id="description-16" class="level4">
<h4 class="anchored" data-anchor-id="description-16">Description</h4>
<p>This function traverses a directory structure to find LAZ files and creates tile location objects for each file. The function looks into the the <code>data</code> directory of the repository/working directory. It then creates <code>tile_location</code> objects based on the folder structure. The folder structure should not be touched by hand, but created by <code>lfa_init_data_structure()</code> which builds the structure based on a shape file.</p>
</section>
<section id="seealso-8" class="level4">
<h4 class="anchored" data-anchor-id="seealso-8">Seealso</h4>
<p><a href="#tilelocation"><code>tile_location</code></a></p>
</section>
<section id="value-16" class="level4">
<h4 class="anchored" data-anchor-id="value-16">Value</h4>
<p>A vector containing tile location objects.</p>
</section>
<section id="examples-16" class="level4">
<h4 class="anchored" data-anchor-id="examples-16">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb256"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb256-1"><a href="#cb256-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_tile_location_objects</span>()</span>
<span id="cb256-2"><a href="#cb256-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb256-3"><a href="#cb256-3" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_tile_location_objects</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-16" class="level4">
<h4 class="anchored" data-anchor-id="usage-16">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb257"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb257-1"><a href="#cb257-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_create_tile_location_objects</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_detection" class="level3" data-number="6.5.18">
<h3 data-number="6.5.18" class="anchored" data-anchor-id="lfa_detection"><span class="header-section-number">6.5.18</span> <code>lfa_detection</code></h3>
<p>Perform tree detection on a lidar catalog and optionally save the results to a file.</p>
<section id="arguments-15" class="level4">
<h4 class="anchored" data-anchor-id="arguments-15">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>catalog</code></td>
<td>A lidar catalog containing point cloud data. If set to NULL, the function attempts to read the catalog from the specified tile location.</td>
</tr>
<tr class="even">
<td><code>tile_location</code></td>
<td>An object specifying the location of the lidar tile. If catalog is NULL, the function attempts to read the catalog from this tile location.</td>
</tr>
<tr class="odd">
<td><code>write_to_file</code></td>
<td>A logical value indicating whether to save the detected tree information to a file. Default is TRUE.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-17" class="level4">
<h4 class="anchored" data-anchor-id="description-17">Description</h4>
<p>This function utilizes lidar data to detect trees within a specified catalog. The detected tree information can be optionally saved to a file in the GeoPackage format. The function uses parallel processing to enhance efficiency.</p>
</section>
<section id="value-17" class="level4">
<h4 class="anchored" data-anchor-id="value-17">Value</h4>
<p>A sf style data frame containing information about the detected trees.</p>
</section>
<section id="examples-17" class="level4">
<h4 class="anchored" data-anchor-id="examples-17">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb258"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb258-1"><a href="#cb258-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform tree detection on a catalog and save the results to a file</span></span>
<span id="cb258-2"><a href="#cb258-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_detection</span>(<span class="at">catalog =</span> my_catalog, <span class="at">tile_location =</span> my_tile_location, <span class="at">write_to_file =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-17" class="level4">
<h4 class="anchored" data-anchor-id="usage-17">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb259"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb259-1"><a href="#cb259-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_detection</span>(catalog, tile_location, <span class="at">write_to_file =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_download_areas" class="level3" data-number="6.5.19">
<h3 data-number="6.5.19" class="anchored" data-anchor-id="lfa_download_areas"><span class="header-section-number">6.5.19</span> <code>lfa_download_areas</code></h3>
<p>Download areas based on spatial features</p>
<section id="arguments-16" class="level4">
<h4 class="anchored" data-anchor-id="arguments-16">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>sf_areas</code></td>
<td>Spatial features representing areas to be downloaded. It must include columns like “species” “name” See details for more information.</td>
</tr>
</tbody>
</table>
</section>
<section id="author-1" class="level4">
<h4 class="anchored" data-anchor-id="author-1">Author</h4>
<p>Jakob Danel</p>
</section>
<section id="description-18" class="level4">
<h4 class="anchored" data-anchor-id="description-18">Description</h4>
<p>This function initiates the data structure and downloads areas based on spatial features.</p>
</section>
<section id="details-5" class="level4">
<h4 class="anchored" data-anchor-id="details-5">Details</h4>
<p>The input data frame, <code>sf_areas</code> , must have the following columns:</p>
<ul>
<li><p>“species”: The species associated with the area.</p></li>
<li><p>“name”: The name of the area.</p></li>
</ul>
<p>The function uses the <code>lfa_init_data_structure</code> function to set up the data structure and then iterates through the rows of <code>sf_areas</code> to download each specified area.</p>
</section>
<section id="value-18" class="level4">
<h4 class="anchored" data-anchor-id="value-18">Value</h4>
<p>None</p>
</section>
<section id="examples-18" class="level4">
<h4 class="anchored" data-anchor-id="examples-18">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb260"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb260-1"><a href="#cb260-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_download_areas</span>(sf_areas)</span>
<span id="cb260-2"><a href="#cb260-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-3"><a href="#cb260-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-4"><a href="#cb260-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Example spatial features data frame</span></span>
<span id="cb260-5"><a href="#cb260-5" aria-hidden="true" tabindex="-1"></a>sf_areas <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb260-6"><a href="#cb260-6" aria-hidden="true" tabindex="-1"></a><span class="at">species =</span> <span class="fu">c</span>(<span class="st">"SpeciesA"</span>, <span class="st">"SpeciesB"</span>),</span>
<span id="cb260-7"><a href="#cb260-7" aria-hidden="true" tabindex="-1"></a><span class="at">name =</span> <span class="fu">c</span>(<span class="st">"Area1"</span>, <span class="st">"Area2"</span>),</span>
<span id="cb260-8"><a href="#cb260-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Must include also other attributes specialized to sf objects</span></span>
<span id="cb260-9"><a href="#cb260-9" aria-hidden="true" tabindex="-1"></a><span class="co"># such as geometry, for processing of the download</span></span>
<span id="cb260-10"><a href="#cb260-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb260-11"><a href="#cb260-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-12"><a href="#cb260-12" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_download_areas</span>(sf_areas)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-18" class="level4">
<h4 class="anchored" data-anchor-id="usage-18">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb261"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb261-1"><a href="#cb261-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_download_areas</span>(sf_areas)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_download" class="level3" data-number="6.5.20">
<h3 data-number="6.5.20" class="anchored" data-anchor-id="lfa_download"><span class="header-section-number">6.5.20</span> <code>lfa_download</code></h3>
<p>Download an las file from the state NRW from a specific location</p>
<section id="arguments-17" class="level4">
<h4 class="anchored" data-anchor-id="arguments-17">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>species</code></td>
<td>The species of the tree which is observed at this location</td>
</tr>
<tr class="even">
<td><code>name</code></td>
<td>The name of the area that is observed</td>
</tr>
<tr class="odd">
<td><code>location</code></td>
<td>An sf object, which holds the location information for the area where the tile should be downloaded from.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-19" class="level4">
<h4 class="anchored" data-anchor-id="description-19">Description</h4>
<p>It will download the file and save it to data/ list(list(“html”), list(list(“<species>”))) / list(list(“html”), list(list(“<name>”))) with the name of the tile</name></species></p>
</section>
<section id="value-19" class="level4">
<h4 class="anchored" data-anchor-id="value-19">Value</h4>
<p>The LASCatalog object of the downloaded file</p>
</section>
<section id="usage-19" class="level4">
<h4 class="anchored" data-anchor-id="usage-19">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb262"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb262-1"><a href="#cb262-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_download</span>(species, name, location)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_find_n_nearest_trees" class="level3" data-number="6.5.21">
<h3 data-number="6.5.21" class="anchored" data-anchor-id="lfa_find_n_nearest_trees"><span class="header-section-number">6.5.21</span> <code>lfa_find_n_nearest_trees</code></h3>
<p>Find n Nearest Trees</p>
<section id="arguments-18" class="level4">
<h4 class="anchored" data-anchor-id="arguments-18">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>trees</code></td>
<td>A sf object containing tree coordinates.</td>
</tr>
<tr class="even">
<td><code>n</code></td>
<td>The number of nearest trees to find for each tree (default is 100).</td>
</tr>
</tbody>
</table>
</section>
<section id="description-20" class="level4">
<h4 class="anchored" data-anchor-id="description-20">Description</h4>
<p>This function calculates the distances to the n nearest trees for each tree in the input dataset.</p>
</section>
<section id="value-20" class="level4">
<h4 class="anchored" data-anchor-id="value-20">Value</h4>
<p>A data frame with additional columns representing the distances to the n nearest trees.</p>
</section>
<section id="examples-19" class="level4">
<h4 class="anchored" data-anchor-id="examples-19">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb263"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb263-1"><a href="#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load tree data using lfa_get_detections() (not provided)</span></span>
<span id="cb263-2"><a href="#cb263-2" aria-hidden="true" tabindex="-1"></a>tree_data <span class="ot">&lt;-</span> <span class="fu">lfa_get_detections</span>()</span>
<span id="cb263-3"><a href="#cb263-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb263-4"><a href="#cb263-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter tree data for a specific species and area</span></span>
<span id="cb263-5"><a href="#cb263-5" aria-hidden="true" tabindex="-1"></a>tree_data <span class="ot">=</span> tree_data[tree_data<span class="sc">$</span>specie <span class="sc">==</span> <span class="st">"pine"</span> <span class="sc">&amp;</span> tree_data<span class="sc">$</span>area <span class="sc">==</span> <span class="st">"greffen"</span>, ]</span>
<span id="cb263-6"><a href="#cb263-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb263-7"><a href="#cb263-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Find the 100 nearest trees for each tree in the filtered dataset</span></span>
<span id="cb263-8"><a href="#cb263-8" aria-hidden="true" tabindex="-1"></a>tree_data <span class="ot">&lt;-</span> <span class="fu">lfa_find_n_nearest_trees</span>(tree_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-20" class="level4">
<h4 class="anchored" data-anchor-id="usage-20">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb264"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb264-1"><a href="#cb264-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_find_n_nearest_trees</span>(trees, <span class="at">n =</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_generate_result_table_tests" class="level3" data-number="6.5.22">
<h3 data-number="6.5.22" class="anchored" data-anchor-id="lfa_generate_result_table_tests"><span class="header-section-number">6.5.22</span> <code>lfa_generate_result_table_tests</code></h3>
<p>Generate Result Table for Tests</p>
<section id="arguments-19" class="level4">
<h4 class="anchored" data-anchor-id="arguments-19">Arguments</h4>
<table class="table">
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>table</code></td>
<td>A data frame representing the result table.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-21" class="level4">
<h4 class="anchored" data-anchor-id="description-21">Description</h4>
<p>This function generates a result table for tests using the knitr::kable function.</p>
</section>
<section id="details-6" class="level4">
<h4 class="anchored" data-anchor-id="details-6">Details</h4>
<p>This function uses the knitr::kable function to create a formatted table, making it suitable for HTML output. The input table is expected to be a data frame with test results, and the resulting table will have capitalized row and column names with lines between columns and rows.</p>
</section>
<section id="value-21" class="level4">
<h4 class="anchored" data-anchor-id="value-21">Value</h4>
<p>A formatted table suitable for HTML output with lines between columns and rows.</p>
</section>
<section id="examples-20" class="level4">
<h4 class="anchored" data-anchor-id="examples-20">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb265"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb265-1"><a href="#cb265-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate a result table for tests</span></span>
<span id="cb265-2"><a href="#cb265-2" aria-hidden="true" tabindex="-1"></a>result_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb265-3"><a href="#cb265-3" aria-hidden="true" tabindex="-1"></a><span class="at">Test1 =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.10</span>, <span class="fl">0.03</span>),</span>
<span id="cb265-4"><a href="#cb265-4" aria-hidden="true" tabindex="-1"></a><span class="at">Test2 =</span> <span class="fu">c</span>(<span class="fl">0.02</span>, <span class="fl">0.08</span>, <span class="fl">0.01</span>),</span>
<span id="cb265-5"><a href="#cb265-5" aria-hidden="true" tabindex="-1"></a><span class="at">Test3 =</span> <span class="fu">c</span>(<span class="fl">0.08</span>, <span class="fl">0.12</span>, <span class="fl">0.05</span>)</span>
<span id="cb265-6"><a href="#cb265-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb265-7"><a href="#cb265-7" aria-hidden="true" tabindex="-1"></a>formatted_table <span class="ot">&lt;-</span> <span class="fu">lfa_generate_result_table_tests</span>(result_table)</span>
<span id="cb265-8"><a href="#cb265-8" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(formatted_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-21" class="level4">
<h4 class="anchored" data-anchor-id="usage-21">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb266"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb266-1"><a href="#cb266-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_generate_result_table_tests</span>(table, <span class="at">caption =</span> <span class="st">"Table Caption"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_get_all_areas" class="level3" data-number="6.5.23">
<h3 data-number="6.5.23" class="anchored" data-anchor-id="lfa_get_all_areas"><span class="header-section-number">6.5.23</span> <code>lfa_get_all_areas</code></h3>
<p>Retrieve a data frame containing all species and corresponding areas.</p>
<section id="description-22" class="level4">
<h4 class="anchored" data-anchor-id="description-22">Description</h4>
<p>This function scans the “data” directory within the current working directory to obtain a list of species. It then iterates through each species to retrieve the list of areas associated with that species. The resulting data frame contains two columns: “specie” representing the species and “area” representing the corresponding area.</p>
</section>
<section id="keyword-6" class="level4">
<h4 class="anchored" data-anchor-id="keyword-6">Keyword</h4>
<p>data</p>
</section>
<section id="seealso-9" class="level4">
<h4 class="anchored" data-anchor-id="seealso-9">Seealso</h4>
<p><a href="#list.dirs"><code>list.dirs</code></a></p>
</section>
<section id="value-22" class="level4">
<h4 class="anchored" data-anchor-id="value-22">Value</h4>
<p>A data frame with columns “specie” and “area” containing information about all species and their associated areas.</p>
</section>
<section id="examples-21" class="level4">
<h4 class="anchored" data-anchor-id="examples-21">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb267"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb267-1"><a href="#cb267-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Retrieve a data frame with information about all species and areas</span></span>
<span id="cb267-2"><a href="#cb267-2" aria-hidden="true" tabindex="-1"></a>all_areas_df <span class="ot">&lt;-</span> <span class="fu">lfa_get_all_areas</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-22" class="level4">
<h4 class="anchored" data-anchor-id="usage-22">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb268"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb268-1"><a href="#cb268-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_get_all_areas</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_get_detection_area" class="level3" data-number="6.5.24">
<h3 data-number="6.5.24" class="anchored" data-anchor-id="lfa_get_detection_area"><span class="header-section-number">6.5.24</span> <code>lfa_get_detection_area</code></h3>
<p>Get Detection for an area</p>
<section id="arguments-20" class="level4">
<h4 class="anchored" data-anchor-id="arguments-20">Arguments</h4>
<table class="table">
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>species</code></td>
<td>A character string specifying the target species.</td>
</tr>
<tr class="even">
<td><code>name</code></td>
<td>A character string specifying the name of the tile.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-23" class="level4">
<h4 class="anchored" data-anchor-id="description-23">Description</h4>
<p>Retrieves the tree detection information for a specified species and tile.</p>
</section>
<section id="details-7" class="level4">
<h4 class="anchored" data-anchor-id="details-7">Details</h4>
<p>This function reads tree detection data from geopackage files within the specified tile location for a given species. It then combines the data into a single SF data frame and returns it. The function assumes that the tree detection files follow a naming convention with the pattern “_detection.gpkg”.</p>
</section>
<section id="keyword-7" class="level4">
<h4 class="anchored" data-anchor-id="keyword-7">Keyword</h4>
<p>spatial</p>
</section>
<section id="references-2" class="level4">
<h4 class="anchored" data-anchor-id="references-2">References</h4>
<p>This function is part of the LiDAR Forest Analysis (LFA) package.</p>
</section>
<section id="seealso-10" class="level4">
<h4 class="anchored" data-anchor-id="seealso-10">Seealso</h4>
<p><a href="#gettiledir"><code>get_tile_dir</code></a></p>
</section>
<section id="value-23" class="level4">
<h4 class="anchored" data-anchor-id="value-23">Value</h4>
<p>A Simple Features (SF) data frame containing tree detection information for the specified species and tile.</p>
</section>
<section id="examples-22" class="level4">
<h4 class="anchored" data-anchor-id="examples-22">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb269"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb269-1"><a href="#cb269-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Retrieve tree detection data for species "example_species" in tile "example_tile"</span></span>
<span id="cb269-2"><a href="#cb269-2" aria-hidden="true" tabindex="-1"></a>trees_data <span class="ot">&lt;-</span> <span class="fu">lfa_get_detection_tile_location</span>(<span class="st">"example_species"</span>, <span class="st">"example_tile"</span>)</span>
<span id="cb269-3"><a href="#cb269-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-4"><a href="#cb269-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage:</span></span>
<span id="cb269-5"><a href="#cb269-5" aria-hidden="true" tabindex="-1"></a>trees_data <span class="ot">&lt;-</span> <span class="fu">lfa_get_detection_tile_location</span>(<span class="st">"example_species"</span>, <span class="st">"example_tile"</span>)</span>
<span id="cb269-6"><a href="#cb269-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-7"><a href="#cb269-7" aria-hidden="true" tabindex="-1"></a><span class="co"># No trees found scenario:</span></span>
<span id="cb269-8"><a href="#cb269-8" aria-hidden="true" tabindex="-1"></a>empty_data <span class="ot">&lt;-</span> <span class="fu">lfa_get_detection_tile_location</span>(<span class="st">"nonexistent_species"</span>, <span class="st">"nonexistent_tile"</span>)</span>
<span id="cb269-9"><a href="#cb269-9" aria-hidden="true" tabindex="-1"></a><span class="co"># The result will be an empty data frame if no trees are found for the specified species and tile.</span></span>
<span id="cb269-10"><a href="#cb269-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-11"><a href="#cb269-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Error handling:</span></span>
<span id="cb269-12"><a href="#cb269-12" aria-hidden="true" tabindex="-1"></a><span class="co"># In case of invalid inputs, the function may throw errors. Ensure correct species and tile names are provided.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-23" class="level4">
<h4 class="anchored" data-anchor-id="usage-23">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb270"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb270-1"><a href="#cb270-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_get_detection_area</span>(species, name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_get_detections_species" class="level3" data-number="6.5.25">
<h3 data-number="6.5.25" class="anchored" data-anchor-id="lfa_get_detections_species"><span class="header-section-number">6.5.25</span> <code>lfa_get_detections_species</code></h3>
<p>Retrieve detections for a specific species.</p>
<section id="arguments-21" class="level4">
<h4 class="anchored" data-anchor-id="arguments-21">Arguments</h4>
<table class="table">
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>species</code></td>
<td>A character string specifying the target species.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-24" class="level4">
<h4 class="anchored" data-anchor-id="description-24">Description</h4>
<p>This function retrieves detection data for a given species from multiple areas.</p>
</section>
<section id="details-8" class="level4">
<h4 class="anchored" data-anchor-id="details-8">Details</h4>
<p>The function looks for detection data in the “data” directory for the specified species. It then iterates through each subdirectory (representing different areas) and consolidates the detection data into a single data frame.</p>
</section>
<section id="value-24" class="level4">
<h4 class="anchored" data-anchor-id="value-24">Value</h4>
<p>A data frame containing detection information for the specified species in different areas.</p>
</section>
<section id="examples-23" class="level4">
<h4 class="anchored" data-anchor-id="examples-23">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb271"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb271-1"><a href="#cb271-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage:</span></span>
<span id="cb271-2"><a href="#cb271-2" aria-hidden="true" tabindex="-1"></a>detections_data <span class="ot">&lt;-</span> <span class="fu">lfa_get_detections_species</span>(<span class="st">"example_species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-24" class="level4">
<h4 class="anchored" data-anchor-id="usage-24">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb272"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb272-1"><a href="#cb272-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_get_detections_species</span>(species)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_get_detections" class="level3" data-number="6.5.26">
<h3 data-number="6.5.26" class="anchored" data-anchor-id="lfa_get_detections"><span class="header-section-number">6.5.26</span> <code>lfa_get_detections</code></h3>
<p>Retrieve aggregated detection data for multiple species.</p>
<section id="concept-1" class="level4">
<h4 class="anchored" data-anchor-id="concept-1">Concept</h4>
<p>data retrieval functions</p>
</section>
<section id="description-25" class="level4">
<h4 class="anchored" data-anchor-id="description-25">Description</h4>
<p>This function obtains aggregated detection data for multiple species by iterating through the list of species obtained from <a href="#lfagetspecies"><code>lfa_get_species</code></a> . For each species, it calls <a href="#lfagetdetectionsspecies"><code>lfa_get_detections_species</code></a> to retrieve the corresponding detection data and aggregates the results into a single data frame. The resulting data frame includes columns for the species, tree detection data, and the area in which the detections occurred.</p>
</section>
<section id="keyword-8" class="level4">
<h4 class="anchored" data-anchor-id="keyword-8">Keyword</h4>
<p>aggregation</p>
</section>
<section id="seealso-11" class="level4">
<h4 class="anchored" data-anchor-id="seealso-11">Seealso</h4>
<p><a href="#lfagetspecies"><code>lfa_get_species</code></a> , <a href="#lfagetdetectionsspecies"><code>lfa_get_detections_species</code></a></p>
<p>Other data retrieval functions: <a href="#lfagetspecies"><code>lfa_get_species</code></a></p>
</section>
<section id="value-25" class="level4">
<h4 class="anchored" data-anchor-id="value-25">Value</h4>
<p>A data frame containing aggregated detection data for multiple species.</p>
</section>
<section id="examples-24" class="level4">
<h4 class="anchored" data-anchor-id="examples-24">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb273"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb273-1"><a href="#cb273-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_get_detections</span>()</span>
<span id="cb273-2"><a href="#cb273-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb273-3"><a href="#cb273-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Retrieve aggregated detection data for multiple species</span></span>
<span id="cb273-4"><a href="#cb273-4" aria-hidden="true" tabindex="-1"></a>detections_data <span class="ot">&lt;-</span> <span class="fu">lfa_get_detections</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-25" class="level4">
<h4 class="anchored" data-anchor-id="usage-25">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb274"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb274-1"><a href="#cb274-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_get_detections</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_get_flag_path" class="level3" data-number="6.5.27">
<h3 data-number="6.5.27" class="anchored" data-anchor-id="lfa_get_flag_path"><span class="header-section-number">6.5.27</span> <code>lfa_get_flag_path</code></h3>
<p>Get the path to a flag file indicating the completion of a specific process.</p>
<section id="arguments-22" class="level4">
<h4 class="anchored" data-anchor-id="arguments-22">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>flag_name</code></td>
<td>A character string specifying the name of the flag file. It should be a descriptive and unique identifier for the process being flagged.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-26" class="level4">
<h4 class="anchored" data-anchor-id="description-26">Description</h4>
<p>This function constructs and returns the path to a hidden flag file, which serves as an indicator that a particular processing step has been completed. The flag file is created in a designated location within the working directory.</p>
</section>
<section id="value-26" class="level4">
<h4 class="anchored" data-anchor-id="value-26">Value</h4>
<p>A character string representing the absolute path to the hidden flag file.</p>
</section>
<section id="examples-25" class="level4">
<h4 class="anchored" data-anchor-id="examples-25">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb275"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb275-1"><a href="#cb275-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the flag path for a process named "data_processing"</span></span>
<span id="cb275-2"><a href="#cb275-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_get_flag_path</span>(<span class="st">"data_processing"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-26" class="level4">
<h4 class="anchored" data-anchor-id="usage-26">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb276"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb276-1"><a href="#cb276-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_get_flag_path</span>(flag_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_get_neighbor_paths" class="level3" data-number="6.5.28">
<h3 data-number="6.5.28" class="anchored" data-anchor-id="lfa_get_neighbor_paths"><span class="header-section-number">6.5.28</span> <code>lfa_get_neighbor_paths</code></h3>
<p>Get Paths to Neighbor GeoPackage Files</p>
<section id="description-27" class="level4">
<h4 class="anchored" data-anchor-id="description-27">Description</h4>
<p>This function retrieves the file paths to GeoPackage files containing neighbor information for each detection area. The GeoPackage files are assumed to be named “neighbours.gpkg” and organized in a directory structure under the “data” folder.</p>
</section>
<section id="value-27" class="level4">
<h4 class="anchored" data-anchor-id="value-27">Value</h4>
<p>A character vector containing file paths to GeoPackage files for each detection area’s neighbors.</p>
</section>
<section id="examples-26" class="level4">
<h4 class="anchored" data-anchor-id="examples-26">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb277"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb277-1"><a href="#cb277-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get paths to neighbor GeoPackage files for all areas</span></span>
<span id="cb277-2"><a href="#cb277-2" aria-hidden="true" tabindex="-1"></a>paths <span class="ot">&lt;-</span> <span class="fu">lfa_get_neighbor_paths</span>()</span>
<span id="cb277-3"><a href="#cb277-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb277-4"><a href="#cb277-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the obtained file paths</span></span>
<span id="cb277-5"><a href="#cb277-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(paths)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-27" class="level4">
<h4 class="anchored" data-anchor-id="usage-27">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb278"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb278-1"><a href="#cb278-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_get_neighbor_paths</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_get_species" class="level3" data-number="6.5.29">
<h3 data-number="6.5.29" class="anchored" data-anchor-id="lfa_get_species"><span class="header-section-number">6.5.29</span> <code>lfa_get_species</code></h3>
<p>Get a list of species from the data directory.</p>
<section id="concept-2" class="level4">
<h4 class="anchored" data-anchor-id="concept-2">Concept</h4>
<p>data retrieval functions</p>
</section>
<section id="description-28" class="level4">
<h4 class="anchored" data-anchor-id="description-28">Description</h4>
<p>This function retrieves a list of species by scanning the “data” directory located in the current working directory.</p>
</section>
<section id="keyword-9" class="level4">
<h4 class="anchored" data-anchor-id="keyword-9">Keyword</h4>
<p>data</p>
</section>
<section id="references-3" class="level4">
<h4 class="anchored" data-anchor-id="references-3">References</h4>
<p>This function relies on the <a href="#list.dirs"><code>list.dirs</code></a> function for directory listing.</p>
</section>
<section id="seealso-12" class="level4">
<h4 class="anchored" data-anchor-id="seealso-12">Seealso</h4>
<p><a href="#list.dirs"><code>list.dirs</code></a></p>
<p>Other data retrieval functions: <a href="#lfagetdetections"><code>lfa_get_detections</code></a></p>
</section>
<section id="value-28" class="level4">
<h4 class="anchored" data-anchor-id="value-28">Value</h4>
<p>A character vector containing the names of species found in the “data” directory.</p>
</section>
<section id="examples-27" class="level4">
<h4 class="anchored" data-anchor-id="examples-27">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb279"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb279-1"><a href="#cb279-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Retrieve the list of species</span></span>
<span id="cb279-2"><a href="#cb279-2" aria-hidden="true" tabindex="-1"></a>species_list <span class="ot">&lt;-</span> <span class="fu">lfa_get_species</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-28" class="level4">
<h4 class="anchored" data-anchor-id="usage-28">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb280"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb280-1"><a href="#cb280-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_get_species</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_ground_correction" class="level3" data-number="6.5.30">
<h3 data-number="6.5.30" class="anchored" data-anchor-id="lfa_ground_correction"><span class="header-section-number">6.5.30</span> <code>lfa_ground_correction</code></h3>
<p>Correct the point clouds for correct ground imagery</p>
<section id="arguments-23" class="level4">
<h4 class="anchored" data-anchor-id="arguments-23">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>ctg</code></td>
<td>An LASCatalog object. If not null, it will perform the actions on this object, if NULL inferring the catalog from the tile_location</td>
</tr>
<tr class="even">
<td><code>tile_location</code></td>
<td>A tile_location type object holding the information about the location of the cataog. This is used to save the catalog after processing too.</td>
</tr>
</tbody>
</table>
</section>
<section id="author-2" class="level4">
<h4 class="anchored" data-anchor-id="author-2">Author</h4>
<p>Jakob Danel</p>
</section>
<section id="description-29" class="level4">
<h4 class="anchored" data-anchor-id="description-29">Description</h4>
<p>This function is needed to correct the Z value of the point cloud, relative to the real ground height. After using this function to your catalog, the Z values can be seen as the real elevation about the ground. At the moment the function uses the <code>tin()</code> function from the <code>lidr</code> package. NOTE : The operation is inplace and can not be reverted, the old values of the point cloud will be deleted!</p>
</section>
<section id="value-29" class="level4">
<h4 class="anchored" data-anchor-id="value-29">Value</h4>
<p>A catalog with the corrected z values. The catalog is always stored at tile_location and holding only the transformed values.</p>
</section>
<section id="usage-29" class="level4">
<h4 class="anchored" data-anchor-id="usage-29">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb281"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb281-1"><a href="#cb281-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_ground_correction</span>(ctg, tile_location)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_init_data_structure" class="level3" data-number="6.5.31">
<h3 data-number="6.5.31" class="anchored" data-anchor-id="lfa_init_data_structure"><span class="header-section-number">6.5.31</span> <code>lfa_init_data_structure</code></h3>
<p>Initialize data structure for species and areas</p>
<section id="arguments-24" class="level4">
<h4 class="anchored" data-anchor-id="arguments-24">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>sf_species</code></td>
<td>A data frame with information about species and associated areas.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-30" class="level4">
<h4 class="anchored" data-anchor-id="description-30">Description</h4>
<p>This function initializes the data structure for storing species and associated areas.</p>
</section>
<section id="details-9" class="level4">
<h4 class="anchored" data-anchor-id="details-9">Details</h4>
<p>The input data frame, <code>sf_species</code> , should have at least the following columns:</p>
<ul>
<li><p>“species”: The names of the species for which the data structure needs to be initialized.</p></li>
<li><p>“name”: The names of the associated areas.</p></li>
</ul>
<p>The function creates directories based on the species and area information provided in the <code>sf_species</code> data frame. It checks whether the directories already exist and creates them if they don’t.</p>
</section>
<section id="value-30" class="level4">
<h4 class="anchored" data-anchor-id="value-30">Value</h4>
<p>None</p>
</section>
<section id="examples-28" class="level4">
<h4 class="anchored" data-anchor-id="examples-28">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb282"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb282-1"><a href="#cb282-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example species data frame</span></span>
<span id="cb282-2"><a href="#cb282-2" aria-hidden="true" tabindex="-1"></a>sf_species <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb282-3"><a href="#cb282-3" aria-hidden="true" tabindex="-1"></a><span class="at">species =</span> <span class="fu">c</span>(<span class="st">"SpeciesA"</span>, <span class="st">"SpeciesB"</span>),</span>
<span id="cb282-4"><a href="#cb282-4" aria-hidden="true" tabindex="-1"></a><span class="at">name =</span> <span class="fu">c</span>(<span class="st">"Area1"</span>, <span class="st">"Area2"</span>),</span>
<span id="cb282-5"><a href="#cb282-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Other necessary columns</span></span>
<span id="cb282-6"><a href="#cb282-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb282-7"><a href="#cb282-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb282-8"><a href="#cb282-8" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_init_data_structure</span>(sf_species)</span>
<span id="cb282-9"><a href="#cb282-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb282-10"><a href="#cb282-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Example species data frame</span></span>
<span id="cb282-11"><a href="#cb282-11" aria-hidden="true" tabindex="-1"></a>sf_species <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb282-12"><a href="#cb282-12" aria-hidden="true" tabindex="-1"></a><span class="at">species =</span> <span class="fu">c</span>(<span class="st">"SpeciesA"</span>, <span class="st">"SpeciesB"</span>),</span>
<span id="cb282-13"><a href="#cb282-13" aria-hidden="true" tabindex="-1"></a><span class="at">name =</span> <span class="fu">c</span>(<span class="st">"Area1"</span>, <span class="st">"Area2"</span>),</span>
<span id="cb282-14"><a href="#cb282-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Other necessary columns</span></span>
<span id="cb282-15"><a href="#cb282-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb282-16"><a href="#cb282-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb282-17"><a href="#cb282-17" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_init_data_structure</span>(sf_species)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-30" class="level4">
<h4 class="anchored" data-anchor-id="usage-30">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb283"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb283-1"><a href="#cb283-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_init_data_structure</span>(sf_species)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_init" class="level3" data-number="6.5.32">
<h3 data-number="6.5.32" class="anchored" data-anchor-id="lfa_init"><span class="header-section-number">6.5.32</span> <code>lfa_init</code></h3>
<p>Initialize LFA (LiDAR forest analysis) data processing</p>
<section id="arguments-25" class="level4">
<h4 class="anchored" data-anchor-id="arguments-25">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>sf_file</code></td>
<td>A character string specifying the path to the shapefile containing spatial features of research areas.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-31" class="level4">
<h4 class="anchored" data-anchor-id="description-31">Description</h4>
<p>This function initializes the LFA data processing by reading a shapefile containing spatial features of research areas, downloading the specified areas, and creating tile location objects for each area.</p>
</section>
<section id="details-10" class="level4">
<h4 class="anchored" data-anchor-id="details-10">Details</h4>
<p>This function reads a shapefile ( <code>sf_file</code> ) using the <code>sf</code> package, which should contain information about research areas. It then calls the <code>lfa_download_areas</code> function to download the specified areas and <code>lfa_create_tile_location_objects</code> to create tile location objects based on Lidar data files in those areas. The shapefile MUST follow the following requirements:</p>
<ul>
<li><p>Each geometry must be a single object of type polygon</p></li>
<li><p>Each entry must have the following attributes:</p></li>
<li><p>species: A string describing the tree species of the area.</p></li>
<li><p>name: A string describing the location of the area.</p></li>
</ul>
</section>
<section id="value-31" class="level4">
<h4 class="anchored" data-anchor-id="value-31">Value</h4>
<p>A vector containing tile location objects.</p>
</section>
<section id="examples-29" class="level4">
<h4 class="anchored" data-anchor-id="examples-29">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb284"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb284-1"><a href="#cb284-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize LFA processing with the default shapefile</span></span>
<span id="cb284-2"><a href="#cb284-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_init</span>()</span>
<span id="cb284-3"><a href="#cb284-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb284-4"><a href="#cb284-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize LFA processing with a custom shapefile</span></span>
<span id="cb284-5"><a href="#cb284-5" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_init</span>(<span class="st">"custom_areas.shp"</span>)</span>
<span id="cb284-6"><a href="#cb284-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb284-7"><a href="#cb284-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage with the default shapefile</span></span>
<span id="cb284-8"><a href="#cb284-8" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_init</span>()</span>
<span id="cb284-9"><a href="#cb284-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb284-10"><a href="#cb284-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage with a custom shapefile</span></span>
<span id="cb284-11"><a href="#cb284-11" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_init</span>(<span class="st">"custom_areas.shp"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-31" class="level4">
<h4 class="anchored" data-anchor-id="usage-31">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb285"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb285-1"><a href="#cb285-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_init</span>(<span class="at">sf_file =</span> <span class="st">"research_areas.shp"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_intersect_areas" class="level3" data-number="6.5.33">
<h3 data-number="6.5.33" class="anchored" data-anchor-id="lfa_intersect_areas"><span class="header-section-number">6.5.33</span> <code>lfa_intersect_areas</code></h3>
<p>Intersect Lidar Catalog with Spatial Features</p>
<section id="arguments-26" class="level4">
<h4 class="anchored" data-anchor-id="arguments-26">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>ctg</code></td>
<td>A LAScatalog object representing the Lidar data to be processed.</td>
</tr>
<tr class="even">
<td><code>tile_location</code></td>
<td>A tile location object representing the specific area of interest.</td>
</tr>
<tr class="odd">
<td><code>areas_sf</code></td>
<td>Spatial features defining areas.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-32" class="level4">
<h4 class="anchored" data-anchor-id="description-32">Description</h4>
<p>This function intersects a Lidar catalog with a specific area defined by spatial features.</p>
</section>
<section id="details-11" class="level4">
<h4 class="anchored" data-anchor-id="details-11">Details</h4>
<p>The function intersects the Lidar catalog specified by <code>ctg</code> with a specific area defined by the <code>tile_location</code> object and <code>areas_sf</code> . It removes points outside the specified area and returns a modified LAScatalog object.</p>
<p>The specified area is identified based on the <code>species</code> and <code>name</code> attributes in the <code>tile_location</code> object. If a matching area is not found in <code>areas_sf</code> , the function stops with an error.</p>
<p>The function then transforms the spatial reference of the identified area to match that of the Lidar catalog using <code>sf::st_transform</code> .</p>
<p>The processing is applied to each chunk in the catalog using the <code>identify_area</code> function, which merges spatial information and filters out points that are not classified as inside the identified area. After processing, the function writes the modified LAS files back to the original file locations, removing points outside the specified area.</p>
<p>If an error occurs during the processing of a chunk, a warning is issued, and the function continues processing the next chunks. If no points are found after filtering, a warning is issued, and NULL is returned.</p>
</section>
<section id="seealso-13" class="level4">
<h4 class="anchored" data-anchor-id="seealso-13">Seealso</h4>
<p>Other functions in the Lidar forest analysis (LFA) package.</p>
</section>
<section id="value-32" class="level4">
<h4 class="anchored" data-anchor-id="value-32">Value</h4>
<p>A modified LAScatalog object with points outside the specified area removed.</p>
</section>
<section id="examples-30" class="level4">
<h4 class="anchored" data-anchor-id="examples-30">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb286"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb286-1"><a href="#cb286-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage</span></span>
<span id="cb286-2"><a href="#cb286-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_intersect_areas</span>(ctg, tile_location, areas_sf)</span>
<span id="cb286-3"><a href="#cb286-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-4"><a href="#cb286-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage</span></span>
<span id="cb286-5"><a href="#cb286-5" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_intersect_areas</span>(ctg, tile_location, areas_sf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-32" class="level4">
<h4 class="anchored" data-anchor-id="usage-32">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb287"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb287-1"><a href="#cb287-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_intersect_areas</span>(ctg, tile_location, areas_sf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_jsd_from_vec" class="level3" data-number="6.5.34">
<h3 data-number="6.5.34" class="anchored" data-anchor-id="lfa_jsd_from_vec"><span class="header-section-number">6.5.34</span> <code>lfa_jsd_from_vec</code></h3>
<p>Compute Jensen-Shannon Divergence from Vectors</p>
<section id="arguments-27" class="level4">
<h4 class="anchored" data-anchor-id="arguments-27">Arguments</h4>
<table class="table">
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>x</code></td>
<td>A numeric vector.</td>
</tr>
<tr class="even">
<td><code>y</code></td>
<td>A numeric vector.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-33" class="level4">
<h4 class="anchored" data-anchor-id="description-33">Description</h4>
<p>This function calculates the Jensen-Shannon Divergence (JSD) between two vectors.</p>
</section>
<section id="value-33" class="level4">
<h4 class="anchored" data-anchor-id="value-33">Value</h4>
<p>Jensen-Shannon Divergence between the density distributions of x and y.</p>
</section>
<section id="examples-31" class="level4">
<h4 class="anchored" data-anchor-id="examples-31">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb288"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb288-1"><a href="#cb288-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>)</span>
<span id="cb288-2"><a href="#cb288-2" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>, <span class="at">mean =</span> <span class="dv">2</span>)</span>
<span id="cb288-3"><a href="#cb288-3" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_jsd_from_vec</span>(x, y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-33" class="level4">
<h4 class="anchored" data-anchor-id="usage-33">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb289"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb289-1"><a href="#cb289-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_jsd_from_vec</span>(x, y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_jsd" class="level3" data-number="6.5.35">
<h3 data-number="6.5.35" class="anchored" data-anchor-id="lfa_jsd"><span class="header-section-number">6.5.35</span> <code>lfa_jsd</code></h3>
<p>Jensen-Shannon Divergence Calculation</p>
<section id="arguments-28" class="level4">
<h4 class="anchored" data-anchor-id="arguments-28">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>p</code></td>
<td>A numeric vector representing the probability distribution P.</td>
</tr>
<tr class="even">
<td><code>q</code></td>
<td>A numeric vector representing the probability distribution Q.</td>
</tr>
<tr class="odd">
<td><code>epsilon</code></td>
<td>A small positive constant added to both P and Q to avoid logarithm of zero. Default is 1e-10.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-34" class="level4">
<h4 class="anchored" data-anchor-id="description-34">Description</h4>
<p>This function calculates the Jensen-Shannon Divergence (JSD) between two probability distributions P and Q.</p>
</section>
<section id="details-12" class="level4">
<h4 class="anchored" data-anchor-id="details-12">Details</h4>
<p>The JSD is computed using the Kullback-Leibler Divergence (KLD) as follows: <code>sum((p * log((p + epsilon) / (m + epsilon)) + q * log((q + epsilon) / (m + epsilon))) / 2)</code> where <code>m = (p + q) / 2</code> .</p>
</section>
<section id="seealso-14" class="level4">
<h4 class="anchored" data-anchor-id="seealso-14">Seealso</h4>
<p><a href="#kld"><code>kld</code></a> , <a href="#sum"><code>sum</code></a> , <a href="#log"><code>log</code></a></p>
</section>
<section id="value-34" class="level4">
<h4 class="anchored" data-anchor-id="value-34">Value</h4>
<p>A numeric value representing the Jensen-Shannon Divergence between P and Q.</p>
</section>
<section id="examples-32" class="level4">
<h4 class="anchored" data-anchor-id="examples-32">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb290"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb290-1"><a href="#cb290-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate JSD between two probability distributions</span></span>
<span id="cb290-2"><a href="#cb290-2" aria-hidden="true" tabindex="-1"></a>p_distribution <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">0.3</span>, <span class="fl">0.5</span>)</span>
<span id="cb290-3"><a href="#cb290-3" aria-hidden="true" tabindex="-1"></a>q_distribution <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="dv">0</span>, <span class="fl">0.9</span>)</span>
<span id="cb290-4"><a href="#cb290-4" aria-hidden="true" tabindex="-1"></a>jsd_result <span class="ot">&lt;-</span> <span class="fu">jsd</span>(p_distribution, q_distribution)</span>
<span id="cb290-5"><a href="#cb290-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(jsd_result)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-34" class="level4">
<h4 class="anchored" data-anchor-id="usage-34">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb291"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb291-1"><a href="#cb291-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_jsd</span>(p, q, <span class="at">epsilon =</span> <span class="fl">1e-10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_kld_from_vec" class="level3" data-number="6.5.36">
<h3 data-number="6.5.36" class="anchored" data-anchor-id="lfa_kld_from_vec"><span class="header-section-number">6.5.36</span> <code>lfa_kld_from_vec</code></h3>
<p>Compute Kullback-Leibler Divergence from Vectors</p>
<section id="arguments-29" class="level4">
<h4 class="anchored" data-anchor-id="arguments-29">Arguments</h4>
<table class="table">
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>x</code></td>
<td>A numeric vector.</td>
</tr>
<tr class="even">
<td><code>y</code></td>
<td>A numeric vector.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-35" class="level4">
<h4 class="anchored" data-anchor-id="description-35">Description</h4>
<p>This function calculates the Kullback-Leibler Divergence (KLD) between two vectors.</p>
</section>
<section id="value-35" class="level4">
<h4 class="anchored" data-anchor-id="value-35">Value</h4>
<p>Kullback-Leibler Divergence between the density distributions of x and y.</p>
</section>
<section id="examples-33" class="level4">
<h4 class="anchored" data-anchor-id="examples-33">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb292"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb292-1"><a href="#cb292-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>)</span>
<span id="cb292-2"><a href="#cb292-2" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>, <span class="at">mean =</span> <span class="dv">2</span>)</span>
<span id="cb292-3"><a href="#cb292-3" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_kld_from_vec</span>(x, y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-35" class="level4">
<h4 class="anchored" data-anchor-id="usage-35">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb293"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb293-1"><a href="#cb293-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_kld_from_vec</span>(x, y)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_kld" class="level3" data-number="6.5.37">
<h3 data-number="6.5.37" class="anchored" data-anchor-id="lfa_kld"><span class="header-section-number">6.5.37</span> <code>lfa_kld</code></h3>
<p>Kullback-Leibler Divergence Calculation</p>
<section id="arguments-30" class="level4">
<h4 class="anchored" data-anchor-id="arguments-30">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>p</code></td>
<td>A numeric vector representing the probability distribution P.</td>
</tr>
<tr class="even">
<td><code>q</code></td>
<td>A numeric vector representing the probability distribution Q.</td>
</tr>
<tr class="odd">
<td><code>epsilon</code></td>
<td>A small positive constant added to both P and Q to avoid logarithm of zero. Default is 1e-10.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-36" class="level4">
<h4 class="anchored" data-anchor-id="description-36">Description</h4>
<p>This function calculates the Kullback-Leibler Divergence (KLD) between two probability distributions P and Q.</p>
</section>
<section id="details-13" class="level4">
<h4 class="anchored" data-anchor-id="details-13">Details</h4>
<p>The KLD is computed using the formula: <code>sum(p * log((p + epsilon) / (q + epsilon)))</code> This avoids issues when the denominator (Q) contains zero probabilities.</p>
</section>
<section id="seealso-15" class="level4">
<h4 class="anchored" data-anchor-id="seealso-15">Seealso</h4>
<p><a href="#sum"><code>sum</code></a> , <a href="#log"><code>log</code></a></p>
</section>
<section id="value-36" class="level4">
<h4 class="anchored" data-anchor-id="value-36">Value</h4>
<p>A numeric value representing the Kullback-Leibler Divergence between P and Q.</p>
</section>
<section id="examples-34" class="level4">
<h4 class="anchored" data-anchor-id="examples-34">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb294"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb294-1"><a href="#cb294-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate KLD between two probability distributions</span></span>
<span id="cb294-2"><a href="#cb294-2" aria-hidden="true" tabindex="-1"></a>p_distribution <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">0.3</span>, <span class="fl">0.5</span>)</span>
<span id="cb294-3"><a href="#cb294-3" aria-hidden="true" tabindex="-1"></a>q_distribution <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="dv">0</span>, <span class="fl">0.9</span>)</span>
<span id="cb294-4"><a href="#cb294-4" aria-hidden="true" tabindex="-1"></a>kld_result <span class="ot">&lt;-</span> <span class="fu">kld</span>(p_distribution, q_distribution)</span>
<span id="cb294-5"><a href="#cb294-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(kld_result)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-36" class="level4">
<h4 class="anchored" data-anchor-id="usage-36">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb295"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb295-1"><a href="#cb295-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_kld</span>(p, q, <span class="at">epsilon =</span> <span class="fl">1e-10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_ks_test" class="level3" data-number="6.5.38">
<h3 data-number="6.5.38" class="anchored" data-anchor-id="lfa_ks_test"><span class="header-section-number">6.5.38</span> <code>lfa_ks_test</code></h3>
<p>Kolmogorov-Smirnov Test Wrapper Function</p>
<section id="arguments-31" class="level4">
<h4 class="anchored" data-anchor-id="arguments-31">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>x</code></td>
<td>A numeric vector representing the first sample.</td>
</tr>
<tr class="even">
<td><code>y</code></td>
<td>A numeric vector representing the second sample.</td>
</tr>
<tr class="odd">
<td><code>output_variable</code></td>
<td>A character string specifying the output variable to extract from the ks.test result. Default is “p.value”. Other possible values include “statistic” and “alternative”.</td>
</tr>
<tr class="even">
<td><code>...</code></td>
<td>Additional arguments to be passed to the ks.test function.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-37" class="level4">
<h4 class="anchored" data-anchor-id="description-37">Description</h4>
<p>This function serves as a wrapper for the Kolmogorov-Smirnov (KS) test between two samples.</p>
</section>
<section id="details-14" class="level4">
<h4 class="anchored" data-anchor-id="details-14">Details</h4>
<p>The function uses the ks.test function to perform a two-sample KS test and returns the specified output variable. The default output variable is the p-value. Other possible output variables include “statistic” and “alternative”.</p>
</section>
<section id="seealso-16" class="level4">
<h4 class="anchored" data-anchor-id="seealso-16">Seealso</h4>
<p><a href="#ks.test"><code>ks.test</code></a></p>
</section>
<section id="value-37" class="level4">
<h4 class="anchored" data-anchor-id="value-37">Value</h4>
<p>A numeric value representing the specified output variable from the KS test result.</p>
</section>
<section id="examples-35" class="level4">
<h4 class="anchored" data-anchor-id="examples-35">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb296"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb296-1"><a href="#cb296-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform KS test and extract the p-value</span></span>
<span id="cb296-2"><a href="#cb296-2" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">lfa_ks_test</span>(sample1, sample2)</span>
<span id="cb296-3"><a href="#cb296-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(result)</span>
<span id="cb296-4"><a href="#cb296-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-5"><a href="#cb296-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform KS test and extract the test statistic</span></span>
<span id="cb296-6"><a href="#cb296-6" aria-hidden="true" tabindex="-1"></a>result_statistic <span class="ot">&lt;-</span> <span class="fu">lfa_ks_test</span>(sample1, sample2, <span class="at">output_variable =</span> <span class="st">"statistic"</span>)</span>
<span id="cb296-7"><a href="#cb296-7" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(result_statistic)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-37" class="level4">
<h4 class="anchored" data-anchor-id="usage-37">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb297"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb297-1"><a href="#cb297-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_ks_test</span>(x, y, <span class="at">output_variable =</span> <span class="st">"p.value"</span>, ...)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_load_ctg_if_not_present" class="level3" data-number="6.5.39">
<h3 data-number="6.5.39" class="anchored" data-anchor-id="lfa_load_ctg_if_not_present"><span class="header-section-number">6.5.39</span> <code>lfa_load_ctg_if_not_present</code></h3>
<p>Loading the catalog if it is not present</p>
<section id="arguments-32" class="level4">
<h4 class="anchored" data-anchor-id="arguments-32">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>ctg</code></td>
<td>Catalog object. Can be NULL</td>
</tr>
<tr class="even">
<td><code>tile_location</code></td>
<td>The location to look for the catalog tiles, if their are not present</td>
</tr>
</tbody>
</table>
</section>
<section id="description-38" class="level4">
<h4 class="anchored" data-anchor-id="description-38">Description</h4>
<p>This function checks if the catalog is <code>NULL</code> . If it is it will load the catalog from the <code>tile_location</code></p>
</section>
<section id="value-38" class="level4">
<h4 class="anchored" data-anchor-id="value-38">Value</h4>
<p>The provided ctg object if not null, else the catalog for the tiles of the tile_location.</p>
</section>
<section id="usage-38" class="level4">
<h4 class="anchored" data-anchor-id="usage-38">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb298"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb298-1"><a href="#cb298-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_load_ctg_if_not_present</span>(ctg, tile_location)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_map_tile_locations" class="level3" data-number="6.5.40">
<h3 data-number="6.5.40" class="anchored" data-anchor-id="lfa_map_tile_locations"><span class="header-section-number">6.5.40</span> <code>lfa_map_tile_locations</code></h3>
<p>Map Function Over Tile Locations</p>
<section id="arguments-33" class="level4">
<h4 class="anchored" data-anchor-id="arguments-33">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>tile_locations</code></td>
<td>A list of tile location objects.</td>
</tr>
<tr class="even">
<td><code>map_function</code></td>
<td>The mapping function to be applied to each tile location.</td>
</tr>
<tr class="odd">
<td><code>...</code></td>
<td>Additional arguments to be passed to the mapping function.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-39" class="level4">
<h4 class="anchored" data-anchor-id="description-39">Description</h4>
<p>This function applies a specified mapping function to each tile location in a list.</p>
</section>
<section id="details-15" class="level4">
<h4 class="anchored" data-anchor-id="details-15">Details</h4>
<p>This function iterates over each tile location in the provided list ( <code>tile_locations</code> ) and applies the specified mapping function ( <code>map_function</code> ) to each tile location. The mapping function should accept a tile location object as its first argument, and additional arguments can be passed using the ellipsis ( <code>...</code> ) syntax.</p>
<p>This function is useful for performing operations on multiple tile locations concurrently, such as loading Lidar data, processing areas, or other tasks that involve tile locations.</p>
</section>
<section id="seealso-17" class="level4">
<h4 class="anchored" data-anchor-id="seealso-17">Seealso</h4>
<p>The mapping function provided should be compatible with the structure and requirements of the tile locations and the specific task being performed.</p>
</section>
<section id="value-39" class="level4">
<h4 class="anchored" data-anchor-id="value-39">Value</h4>
<p>None</p>
</section>
<section id="examples-36" class="level4">
<h4 class="anchored" data-anchor-id="examples-36">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb299"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb299-1"><a href="#cb299-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage</span></span>
<span id="cb299-2"><a href="#cb299-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_map_tile_locations</span>(tile_locations, my_mapping_function, <span class="at">param1 =</span> <span class="st">"value"</span>)</span>
<span id="cb299-3"><a href="#cb299-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb299-4"><a href="#cb299-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage</span></span>
<span id="cb299-5"><a href="#cb299-5" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_map_tile_locations</span>(tile_locations, my_mapping_function, <span class="at">param1 =</span> <span class="st">"value"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-39" class="level4">
<h4 class="anchored" data-anchor-id="usage-39">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb300"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb300-1"><a href="#cb300-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_map_tile_locations</span>(tile_locations, map_function, <span class="at">check_flag =</span> <span class="cn">NULL</span>, ...)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_merge_and_save" class="level3" data-number="6.5.41">
<h3 data-number="6.5.41" class="anchored" data-anchor-id="lfa_merge_and_save"><span class="header-section-number">6.5.41</span> <code>lfa_merge_and_save</code></h3>
<p>Merge and Save Text Files in a Directory</p>
<section id="arguments-34" class="level4">
<h4 class="anchored" data-anchor-id="arguments-34">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>input_directory</code></td>
<td>The path to the input directory containing text files.</td>
</tr>
<tr class="even">
<td><code>output_name</code></td>
<td>The name for the output file where the merged content will be saved.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-40" class="level4">
<h4 class="anchored" data-anchor-id="description-40">Description</h4>
<p>This function takes an input directory and an output name as arguments. It merges the textual content of all files in the specified directory into a single string, with each file’s content separated by a newline character. The merged content is then saved into a file named after the output name in the same directory. After the merging is complete, all input files are deleted.</p>
</section>
<section id="details-16" class="level4">
<h4 class="anchored" data-anchor-id="details-16">Details</h4>
<p>This function reads the content of each text file in the specified input directory and concatenates them into a single string. Each file’s content is separated by a newline character. The merged content is then saved into a file named after the output name in the same directory. Finally, all input files are deleted from the directory.</p>
</section>
<section id="seealso-18" class="level4">
<h4 class="anchored" data-anchor-id="seealso-18">Seealso</h4>
<p><a href="#readlines"><code>readLines</code></a> , <a href="#writelines"><code>writeLines</code></a> , <a href="#file.remove"><code>file.remove</code></a></p>
</section>
<section id="value-40" class="level4">
<h4 class="anchored" data-anchor-id="value-40">Value</h4>
<p>This function does not explicitly return any value. It prints a message indicating the successful completion of the merging and saving process.</p>
</section>
<section id="examples-37" class="level4">
<h4 class="anchored" data-anchor-id="examples-37">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb301"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb301-1"><a href="#cb301-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge text files in the "data_files" directory and save the result in "merged_output"</span></span>
<span id="cb301-2"><a href="#cb301-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_merge_and_save</span>(<span class="st">"data_files"</span>, <span class="st">"merged_output"</span>)</span>
<span id="cb301-3"><a href="#cb301-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb301-4"><a href="#cb301-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge text files in the "data_files" directory and save the result in "merged_output"</span></span>
<span id="cb301-5"><a href="#cb301-5" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_merge_and_save</span>(<span class="st">"data_files"</span>, <span class="st">"merged_output"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-40" class="level4">
<h4 class="anchored" data-anchor-id="usage-40">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb302"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb302-1"><a href="#cb302-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_merge_and_save</span>(input_directory, output_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_plot_confusion_matrix" class="level3" data-number="6.5.42">
<h3 data-number="6.5.42" class="anchored" data-anchor-id="lfa_plot_confusion_matrix"><span class="header-section-number">6.5.42</span> <code>lfa_plot_confusion_matrix</code></h3>
<p>Plot Confusion Matrix</p>
<section id="arguments-35" class="level4">
<h4 class="anchored" data-anchor-id="arguments-35">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>conf_matrix</code></td>
<td>Confusion matrix, typically obtained from classification evaluation.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-41" class="level4">
<h4 class="anchored" data-anchor-id="description-41">Description</h4>
<p>This function generates a heatmap plot of a confusion matrix using ggplot2.</p>
</section>
<section id="details-17" class="level4">
<h4 class="anchored" data-anchor-id="details-17">Details</h4>
<p>The function takes a confusion matrix as input and generates a heatmap plot using ggplot2. The plot represents the relationship between the predicted and actual classes, with cell colors indicating the frequency of each combination. Additionally, the plot includes labels for accuracy and kappa statistics based on the confusion matrix.</p>
</section>
<section id="value-41" class="level4">
<h4 class="anchored" data-anchor-id="value-41">Value</h4>
<p>A ggplot object representing the confusion matrix heatmap plot.</p>
</section>
<section id="examples-38" class="level4">
<h4 class="anchored" data-anchor-id="examples-38">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb303"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb303-1"><a href="#cb303-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example confusion matrix</span></span>
<span id="cb303-2"><a href="#cb303-2" aria-hidden="true" tabindex="-1"></a>cm <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="at">predicted =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>), <span class="at">actual =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"B"</span>))</span>
<span id="cb303-3"><a href="#cb303-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot confusion matrix</span></span>
<span id="cb303-4"><a href="#cb303-4" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_plot_confusion_matrix</span>(cm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-41" class="level4">
<h4 class="anchored" data-anchor-id="usage-41">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb304"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb304-1"><a href="#cb304-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_plot_confusion_matrix</span>(conf_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_precision_per_class" class="level3" data-number="6.5.43">
<h3 data-number="6.5.43" class="anchored" data-anchor-id="lfa_precision_per_class"><span class="header-section-number">6.5.43</span> <code>lfa_precision_per_class</code></h3>
<p>Calculate Precision per Class from Confusion Matrix</p>
<section id="arguments-36" class="level4">
<h4 class="anchored" data-anchor-id="arguments-36">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>confusion_matrix</code></td>
<td>Confusion matrix obtained from a classification evaluation.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-42" class="level4">
<h4 class="anchored" data-anchor-id="description-42">Description</h4>
<p>This function calculates precision for each class based on the provided confusion matrix.</p>
</section>
<section id="details-18" class="level4">
<h4 class="anchored" data-anchor-id="details-18">Details</h4>
<p>Precision is a measure of the accuracy of the positive predictions for a specific class. It is calculated as the ratio of true positives to the sum of true positives and false positives.</p>
</section>
<section id="seealso-19" class="level4">
<h4 class="anchored" data-anchor-id="seealso-19">Seealso</h4>
<p><a href="#lfarecallperclass"><code>lfa_recall_per_class</code></a> , <a href="#lfaf1scoreperclass"><code>lfa_f1_score_per_class</code></a></p>
</section>
<section id="value-42" class="level4">
<h4 class="anchored" data-anchor-id="value-42">Value</h4>
<p>A numeric vector representing precision for each class.</p>
</section>
<section id="examples-39" class="level4">
<h4 class="anchored" data-anchor-id="examples-39">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb305"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb305-1"><a href="#cb305-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example confusion matrix</span></span>
<span id="cb305-2"><a href="#cb305-2" aria-hidden="true" tabindex="-1"></a>cm <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="at">predicted =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>), <span class="at">actual =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"B"</span>))</span>
<span id="cb305-3"><a href="#cb305-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate precision per class</span></span>
<span id="cb305-4"><a href="#cb305-4" aria-hidden="true" tabindex="-1"></a>precision_vector <span class="ot">&lt;-</span> <span class="fu">lfa_precision_per_class</span>(cm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-42" class="level4">
<h4 class="anchored" data-anchor-id="usage-42">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb306"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb306-1"><a href="#cb306-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_precision_per_class</span>(confusion_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_random_forest" class="level3" data-number="6.5.44">
<h3 data-number="6.5.44" class="anchored" data-anchor-id="lfa_random_forest"><span class="header-section-number">6.5.44</span> <code>lfa_random_forest</code></h3>
<p>Random Forest Classifier with Leave-One-Out Cross-Validation</p>
<section id="arguments-37" class="level4">
<h4 class="anchored" data-anchor-id="arguments-37">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>tree_data</code></td>
<td>A data frame containing the tree data, including the response variable (“specie”) and predictor variables.</td>
</tr>
<tr class="even">
<td><code>excluded_input_columns</code></td>
<td>A character vector specifying columns to be excluded from predictor variables.</td>
</tr>
<tr class="odd">
<td><code>response_variable</code></td>
<td>The response variable to be predicted (default is “specie”).</td>
</tr>
<tr class="even">
<td><code>seed</code></td>
<td>An integer to set the seed for reproducibility (default is 123).</td>
</tr>
<tr class="odd">
<td><code>...</code></td>
<td>Additional parameters to be passed to the randomForest function.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-43" class="level4">
<h4 class="anchored" data-anchor-id="description-43">Description</h4>
<p>This function performs a random forest classification using leave-one-out cross-validation for each area in the input tree data. It returns a list containing various results, including predicted species, confusion matrix, accuracy, and the formula used for modeling.</p>
</section>
<section id="value-43" class="level4">
<h4 class="anchored" data-anchor-id="value-43">Value</h4>
<p>A list containing the following elements:</p>
<ul>
<li><p><code>predicted_species_absolute</code> : A data frame with observed and predicted species for each area.</p></li>
<li><p><code>predicted_species_relative</code> : A data frame wit the relative precictions per speices and areas, normalized by the total predictions in each area.</p></li>
<li><p><code>confusion_matrix</code> : A confusion matrix showing the counts of predicted vs.&nbsp;observed species.</p></li>
<li><p><code>accuracy</code> : The accuracy of the model, calculated as the sum of diagonal elements in the confusion matrix divided by the total count.</p></li>
<li><p><code>formula</code> : The formula used for modeling.</p></li>
</ul>
</section>
<section id="examples-40" class="level4">
<h4 class="anchored" data-anchor-id="examples-40">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb307"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb307-1"><a href="#cb307-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming tree_data is defined</span></span>
<span id="cb307-2"><a href="#cb307-2" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">lfa_random_forest</span>(tree_data, <span class="at">excluded_input_columns =</span> <span class="fu">c</span>(<span class="st">"column1"</span>, <span class="st">"column2"</span>))</span>
<span id="cb307-3"><a href="#cb307-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb307-4"><a href="#cb307-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the list of results</span></span>
<span id="cb307-5"><a href="#cb307-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-43" class="level4">
<h4 class="anchored" data-anchor-id="usage-43">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb308"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb308-1"><a href="#cb308-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_random_forest</span>(</span>
<span id="cb308-2"><a href="#cb308-2" aria-hidden="true" tabindex="-1"></a>  tree_data,</span>
<span id="cb308-3"><a href="#cb308-3" aria-hidden="true" tabindex="-1"></a>  excluded_input_columns,</span>
<span id="cb308-4"><a href="#cb308-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">response_variable =</span> <span class="st">"specie"</span>,</span>
<span id="cb308-5"><a href="#cb308-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">ntree =</span> <span class="dv">100</span>,</span>
<span id="cb308-6"><a href="#cb308-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">seed =</span> <span class="dv">123</span>,</span>
<span id="cb308-7"><a href="#cb308-7" aria-hidden="true" tabindex="-1"></a>  ...</span>
<span id="cb308-8"><a href="#cb308-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_rasterize_chunk" class="level3" data-number="6.5.45">
<h3 data-number="6.5.45" class="anchored" data-anchor-id="lfa_rasterize_chunk"><span class="header-section-number">6.5.45</span> <code>lfa_rasterize_chunk</code></h3>
<p>Rasterize Lidar Chunk</p>
<section id="arguments-38" class="level4">
<h4 class="anchored" data-anchor-id="arguments-38">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>chunk</code></td>
<td>Lidar chunk object to be rasterized.</td>
</tr>
<tr class="even">
<td><code>...</code></td>
<td>Additional arguments to be passed to the underlying lidR::rasterize_canopy function.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-44" class="level4">
<h4 class="anchored" data-anchor-id="description-44">Description</h4>
<p>This function rasterizes a Lidar chunk to generate a raster representation of the canopy.</p>
</section>
<section id="details-19" class="level4">
<h4 class="anchored" data-anchor-id="details-19">Details</h4>
<p>The function takes a Lidar chunk as input and uses lidR::rasterize_canopy to generate a raster representation of the canopy. Additional arguments can be passed to customize the rasterization process.</p>
</section>
<section id="value-44" class="level4">
<h4 class="anchored" data-anchor-id="value-44">Value</h4>
<p>A raster layer representing the rasterized canopy.</p>
</section>
<section id="examples-41" class="level4">
<h4 class="anchored" data-anchor-id="examples-41">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb309"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb309-1"><a href="#cb309-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example Lidar chunk</span></span>
<span id="cb309-2"><a href="#cb309-2" aria-hidden="true" tabindex="-1"></a>lidar_chunk <span class="ot">&lt;-</span> <span class="fu">readLAS</span>(<span class="st">"lidar_data.las"</span>, <span class="at">select =</span> <span class="st">"xyz"</span>)</span>
<span id="cb309-3"><a href="#cb309-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Rasterize Lidar chunk</span></span>
<span id="cb309-4"><a href="#cb309-4" aria-hidden="true" tabindex="-1"></a>rasterized_canopy <span class="ot">&lt;-</span> <span class="fu">lfa_rasterize_chunk</span>(lidar_chunk)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-44" class="level4">
<h4 class="anchored" data-anchor-id="usage-44">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb310"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb310-1"><a href="#cb310-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_rasterize_chunk</span>(chunk, ...)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_rd_to_qmd" class="level3" data-number="6.5.46">
<h3 data-number="6.5.46" class="anchored" data-anchor-id="lfa_rd_to_qmd"><span class="header-section-number">6.5.46</span> <code>lfa_rd_to_qmd</code></h3>
<p>Convert Rd File to Markdown</p>
<section id="arguments-39" class="level4">
<h4 class="anchored" data-anchor-id="arguments-39">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>rdfile</code></td>
<td>The path to the Rd file or a parsed Rd object.</td>
</tr>
<tr class="even">
<td><code>outfile</code></td>
<td>The path to the output Markdown file (including the file extension).</td>
</tr>
<tr class="odd">
<td><code>append</code></td>
<td>Logical, indicating whether to append to an existing file (default is FALSE).</td>
</tr>
</tbody>
</table>
</section>
<section id="description-45" class="level4">
<h4 class="anchored" data-anchor-id="description-45">Description</h4>
<p>IMPORTANT NOTE: This function is nearly identical to the <code>Rd2md::Rd2markdown</code> function from the <code>Rd2md</code> package. We needed to implement our own version of it because of various reasons:</p>
<ul>
<li><p>The algorithm uses hardcoded header sizes (h1 and h2 in original) which is not feasible for our use-case of the markdown.</p></li>
<li><p>We needed to add some Quarto Markdown specifics, e.g.&nbsp;to make sure that the examples will not be runned.</p></li>
<li><p>We want to exclude certain tags from our implementation.</p></li>
</ul>
</section>
<section id="details-20" class="level4">
<h4 class="anchored" data-anchor-id="details-20">Details</h4>
<p>For that reason we copied the method and made changes as needed and also added this custom documentation.</p>
<p>This function converts an Rd (R documentation) file to Markdown format (.md) and saves the converted file at the specified location. The function allows appending to an existing file or creating a new one. The resulting Markdown file includes sections for the function’s name, title, and additional content such as examples, usage, arguments, and other sections present in the Rd file.</p>
<p>The function performs the following steps:</p>
<ul>
<li><p>Parses the Rd file using the Rd2md package.</p></li>
<li><p>Creates a Markdown file with sections for the function’s name, title, and additional content.</p></li>
<li><p>Appends the content to an existing file if <code>append</code> is set to TRUE.</p></li>
<li><p>Saves the resulting Markdown file at the specified location.</p></li>
</ul>
</section>
<section id="seealso-20" class="level4">
<h4 class="anchored" data-anchor-id="seealso-20">Seealso</h4>
<p><a href="#rd2md::parserd"><code>Rd2md::parseRd</code></a></p>
</section>
<section id="value-45" class="level4">
<h4 class="anchored" data-anchor-id="value-45">Value</h4>
<p>This function does not explicitly return any value. It saves the converted Markdown file at the specified location as described in the details section.</p>
</section>
<section id="examples-42" class="level4">
<h4 class="anchored" data-anchor-id="examples-42">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb311"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb311-1"><a href="#cb311-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert Rd file to Markdown and save it</span></span>
<span id="cb311-2"><a href="#cb311-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_rd_to_md</span>(<span class="st">"path/to/your/file.Rd"</span>, <span class="st">"path/to/your/output/file.md"</span>)</span>
<span id="cb311-3"><a href="#cb311-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb311-4"><a href="#cb311-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert Rd file to Markdown and append to an existing file</span></span>
<span id="cb311-5"><a href="#cb311-5" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_rd_to_md</span>(<span class="st">"path/to/your/file.Rd"</span>, <span class="st">"path/to/existing/output/file.md"</span>, <span class="at">append =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-45" class="level4">
<h4 class="anchored" data-anchor-id="usage-45">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb312"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb312-1"><a href="#cb312-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_rd_to_qmd</span>(rdfile, outfile, <span class="at">append =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_rd_to_results" class="level3" data-number="6.5.47">
<h3 data-number="6.5.47" class="anchored" data-anchor-id="lfa_rd_to_results"><span class="header-section-number">6.5.47</span> <code>lfa_rd_to_results</code></h3>
<p>Convert Rd Files to Markdown and Merge Results</p>
<section id="description-46" class="level4">
<h4 class="anchored" data-anchor-id="description-46">Description</h4>
<p>This function converts all Rd (R documentation) files in the “man” directory to Markdown format (.qmd) and saves the converted files in the “results/appendix/package-docs” directory. It then merges the converted Markdown files into a single string and saves the merged content into a file named “docs.qmd” in the “results/appendix/package-docs” directory.</p>
</section>
<section id="details-21" class="level4">
<h4 class="anchored" data-anchor-id="details-21">Details</h4>
<p>The function performs the following steps:</p>
<ul>
<li><p>Removes any existing “docs.qmd” file in the “results/appendix/package-docs” directory.</p></li>
<li><p>Finds all Rd files in the “man” directory.</p></li>
<li><p>Converts each Rd file to Markdown format (.qmd) using the <code>lfa_rd_to_qmd</code> function.</p></li>
<li><p>Saves the converted Markdown files in the “results/appendix/package-docs” directory.</p></li>
<li><p>Merges the content of all converted Markdown files into a single string.</p></li>
<li><p>Saves the merged content into a file named “docs.qmd” in the “results/appendix/package-docs” directory.</p></li>
</ul>
</section>
<section id="seealso-21" class="level4">
<h4 class="anchored" data-anchor-id="seealso-21">Seealso</h4>
<p><a href="#lfardtoqmd"><code>lfa_rd_to_qmd</code></a> , <a href="#lfamergeandsave"><code>lfa_merge_and_save</code></a></p>
</section>
<section id="value-46" class="level4">
<h4 class="anchored" data-anchor-id="value-46">Value</h4>
<p>This function does not explicitly return any value. It performs the conversion, merging, and saving operations as described in the details section.</p>
</section>
<section id="examples-43" class="level4">
<h4 class="anchored" data-anchor-id="examples-43">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb313"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb313-1"><a href="#cb313-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert Rd files to Markdown and merge the results</span></span>
<span id="cb313-2"><a href="#cb313-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_rd_to_results</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-46" class="level4">
<h4 class="anchored" data-anchor-id="usage-46">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb314"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb314-1"><a href="#cb314-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_rd_to_results</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_read_area_as_catalog" class="level3" data-number="6.5.48">
<h3 data-number="6.5.48" class="anchored" data-anchor-id="lfa_read_area_as_catalog"><span class="header-section-number">6.5.48</span> <code>lfa_read_area_as_catalog</code></h3>
<p>Read LiDAR data from a specified species and location as a catalog.</p>
<section id="arguments-40" class="level4">
<h4 class="anchored" data-anchor-id="arguments-40">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>specie</code></td>
<td>A character string specifying the species of interest.</td>
</tr>
<tr class="even">
<td><code>location_name</code></td>
<td>A character string specifying the name of the location.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-47" class="level4">
<h4 class="anchored" data-anchor-id="description-47">Description</h4>
<p>This function constructs the file path based on the specified <code>specie</code> and <code>location_name</code> , lists the directories at that path, and reads the LiDAR data into a <code>lidR::LAScatalog</code> .</p>
</section>
<section id="value-47" class="level4">
<h4 class="anchored" data-anchor-id="value-47">Value</h4>
<p>A <code>lidR::LAScatalog</code> object containing the LiDAR data from the specified location and species.</p>
</section>
<section id="examples-44" class="level4">
<h4 class="anchored" data-anchor-id="examples-44">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb315"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb315-1"><a href="#cb315-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_read_area_as_catalog</span>(<span class="st">"beech"</span>, <span class="st">"location1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-47" class="level4">
<h4 class="anchored" data-anchor-id="usage-47">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb316"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb316-1"><a href="#cb316-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_read_area_as_catalog</span>(specie, location_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_recall_per_class" class="level3" data-number="6.5.49">
<h3 data-number="6.5.49" class="anchored" data-anchor-id="lfa_recall_per_class"><span class="header-section-number">6.5.49</span> <code>lfa_recall_per_class</code></h3>
<p>Calculate Recall per Class from Confusion Matrix</p>
<section id="arguments-41" class="level4">
<h4 class="anchored" data-anchor-id="arguments-41">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>confusion_matrix</code></td>
<td>Confusion matrix obtained from a classification evaluation.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-48" class="level4">
<h4 class="anchored" data-anchor-id="description-48">Description</h4>
<p>This function calculates recall for each class based on the provided confusion matrix.</p>
</section>
<section id="details-22" class="level4">
<h4 class="anchored" data-anchor-id="details-22">Details</h4>
<p>Recall (Sensitivity or True Positive Rate) is a measure of the ability of a classification model to identify all relevant instances. It is calculated as the ratio of true positives to the sum of true positives and false negatives.</p>
</section>
<section id="seealso-22" class="level4">
<h4 class="anchored" data-anchor-id="seealso-22">Seealso</h4>
<p><a href="#lfaprecisionperclass"><code>lfa_precision_per_class</code></a></p>
</section>
<section id="value-48" class="level4">
<h4 class="anchored" data-anchor-id="value-48">Value</h4>
<p>A numeric vector representing recall for each class.</p>
</section>
<section id="examples-45" class="level4">
<h4 class="anchored" data-anchor-id="examples-45">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb317"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb317-1"><a href="#cb317-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example confusion matrix</span></span>
<span id="cb317-2"><a href="#cb317-2" aria-hidden="true" tabindex="-1"></a>cm <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="at">predicted =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>), <span class="at">actual =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"B"</span>))</span>
<span id="cb317-3"><a href="#cb317-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate recall per class</span></span>
<span id="cb317-4"><a href="#cb317-4" aria-hidden="true" tabindex="-1"></a>recall_vector <span class="ot">&lt;-</span> <span class="fu">lfa_recall_per_class</span>(cm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-48" class="level4">
<h4 class="anchored" data-anchor-id="usage-48">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb318"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb318-1"><a href="#cb318-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_recall_per_class</span>(confusion_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_run_test_asymmetric" class="level3" data-number="6.5.50">
<h3 data-number="6.5.50" class="anchored" data-anchor-id="lfa_run_test_asymmetric"><span class="header-section-number">6.5.50</span> <code>lfa_run_test_asymmetric</code></h3>
<p>Asymmetric Pairwise Test for Categories</p>
<section id="arguments-42" class="level4">
<h4 class="anchored" data-anchor-id="arguments-42">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>data</code></td>
<td>A data frame containing the relevant columns.</td>
</tr>
<tr class="even">
<td><code>data_column</code></td>
<td>A character string specifying the column containing the numerical data.</td>
</tr>
<tr class="odd">
<td><code>category_column</code></td>
<td>A character string specifying the column containing the categorical variable.</td>
</tr>
<tr class="even">
<td><code>test_function</code></td>
<td>A function used to perform the pairwise test between two sets of data. It should accept two vectors of numeric data and additional parameters specified by <code>...</code> . The function should return a numeric value representing the test result.</td>
</tr>
<tr class="odd">
<td><code>...</code></td>
<td>Additional parameters to be passed to the <code>test_function</code> .</td>
</tr>
</tbody>
</table>
</section>
<section id="description-49" class="level4">
<h4 class="anchored" data-anchor-id="description-49">Description</h4>
<p>This function performs an asymmetric pairwise test for categories using a user-defined <code>test_function</code> .</p>
</section>
<section id="details-23" class="level4">
<h4 class="anchored" data-anchor-id="details-23">Details</h4>
<p>The function calculates the test results for each unique combination of categories using the specified <code>test_function</code> . The resulting table is asymmetric, containing the test results for comparisons from the rows to the columns.</p>
</section>
<section id="seealso-23" class="level4">
<h4 class="anchored" data-anchor-id="seealso-23">Seealso</h4>
<p><a href="#outer"><code>outer</code></a> , <a href="#vectorize"><code>Vectorize</code></a></p>
</section>
<section id="value-49" class="level4">
<h4 class="anchored" data-anchor-id="value-49">Value</h4>
<p>A data frame representing the results of the asymmetric pairwise tests between categories.</p>
</section>
<section id="examples-46" class="level4">
<h4 class="anchored" data-anchor-id="examples-46">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb319"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb319-1"><a href="#cb319-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a custom test function</span></span>
<span id="cb319-2"><a href="#cb319-2" aria-hidden="true" tabindex="-1"></a>custom_test_function <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb319-3"><a href="#cb319-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Your test logic here</span></span>
<span id="cb319-4"><a href="#cb319-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Return a numeric result</span></span>
<span id="cb319-5"><a href="#cb319-5" aria-hidden="true" tabindex="-1"></a><span class="fu">return</span>(<span class="fu">mean</span>(x) <span class="sc">-</span> <span class="fu">mean</span>(y))</span>
<span id="cb319-6"><a href="#cb319-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb319-7"><a href="#cb319-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb319-8"><a href="#cb319-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform an asymmetric pairwise test</span></span>
<span id="cb319-9"><a href="#cb319-9" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">lfa_run_test_asymmetric</span>(your_data, <span class="st">"numeric_column"</span>, <span class="st">"category_column"</span>, custom_test_function)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-49" class="level4">
<h4 class="anchored" data-anchor-id="usage-49">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb320"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb320-1"><a href="#cb320-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_run_test_asymmetric</span>(data, data_column, category_column, test_function, ...)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_run_test_symmetric" class="level3" data-number="6.5.51">
<h3 data-number="6.5.51" class="anchored" data-anchor-id="lfa_run_test_symmetric"><span class="header-section-number">6.5.51</span> <code>lfa_run_test_symmetric</code></h3>
<p>Symmetric Pairwise Test for Categories</p>
<section id="arguments-43" class="level4">
<h4 class="anchored" data-anchor-id="arguments-43">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>data</code></td>
<td>A data frame containing the relevant columns.</td>
</tr>
<tr class="even">
<td><code>data_column</code></td>
<td>A character string specifying the column containing the numerical data.</td>
</tr>
<tr class="odd">
<td><code>category_column</code></td>
<td>A character string specifying the column containing the categorical variable.</td>
</tr>
<tr class="even">
<td><code>test_function</code></td>
<td>A function used to perform the pairwise test between two sets of data. It should accept two vectors of numeric data and additional parameters specified by <code>...</code> . The function should return a numeric value representing the test result.</td>
</tr>
<tr class="odd">
<td><code>...</code></td>
<td>Additional parameters to be passed to the <code>test_function</code> .</td>
</tr>
</tbody>
</table>
</section>
<section id="description-50" class="level4">
<h4 class="anchored" data-anchor-id="description-50">Description</h4>
<p>This function performs a symmetric pairwise test for categories using a user-defined <code>test_function</code> .</p>
</section>
<section id="details-24" class="level4">
<h4 class="anchored" data-anchor-id="details-24">Details</h4>
<p>The function calculates the test results for each unique combination of categories using the specified <code>test_function</code> . The resulting table is symmetric, containing the test results for comparisons from the rows to the columns. The upper triangle of the matrix is filled with <code>NA</code> to avoid duplicate results.</p>
</section>
<section id="seealso-24" class="level4">
<h4 class="anchored" data-anchor-id="seealso-24">Seealso</h4>
<p><a href="#outer"><code>outer</code></a> , <a href="#vectorize"><code>Vectorize</code></a></p>
</section>
<section id="value-50" class="level4">
<h4 class="anchored" data-anchor-id="value-50">Value</h4>
<p>A data frame representing the results of the symmetric pairwise tests between categories.</p>
</section>
<section id="examples-47" class="level4">
<h4 class="anchored" data-anchor-id="examples-47">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb321"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb321-1"><a href="#cb321-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a custom test function</span></span>
<span id="cb321-2"><a href="#cb321-2" aria-hidden="true" tabindex="-1"></a>custom_test_function <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb321-3"><a href="#cb321-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Your test logic here</span></span>
<span id="cb321-4"><a href="#cb321-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Return a numeric result</span></span>
<span id="cb321-5"><a href="#cb321-5" aria-hidden="true" tabindex="-1"></a><span class="fu">return</span>(<span class="fu">mean</span>(x) <span class="sc">-</span> <span class="fu">mean</span>(y))</span>
<span id="cb321-6"><a href="#cb321-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb321-7"><a href="#cb321-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb321-8"><a href="#cb321-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform a symmetric pairwise test</span></span>
<span id="cb321-9"><a href="#cb321-9" aria-hidden="true" tabindex="-1"></a>result <span class="ot">&lt;-</span> <span class="fu">lfa_run_test_symmetric</span>(your_data, <span class="st">"numeric_column"</span>, <span class="st">"category_column"</span>, custom_test_function)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-50" class="level4">
<h4 class="anchored" data-anchor-id="usage-50">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb322"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb322-1"><a href="#cb322-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_run_test_symmetric</span>(data, data_column, category_column, test_function, ...)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_save_all_neighbours" class="level3" data-number="6.5.52">
<h3 data-number="6.5.52" class="anchored" data-anchor-id="lfa_save_all_neighbours"><span class="header-section-number">6.5.52</span> <code>lfa_save_all_neighbours</code></h3>
<p>Save Neighbors for All Areas</p>
<section id="arguments-44" class="level4">
<h4 class="anchored" data-anchor-id="arguments-44">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>n</code></td>
<td>The number of nearest trees to find for each tree (default is 100).</td>
</tr>
</tbody>
</table>
</section>
<section id="description-51" class="level4">
<h4 class="anchored" data-anchor-id="description-51">Description</h4>
<p>This function iterates through all detection areas, finds the n nearest trees for each tree, and saves the result to a GeoPackage file for each area.</p>
</section>
<section id="examples-48" class="level4">
<h4 class="anchored" data-anchor-id="examples-48">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb323"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb323-1"><a href="#cb323-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save neighbors for all areas with default value (n=100)</span></span>
<span id="cb323-2"><a href="#cb323-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_save_all_neighbours</span>()</span>
<span id="cb323-3"><a href="#cb323-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb323-4"><a href="#cb323-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Save neighbors for all areas with a specific value of n (e.g., n=50)</span></span>
<span id="cb323-5"><a href="#cb323-5" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_save_all_neighbours</span>(<span class="at">n =</span> <span class="dv">50</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-51" class="level4">
<h4 class="anchored" data-anchor-id="usage-51">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb324"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb324-1"><a href="#cb324-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_save_all_neighbours</span>(<span class="at">n =</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_segmentation" class="level3" data-number="6.5.53">
<h3 data-number="6.5.53" class="anchored" data-anchor-id="lfa_segmentation"><span class="header-section-number">6.5.53</span> <code>lfa_segmentation</code></h3>
<p>Segment the elements of an point cloud by trees</p>
<section id="arguments-45" class="level4">
<h4 class="anchored" data-anchor-id="arguments-45">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>ctg</code></td>
<td>An LASCatalog object. If not null, it will perform the actions on this object, if NULL inferring the catalog from the tile_location</td>
</tr>
<tr class="even">
<td><code>tile_location</code></td>
<td>A tile_location type object holding the information about the location of the catalog. This is used to save the catalog after processing too.</td>
</tr>
</tbody>
</table>
</section>
<section id="author-3" class="level4">
<h4 class="anchored" data-anchor-id="author-3">Author</h4>
<p>Jakob Danel</p>
</section>
<section id="description-52" class="level4">
<h4 class="anchored" data-anchor-id="description-52">Description</h4>
<p>This function will try to to divide the hole point cloud into unique trees. Therefore it is assigning for each chunk of the catalog a <code>treeID</code> for each point. Therefore the algorithm uses the <code>li2012</code> implementation with the following parameters: <code>li2012(dt1 = 2, dt2 = 3, R = 2, Zu = 10, hmin = 5, speed_up = 12)</code> NOTE : The operation is in place and can not be reverted, the old values of the point cloud will be deleted!</p>
</section>
<section id="value-51" class="level4">
<h4 class="anchored" data-anchor-id="value-51">Value</h4>
<p>A catalog where each chunk has additional <code>treeID</code> values indicating the belonging tree.</p>
</section>
<section id="usage-52" class="level4">
<h4 class="anchored" data-anchor-id="usage-52">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb325"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb325-1"><a href="#cb325-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_segmentation</span>(ctg, tile_location)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_set_flag" class="level3" data-number="6.5.54">
<h3 data-number="6.5.54" class="anchored" data-anchor-id="lfa_set_flag"><span class="header-section-number">6.5.54</span> <code>lfa_set_flag</code></h3>
<p>Set a flag to indicate the completion of a specific process.</p>
<section id="arguments-46" class="level4">
<h4 class="anchored" data-anchor-id="arguments-46">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>flag_name</code></td>
<td>A character string specifying the name of the flag file. It should be a descriptive and unique identifier for the process being flagged.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-53" class="level4">
<h4 class="anchored" data-anchor-id="description-53">Description</h4>
<p>This function creates a hidden flag file at a specified location within the working directory to indicate that a particular processing step has been completed. If the flag file already exists, a warning is issued.</p>
</section>
<section id="value-52" class="level4">
<h4 class="anchored" data-anchor-id="value-52">Value</h4>
<p>This function does not have a formal return value.</p>
</section>
<section id="examples-49" class="level4">
<h4 class="anchored" data-anchor-id="examples-49">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb326"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb326-1"><a href="#cb326-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the flag for a process named "data_processing"</span></span>
<span id="cb326-2"><a href="#cb326-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_set_flag</span>(<span class="st">"data_processing"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-53" class="level4">
<h4 class="anchored" data-anchor-id="usage-53">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb327"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb327-1"><a href="#cb327-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_set_flag</span>(flag_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_visit_all_areas" class="level3" data-number="6.5.55">
<h3 data-number="6.5.55" class="anchored" data-anchor-id="lfa_visit_all_areas"><span class="header-section-number">6.5.55</span> <code>lfa_visit_all_areas</code></h3>
<p>Visit All Areas and Apply Preprocessing Function</p>
<section id="arguments-47" class="level4">
<h4 class="anchored" data-anchor-id="arguments-47">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>preprocessing_function</code></td>
<td>The preprocessing function to be applied to each area. It should take specie, area, and additional parameters as inputs.</td>
</tr>
<tr class="even">
<td><code>areas</code></td>
<td>Data frame containing information about different areas, including columns “specie” and “area.”</td>
</tr>
<tr class="odd">
<td><code>...</code></td>
<td>Additional arguments to be passed to the preprocessing function.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-54" class="level4">
<h4 class="anchored" data-anchor-id="description-54">Description</h4>
<p>This function iterates over all specified areas and applies a preprocessing function to each one.</p>
</section>
<section id="details-25" class="level4">
<h4 class="anchored" data-anchor-id="details-25">Details</h4>
<p>The function iterates over all areas specified in the ‘areas’ parameter, and for each area, it applies the provided preprocessing function. The ‘areas’ parameter is expected to be a data frame with columns “specie” and “area,” containing information about different areas to visit. Additional arguments passed via ‘…’ are forwarded to the preprocessing function.</p>
</section>
<section id="seealso-25" class="level4">
<h4 class="anchored" data-anchor-id="seealso-25">Seealso</h4>
<p><a href="#lfagetallareas"><code>lfa_get_all_areas</code></a></p>
</section>
<section id="value-53" class="level4">
<h4 class="anchored" data-anchor-id="value-53">Value</h4>
<p>A list containing the results of applying the preprocessing function to each area.</p>
</section>
<section id="examples-50" class="level4">
<h4 class="anchored" data-anchor-id="examples-50">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb328"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb328-1"><a href="#cb328-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example preprocessing function</span></span>
<span id="cb328-2"><a href="#cb328-2" aria-hidden="true" tabindex="-1"></a>my_preprocessing_function <span class="ot">&lt;-</span> <span class="cf">function</span>(specie, area, ...) {</span>
<span id="cb328-3"><a href="#cb328-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Your preprocessing logic here</span></span>
<span id="cb328-4"><a href="#cb328-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Return the result</span></span>
<span id="cb328-5"><a href="#cb328-5" aria-hidden="true" tabindex="-1"></a><span class="fu">return</span>(result)</span>
<span id="cb328-6"><a href="#cb328-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb328-7"><a href="#cb328-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Visit all areas and apply the preprocessing function</span></span>
<span id="cb328-8"><a href="#cb328-8" aria-hidden="true" tabindex="-1"></a>results_list <span class="ot">&lt;-</span> <span class="fu">lfa_visit_all_areas</span>(my_preprocessing_function)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-54" class="level4">
<h4 class="anchored" data-anchor-id="usage-54">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb329"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb329-1"><a href="#cb329-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_visit_all_areas</span>(preprocessing_function, <span class="at">areas =</span> <span class="fu">lfa_get_all_areas</span>(), ...)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="lfa_visualize_rf_metrics" class="level3" data-number="6.5.56">
<h3 data-number="6.5.56" class="anchored" data-anchor-id="lfa_visualize_rf_metrics"><span class="header-section-number">6.5.56</span> <code>lfa_visualize_rf_metrics</code></h3>
<p>Visualize Precision and Recall Metrics for Random Forest Classification</p>
<section id="arguments-48" class="level4">
<h4 class="anchored" data-anchor-id="arguments-48">Arguments</h4>
<table class="table">
<colgroup>
<col style="width: 44%">
<col style="width: 55%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>metrics_df</code></td>
<td>Data frame containing precision and recall metrics for each class.</td>
</tr>
</tbody>
</table>
</section>
<section id="description-55" class="level4">
<h4 class="anchored" data-anchor-id="description-55">Description</h4>
<p>This function creates a bar plot to visualize precision and recall metrics for each class obtained from a Random Forest classification.</p>
</section>
<section id="details-26" class="level4">
<h4 class="anchored" data-anchor-id="details-26">Details</h4>
<p>The function creates a bar plot to visualize precision and recall metrics for each class obtained from a Random Forest classification.</p>
</section>
<section id="seealso-26" class="level4">
<h4 class="anchored" data-anchor-id="seealso-26">Seealso</h4>
<p><a href="#lfacalculaterfmetrics"><code>lfa_calculate_rf_metrics</code></a></p>
</section>
<section id="value-54" class="level4">
<h4 class="anchored" data-anchor-id="value-54">Value</h4>
<p>A ggplot object representing the bar plot of precision and recall metrics.</p>
</section>
<section id="examples-51" class="level4">
<h4 class="anchored" data-anchor-id="examples-51">Examples</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb330"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb330-1"><a href="#cb330-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example data frame containing precision and recall metrics</span></span>
<span id="cb330-2"><a href="#cb330-2" aria-hidden="true" tabindex="-1"></a>example_metrics_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb330-3"><a href="#cb330-3" aria-hidden="true" tabindex="-1"></a><span class="at">Class =</span> <span class="fu">c</span>(<span class="st">"ClassA"</span>, <span class="st">"ClassB"</span>),</span>
<span id="cb330-4"><a href="#cb330-4" aria-hidden="true" tabindex="-1"></a><span class="at">Precision =</span> <span class="fu">c</span>(<span class="fl">0.85</span>, <span class="fl">0.92</span>),</span>
<span id="cb330-5"><a href="#cb330-5" aria-hidden="true" tabindex="-1"></a><span class="at">Recall =</span> <span class="fu">c</span>(<span class="fl">0.78</span>, <span class="fl">0.88</span>)</span>
<span id="cb330-6"><a href="#cb330-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb330-7"><a href="#cb330-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize precision and recall metrics</span></span>
<span id="cb330-8"><a href="#cb330-8" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_visualize_rf_metrics</span>(example_metrics_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="usage-55" class="level4">
<h4 class="anchored" data-anchor-id="usage-55">Usage</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb331"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb331-1"><a href="#cb331-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lfa_visualize_rf_metrics</span>(metrics_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>
</section>
</section>
</section>


<div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p>https://www.opengeodata.nrw.de/produkte/geobasis/hm/3dm_l_las/3dm_l_las/, last visited 7th Dec 2023<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>https://github.com/joheisig/GEDIcalibratoR, last visited 7th Dec 2023<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>